---
title: "Fibrex Compositional Analysis"
author: "Claus Lykkebo"
date: "`r Sys.Date()`"
output:
  rmdformats::readthedown:
    toc_depth: 4
    collapsed: false
    code_folding: hide
    number_sections: true
knit: (function(inputFile, encoding) { 
    rmarkdown::render(
        inputFile, encoding = encoding,
        output_dir = file.path(dirname(inputFile),"output"),
        output_file = paste0("Fibrex_", Sys.Date(), "_differential_abundance.html")) 
    })
bibliography: references.bib
link-citations: true
params:
    input: "R_objects/Phyloseq.Rdata"
    meta: "input/metadata_fibrex.csv"
    batch: "Run"
    indeces: "Observed|Shannon|FaithPD|Chao1"
    COL: !r c("A_CTRL" = "#a5cee3","A_PFOS" = "#1778b6","B_CTRL" = "#b4d88a","B_PFOS" = "#30a148")
---

# GMH ASV ANALYSIS PIPELINE

This Rmarkdown contains the commands necessary to perform initial analyses of the output from the DF_GMH_PIPELINE There will be things that should be updated to fit the exact project. It is important that metadata is loaded correctly and it should contain a variable identifying negative controls and mock samples. I recommend visiting the ["Analysis of community ecology data in R"](https://www.davidzeleny.net/anadat-r/doku.php/en:start) to read about the theory behind the alpha and beta diversity and examples of the necessary R-code to execute them. For analyses of differential abundance I will use the DAtest package [(Russel et al., 2018)](https://www.biorxiv.org/content/10.1101/241802v1). Another excellent source of help is the R [cheat-sheets](https://www.rstudio.com/resources/cheatsheets/).

```{r setup, eval=TRUE, echo=TRUE, message=FALSE,warning = FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Load libraries
library(tidyverse)
library(phyloseq)
library(decontam)
library(pals)
library(ggpubr)
library(rstatix)
library(vegan)
library(reshape2)
library(DAtest)
library(ape)
library(kableExtra)
library(plotly)
library(magick)
library(forcats)

# Create used folders if missing
if (!file.exists("R_objects")) dir.create(file.path(getwd(), "R_objects"))
if (!file.exists("plots")) dir.create(file.path(getwd(), "plots"))
if (!file.exists("plots/cabu")) dir.create(file.path(getwd(), "plots/cabu"))
if (!file.exists("plots/dabu")) dir.create(file.path(getwd(), "plots/dabu"))
if (!file.exists("tables")) dir.create(file.path(getwd(), "tables"))
if (!file.exists("scripts")) dir.create(file.path(getwd(), "scripts"))

saveRDS(params, "R_objects/comp_params.RDS")
```

## SCRIPTS {.tabset .tabset-fade .tabset-pills}

### INFO

This section contains the scripts for the custom functions used in this pipeline

### FILTER TAXA

These functions are used to merge ASV that are not one of the most abundant, or below a defined threshold, into "Other"

```{r merge_taxa_scripts, eval=TRUE, echo=TRUE}

# This function merges anything not in the top (10 by default) most abundant ASVs
merge_less_than_top <- function(pobject, top=10){
  # transform to sample counts to relative values
  transformed <- transform_sample_counts(pobject, function(x) x/sum(x))

    # Orientate and export otu table
  if (taxa_are_rows(transformed)) otu.table <- as.data.frame(otu_table(transformed)) else otu.table <- as.data.frame(t(otu_table(transformed)))
  
  # sort otu table by decreasing abundance
  otu.sort <- otu.table[order(rowMeans(otu.table), decreasing = TRUE),]
  
  # Create list of ASVs to merge
  otu.list <- row.names(otu.sort[(top+1):nrow(otu.sort),])
  if(any("Others" %in% rownames(otu.table))) otu.list <- c(otu.list,"Others")
  
  # perform the merger with the original sample counts
  merged <- merge_taxa(pobject, otu.list, 1)
  
  # change the taxa name
  taxa_names(merged)[taxa_names(merged) %in% otu.list[1]] <- "Others"
  
  # change taxa to "Other" for all levels not being NAs
  for (i in 1:ncol(tax_table(merged))){
    if (sum(!is.na(tax_table(merged)[,i]))){
      tax_table(merged)["Others",i] <- "Others"
    }
  }
  return(merged)
}

# This function merges ASVs that are below a defined ratio, with the default being 0.001 (0.1%)
merge_low_abundance <- function(pobject, threshold=0.001){
  
  # transform to sample counts to relative values
  transformed <- transform_sample_counts(pobject, function(x) x/sum(x))
  
  # Orientate and export otu table
  if (taxa_are_rows(transformed)) otu.table <- as.data.frame(otu_table(transformed)) else otu.table <- as.data.frame(t(otu_table(transformed)))
  
  # Create list of ASVs to merge
  otu.list <- row.names(otu.table[rowMeans(otu.table) < threshold,])
  if(any("Others" %in% rownames(otu.table))) otu.list <- c(otu.list,"Others")

  # perform the merger with the original sample counts
  merged <- merge_taxa(pobject, otu.list, 1)
  
  # change the taxa name
  taxa_names(merged)[taxa_names(merged) %in% otu.list[1]] <- "Others"
  
  # change taxa to "Other" for all levels not being NAs
  for (i in 1:ncol(tax_table(merged))){
    if (sum(!is.na(tax_table(merged)[,i]))){
      tax_table(merged)["Others",i] <- "Others"
    }
  }
  return(merged)
}

merge_prevalence <- function(pobject, variable, min.samples){
  # transform to sample counts to relative values
  if (taxa_are_rows(transformed)) otu.table <- as.data.frame(otu_table(transformed)) else otu.table <- as.data.frame(t(otu_table(transformed)))
  
  # make binary
  otu.table[otu.table != 0] <- 1
  
  # extract metadata
  dat <- data.frame(sample_data(pobject))
  
  # set groups
  vgroup <- unique(dat[,variable])
  
  # create count table
  counts <- data.frame(row.names = row.names(otu.table))
  for (i in seq(length(vgroup))) counts[,vgroup[i]] <- rowSums(otu.table[,dat[,variable]==vgroup[i]])
  
  counts$keep <- ifelse(rowSums(counts) > min.sample*length(vgroup), TRUE, FALSE)
  
  # Create list of ASVs to merge
  otu.list <- row.names(counts[!counts$keep,])
  if(any("Others" %in% rownames(otu.table))) otu.list <- c(otu.list,"Others")

  # perform the merger with the original sample counts
  merged <- merge_taxa(pobject, otu.list, 1)
  
  # change the taxa name
  taxa_names(merged)[taxa_names(merged) %in% otu.list[1]] <- "Others"
  
  # change taxa to "Other" for all levels not being NAs
  for (i in 1:ncol(tax_table(merged))){
    if (sum(!is.na(tax_table(merged)[,i]))){
      tax_table(merged)["Others",i] <- "Others"
    }
  }
  return(merged)
  
}
# save functions
save(merge_less_than_top, merge_low_abundance, file = "scripts/merge_abundance.Rdata")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)) #will clear all objects includes hidden objects.
invisible(gc()) #free up memory and report the memory usage.

```

# COMPOSITIONAL DIFFERENCES

The last part of the initial analyses is to compare the composition of the samples and test the differential abundance of individual ASVs or taxa.

## AGGLOMERATE DATA - All data

Depending on the project and data it might be relevant to agglomerate data at species or higher taxonomic level for this type of analysis.

```{r agglomerate, eval=TRUE, echo=TRUE}
params <- readRDS("R_objects/comp_params.RDS")
# load
load(params$input)

# agglomerate at species and genus level
phy.sp <- tax_glom(phy, taxrank = "Species")
phy.ge <- tax_glom(phy, taxrank = "Genus")
phy.fa <- tax_glom(phy, taxrank = "Family")
phy.or <- tax_glom(phy, taxrank = "Order")
phy.cl <- tax_glom(phy, taxrank = "Class")
phy.ph <- tax_glom(phy, taxrank = "Phylum")

# save agglomerated phyloseq objects
save(phy.sp, phy.ge, phy.fa, phy.or, phy.cl, phy.ph, file = "R_objects/Agglomerated_all.RData")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## AGGLOMERATE DATA - ITERATIONS

Depending on the project and data it might be relevant to agglomerate data at species or higher taxonomic level for this type of analysis.

```{r agglomerate_day, eval=TRUE, echo=TRUE}
params <- readRDS("R_objects/comp_params.RDS")
# load
load(params$input)
str.agg <- c("Feces","Feces_d0","Feces_d2","Feces_d4","Feces_d8","Feces_d0d8","Feces_d2d8","Cecum_d8","Ileum_d8","Feces_CTRL_d8", "Feces_PFOS_d8","Feces_VANPFOS_d8","Feces_VAN_d8","Feces_CTRL_all", "Feces_PFOS_all","Feces_VANPFOS_all","Feces_VAN_all")

for (SUBNAME in str.agg) {
  # Determine material parameter
  if (grepl("Feces",SUBNAME) == TRUE) {
    MTRL <- "Feces"
    mn <- MTRL
  } else if (grepl("Cecum", SUBNAME) == TRUE) {
    MTRL <- "Cecum"
    mn <- MTRL
  } else if (grepl("Ileum", SUBNAME) == TRUE) {
    MTRL <- "Ileum"
    mn <- "MTRL"
  }
  # Determine parameters for differential abundance on specific days
  if (grepl("_d0d8", SUBNAME) ==TRUE) {
    DAY <- c("d0","d8")
    dn <- "day08"
  } else if (grepl("_d2d8",SUBNAME) == TRUE) {
    DAY <- c("d2","d8")
    dn <- "day2+8"
  } else if (grepl("_d0",SUBNAME) == TRUE) {
    DAY <- "d0"
    dn <- "day0"
  } else if (grepl("_d2",SUBNAME) == TRUE) {
    DAY <- "d2"
    dn <- "day2"
  } else if (grepl("_d4",SUBNAME) == TRUE) {
    DAY <- "d4"
    dn <- "day4"
  } else if (grepl("_d8",SUBNAME) == TRUE) {
    DAY <- "d8"
    dn <- "day8"
  } else {
    DAY <- c("d0","d2","d4","d8")
    dn <- "all_days"
  }
  # Determine parameters for differential abundance over time
  if (grepl("_CTRL",SUBNAME) == TRUE) {
    TREAT <- "CTRL"
  } else if (grepl("_PFOS",SUBNAME) == TRUE) {
    TREAT <- "PFOS"
  } else if (grepl("_VANPFOS",SUBNAME) == TRUE) {
    TREAT <- "VAN+PFOS"
  } else if (grepl("_VAN",SUBNAME) == TRUE) {
    TREAT <- "VAN"
  } else {
    TREAT <- ""
    print("No treatment specified")
  }
  
# MTRL <- "Feces"
# DAY <- c("d2","d8")
# VAN <- "van"
#Subset data
  if (TREAT == "") {
    phy.sub <- subset_samples(phy, material == MTRL & day %in% DAY) 
  } else if (TREAT != "") {
    phy.sub <- subset_samples(phy, material == MTRL & day %in% DAY & treatment == TREAT) 
  }

# agglomerate at species and genus level
phy.sp <- tax_glom(phy.sub, taxrank = "Species")
phy.ge <- tax_glom(phy.sub, taxrank = "Genus")
phy.fa <- tax_glom(phy.sub, taxrank = "Family")
phy.or <- tax_glom(phy.sub, taxrank = "Order")
phy.cl <- tax_glom(phy.sub, taxrank = "Class")
phy.ph <- tax_glom(phy.sub, taxrank = "Phylum")

# save agglomerated phyloseq objects

save(phy.sp, phy.ge, phy.fa, phy.or, phy.cl, phy.ph, file = paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## CREATE TAXA PLOTS for subsets {.tabset .tabset-fade .tabset-pills}
The following chunks create relative abundance plots for agglomerations created above. 

### RELATIVE ABUNDANCE OVERVIEW
Relative abundance plots for Feces for all days and for day 0+8.

```{r plot_taxa_overview, eval=TRUE, echo=TRUE}
params <- readRDS("R_objects/comp_params.RDS")

# Function to sort by phylogenetic order
sort_taxa <- function(dat) {
  # set columns
  tax.levels <- c("Kingdom", "Phylum", "Class","Order","Family","Genus","Species", "Taxa")
  tax.use <- tax.levels[tax.levels %in% colnames(dat)]
  # Sort data
  dat.ordered <- dat %>% arrange_at(tax.use)
  dat.ordered <- as.data.frame(dat.ordered)
  # extract sorted levels
  new.levels <- as.character(unique(dat.ordered[,last(tax.use)]))
  dat.ordered[,last(tax.use)] <- factor(dat.ordered[,last(tax.use)], levels = new.levels)
  return(dat.ordered)
  }

# OVERVIEW
str.agg <- c("Feces","Feces_d0d8")
LVLS <- c("Phylum","Class","Order","Family","Genus","Species")

for (SUBNAME in str.agg) {
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  load("scripts/merge_abundance.Rdata")
  if (!file.exists(paste0("plots/cabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/cabu/",SUBNAME)))
  for (LVL in LVLS) {
  # Transform sample counts to abundances
    if (LVL == "Phylum") {
      phy.rel <- transform_sample_counts(phy.ph, fun = function(x) x/sum(x)*100)
    } else if (LVL == "Class") {
      phy.rel <- transform_sample_counts(phy.cl, fun = function(x) x/sum(x)*100)
    } else if (LVL == "Order") {
      phy.rel <- transform_sample_counts(phy.or, fun = function(x) x/sum(x)*100)
    } else if (LVL == "Family") {
      phy.rel <- transform_sample_counts(phy.fa, fun = function(x) x/sum(x)*100)
    } else if (LVL == "Genus") {
      phy.rel <- transform_sample_counts(phy.ge, fun = function(x) x/sum(x)*100)
    } else if (LVL == "Species") {
      phy.rel <- transform_sample_counts(phy.sp, fun = function(x) x/sum(x)*100)
    }
    phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)
    dat <- suppressWarnings(psmelt(phy.top))
    colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

    # Sort taxa
    dat.sort <- sort_taxa(dat)
    
    # rename taxa
    levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

    # Create plot treatment and day
    p <- ggplot(dat.sort, aes(x = day, 
                         y = Abundance, 
                         fill = Taxa, 
                         color = Taxa)) +
      geom_col(position = "fill") +
      xlab("") + 
      facet_grid(.~treatment,space = "free_x",scales = "free_x") +
      ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") +
      theme_pubr(legend = "right") +
  scale_x_discrete(breaks=c("d0","d2","d4","d8"), labels=c("0","2","4","8"))
      p
    suppressMessages(ggsave(filename = paste0("plots/cabu/",SUBNAME,"/",SUBNAME,"_",LVL,"_barplot_timetreat.pdf"), plot = p, device = "pdf", units = "mm", width = 200, height = 150, dpi = 300))
    suppressMessages(ggsave(filename = paste0("plots/cabu/",SUBNAME,"/",SUBNAME,"_",LVL,"_barplot_timetreat.png"), plot = p, device = "png", units = "mm", width = 200, height = 150, dpi = 300))
    
    # print tax table
    if (LVL == "Phylum") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Class") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Order") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Family") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Genus") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Species") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus, Species) %>% summarise(Abundance = mean(Abundance))
    }
    try(kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
      kable_classic(full_width = F, position = "left") %>%
      save_kable(file = paste0("plots/cabu/",SUBNAME,"/",SUBNAME,"_",LVL,"_complist_timetreat.pdf")))
    
    # Create plot individual treatment
    day.labs <- c("Day 0","Day 2","Day 4","Day 8")
    names(day.labs) <- c("d0","d2","d4","d8")
    p <- ggplot(dat.sort, aes(x = rat_name, 
                         y = Abundance, 
                         fill = Taxa, 
                         color = Taxa)) +
      geom_col(position = "fill") +
      xlab("") + 
      facet_grid(day~treatment,space = "free_x",scales = "free_x", labeller = labeller(day = day.labs)) +
      ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") +
      theme_pubr(legend = "right") + rotate_x_text(angle = 90)
    p
    suppressMessages(ggsave(filename = paste0("plots/cabu/",SUBNAME,"/",SUBNAME,"_",LVL,"_barplot_rats_treat.pdf"), plot = p, device = "pdf", units = "mm", width = 200, height = 150, dpi = 300))
    suppressMessages(ggsave(filename = paste0("plots/cabu/",SUBNAME,"/",SUBNAME,"_",LVL,"_barplot_rats_treat.png"), plot = p, device = "png", units = "mm", width = 200, height = 150, dpi = 300))
    
    # print tax table
    if (LVL == "Phylum") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Class") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Order") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Family") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Genus") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus) %>% summarise(Abundance = mean(Abundance))
    } else if (LVL == "Species") {
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus, Species) %>% summarise(Abundance = mean(Abundance))
    }
      
    try(kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
      kable_classic(full_width = F, position = "left") %>%
      save_kable(file = paste0("plots/cabu/",SUBNAME,"/",SUBNAME,"_",LVL,"_complist_rats_treat.pdf")))
    
    print(paste0("Processing ",SUBNAME," on level ",LVL," done!"))
  }
  print(paste0("Finished processing ",SUBNAME,". Moving on..."))
}
    
print("Finished everything on the list! Bye bye :-)")
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```


# DIFFERENTIAL ABUNDANCE

## FECES DAY 8 TREATMENT - DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "all"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
#phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "all"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
#phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 400, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 400, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```







### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "all"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
#phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, ,effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lao(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

## FECES DAY 8 CTRL vs. VAN - DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```







### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, ,effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### ORDER
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Order"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.or <- subset_samples(phy.or, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.or, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Order, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### CLASS
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Class"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.cl <- subset_samples(phy.cl, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.cl, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Class, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### PHYLUM
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Phylum"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ph <- subset_samples(phy.ph, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ph, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Phylum, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## FECES DAY 8 VAN vs. VAN+PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0("SPECIES: Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### ORDER
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Order"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.or <- subset_samples(phy.or, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.or, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Order, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### CLASS
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Class"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.cl <- subset_samples(phy.cl, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.cl, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Class, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### PHYLUM
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Phylum"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ph <- subset_samples(phy.ph, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ph, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Phylum, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

## Figure for F8 PFOS effect
``` {r dabu_f8_pfos_effect, eval=TRUE, echo=TRUE, error=TRUE}
# Using Welch t-test (TTA) as test for all days on Genus level based on significant divergences from Family level
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
SUBNAME <- "Feces_d8"
PDI <- "treatment"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# "fams" is a manual list of all families generating significant results from differential abundance tests above on either family or lower taxonomical levels.
fams <- c("Veillonellaceae",
"Streptococcaceae",
"Staphylococcaceae",
"Ruminococcaceae",
"Phylum_Firmicutes",
"Phylum_Actinobacteria",
"Peptostreptococcaceae",
"Micrococcaceae",
"Lactobacillaceae",
"Erysipelotrichaceae",
"Erysipelatoclostridiaceae",
"Enterococcaceae",
"Enterobacteriaceae",
"Eggerthellaceae",
"Clostridiales_Incertae_Sedis_XIII",
"Corynebacteriaceae",
"Coriobacteriaceae",
"Bifidobacteriaceae",
"Bacillaceae_1",
"Aerococcaceae",
"Ruminococcaceae",
"Erysipelotrichaceae",
"Desulfovibrionaceae",
"Synergistaceae")

# Genera in Families showing significant differences in either CTRL <> PFOS and VAN <> VAN+PFOS
LVL <- "Genus"
for (TX in fams) {
  print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
  p.dT <-  p + ggtitle(paste0("Abundance of ",LVL," taxa in the ",TX," family for ",MTRL," ",DAY))
  p.dT
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
}

# Species in Families showing significant differences in either CTRL <> PFOS and VAN <> VAN+PFOS
LVL <- "Species"
for (TX in fams) {
  print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Species, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
  p.dT <-  p + ggtitle(paste0("Abundance of ",LVL," taxa in the ",TX," family for ",MTRL," ",DAY))
  p.dT
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
}


```
## Figure for specific families + genera
``` {r eval=TRUE, echo=TRUE}
# Collected figures for Families showing significant differences
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
SUBNAME <- "Feces_d8"
PDI <- "treatment"
fams <- c("Erysipelotrichaceae",
"Enterobacteriaceae")

#for (TX in fams) {
  LVL <- "Genus"
  #print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.aoa(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Sort samples to phylogenetic levels (alphabetically)
t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
genus.sorted <- as.character(unique(t.sorted$Genus))
t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)

levels(t.melt$genus_new) %>% head
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + 
    geom_boxplot() + 
    scale_y_log10() + 
    coord_flip() +
    scale_color_manual(values = params$COL) + 
    theme_bw() +
    facet_grid(Family ~ ., scales = "free_y", space = "free") +
    labs(color = "Treatment")
  p.ge <-  p #+ ggtitle(paste0("Abundance of ",LVL," taxa"))
  p.ge
  
    suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.ge, device = "pdf", units = "mm", width = 200, height = 250, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_",SUBNAME,"_",PDI,"_",LVL,".png"), p.ge, device = "png", units = "mm", width = 200, height = 250, dpi = 300))
  # # Species
  # LVL <- "Species"
  # phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY)
  # # Filter data
  # filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)
  # # Run tha selected analysis TTA
  # filt.DA <- DA.tta(filt, predictor = PDI)
  # # Create a subset of the samples
  # filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  # filt.t <- subset_taxa(filt.ra, Family == TX)
  # # Melt the data
  # t.melt <- psmelt(filt.t)
  # # Create pseudocounts
  # pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # # Plot data and save as output formats
  # p <- ggplot(t.melt, aes(Species, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + theme(axis.text.y = element_text(angle = 90))
  # p.sp <-  p + ggtitle(paste0("Abundance of ",LVL," taxa in the ",TX," family for ",MTRL," ",DAY))
  ####
  # CLEAN UP UNDER THIS LINE
  # Family
  LVL <- "Family"
  phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.aoa(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  t.sorted <- t.melt %>% select(Phylum:Family) %>% arrange(Phylum, Class, Order, Family)
family.sorted <- as.character(unique(t.sorted$Family))
t.melt$family_new <- factor(t.melt$Family, levels = family.sorted)

levels(t.melt$family_new) %>% head
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(family_new, Abundance+pseudocount, color = fct_rev(treatment))) + 
    geom_boxplot() + 
    scale_y_log10() + 
    coord_flip() + 
    scale_color_manual(values = params$COL) + 
    facet_grid(. ~ Family , scales = "free_y", space = "free") +
        labs(color = "Treatment")
    #theme(axis.text.y = element_text(angle = 90))
  p.fa <-  p #+ ggtitle(paste0("Abundance of ",LVL," taxa"))
  p.legend <- get_legend(p.fa)
  # p.done <- ggdraw() +
  #     draw_plot(p.fa, 0, 0.1, 0.5, 0.9) + 
  #     draw_plot(p.ge, 0.5, 0.1, 0.5, 0.9) +
  #     draw_grob(p.legend, 0, 0, 1, .1) +
  #     draw_plot_label("Family") + 
  #     draw_plot_label("Genus", x = 0.5) + 
  #     theme(plot.background = element_rect(fill="white", color = NA))
  # p.done
  done <- plot_grid(p.fa, p.ge, ncol = 1, align = "v", axis = "l", rel_heights = c(1,5))
  done
  # p.all <- ggarrange(p.fa, p.ge, labels = c("A","B"), ncol = 1, nrow = 2, common.legend = TRUE)
  # p.all
  
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_done_",SUBNAME,"_",PDI,".pdf"), done, device = "pdf", units = "mm", width = 300, height = 300, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_done_",SUBNAME,"_",PDI,".png"), done, device = "png", units = "mm", width = 300, height = 300, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
#}
```


``` {r , eval=TRUE, echo=TRUE}

# Plot for all taxonomy in phyla Actinobacteria and Firmicutes  
phs <- c("Firmicutes","Actinobacteria")  

for (TX in phs) {
  LVL <- "Family"
  print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Family, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.fa <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  
  LVL <- "Genus"
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.ge <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  
  # Species
  LVL <- "Species"
  phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Species, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.sp <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  
  p.all <- ggarrange(p.fa, p.ge, p.sp, labels = c("A","B","C"), ncol = 3, nrow = 1, common.legend = TRUE)
  p.allT <- annotate_figure(p.all, top = text_grob(paste0("Significant differential abundance for ",TX," in ", MTRL," ",DAY)))
  
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotphylum_",SUBNAME,"_",PDI,"_",TX,".pdf"), p.allT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotphylum_",SUBNAME,"_",PDI,"_",TX,".png"), p.allT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
}

# Plot for all phyla
  LVL <- "Phylum"
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ph <- subset_samples(phy.ph, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ph, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  #filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.ra)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Phylum, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.ph <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  p.ph

  # clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## Plot for significant genera in all groups
``` {r sig_genera, eval=TRUE, echo=TRUE}
SUBNAME <- "Feces_d8"  
MTRL <- "Feces"
DAY <- "d8"
PDI <- "treatment"


#gen.s <- c("Acetatifactor","Family_Clostridiales_Incertae_Sedis_XIII","Family_Desulfovibrionaceae","Family_Enterobacteriaceae","Family_Erysipelotrichaceae","Holdemania","Mucinivorans","Intestinimonas","Anaerotruncus","Breznakia","Clostridium_XIVa","Dysosmobacter","Enterococcus","Flavonifractor","Lachnospira","Marvinbryantia","Neglecta","Oscillibacter","Pseudoflavonifractor")

gen.s <- c("Pseudoflavonifractor","Oscillibacter","Neglecta","Marvinbryantia","Lachnospira","Flavonifractor","Enterococcus","Dysosmobacter","Clostridium_XIVa","Breznakia","Anaerotruncus","Intestinimonas","Mucinivorans","Holdemania","Family_Erysipelotrichaceae","Family_Enterobacteriaceae","Family_Desulfovibrionaceae","Family_Clostridiales_Incertae_Sedis_XIII","Acetatifactor")

LVL <- "Genus"
# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# Filter samples based on variables given
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
# Run tha selected analysis TTA
filt.DA <- DA.tta(filt, predictor = PDI)
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.t <- subset_taxa(filt.ra, Genus %in% gen.s)
# Melt the data
t.melt <- psmelt(filt.t)
# Create pseudocounts
pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
# Reorder to sample list above
t.melt$Genus <- factor(t.melt$Genus, levels = gen.s)
# Sort samples to phylogenetic levels (alphabetically)
# t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
# genus.sorted <- as.character(unique(t.sorted$Genus))
# t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)
# 
# levels(t.melt$genus_new) %>% head

# Plot data and save as output formats
p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + guides(color = guide_legend(title = "Treatment")) + theme_bw()
            
p.ph <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))

p.ph
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_allgenus_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.ph, device = "pdf", units = "mm", width = 200, height = 300, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_allgenus_",SUBNAME,"_",PDI,"_",LVL,".png"), p.ph, device = "png", units = "mm", width = 200, height = 300, dpi = 300))

  # clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```



## Plot for high relative abundance taxa (GENUS)
``` {r sig_genera, eval=TRUE, echo=TRUE}
SUBNAME <- "Feces_d8"  
MTRL <- "Feces"
DAY <- "d8"
PDI <- "treatment"

GENS <- c("Bifidobacterium",
"Bacteroides",
"Parabacteroides",
"Prevotellamassilia",
"Lactobacillus",
"Ligilactobacillus",
"Family_Lachnospiraceae",
"Order_Clostridiales",
"Family_Ruminococcaceae",
"Intestinimonas",
"Holdemania",
"Turicibacter",
"Escherichia/Shigella",
"Blautia")

LVL <- "Genus"
# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# Filter samples based on variables given
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
#filt.test <- testDA(filt, predictor = PDI, effectSize = 10)
# Run tha selected analysis TTA
filt.DA <- DA.aoa(filt, predictor = PDI)
filt.DA %>% select(Genus,pval.adj) %>% head(100)
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.t <- subset_taxa(filt.ra, Genus %in% GENS)
# Melt the data
t.melt <- psmelt(filt.t)
# Create pseudocounts
pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
# Reorder to sample list above
#t.melt$Genus <- factor(t.melt$Genus, levels = gen.s)
# Sort samples to phylogenetic levels (alphabetically)
t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
genus.sorted <- as.character(unique(t.sorted$Genus))
t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)

#levels(t.melt$genus_new) %>% head

# Plot data and save as output formats
p <- ggplot(t.melt, aes(genus_new, Abundance+pseudocount, color = fct_rev(treatment))) + 
  geom_boxplot() + 
  scale_y_log10() + 
  coord_flip() + 
  scale_color_manual(values = params$COL) + 
  facet_grid(Phylum ~ . , scales = "free_y", space = "free") + 
  guides(color = guide_legend(title = "Treatment")) +
  theme_bw()
            
#p.ph <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
p

  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_hi_relabundance_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p, device = "pdf", units = "mm", width = 200, height = 300, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_ahi_relabundance_",SUBNAME,"_",PDI,"_",LVL,".png"), p, device = "png", units = "mm", width = 200, height = 300, dpi = 300))

  # clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```



## Figure for all significant families in all four treatment groups


``` {r , eval=TRUE, error=TRUE}
params <- readRDS("R_objects/comp_params.RDS")
MTRL <- "Feces"
DAY <- "d8"
SUBNAME <- "Feces_d8"
PDI <- "treatment"
# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# Make sure there is a folder
if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))


fams <- c("Veillonellaceae","Streptococcaceae","Staphylococcaceae","Ruminococcaceae","Phylum_Firmicutes","Phylum_Actinobacteria","Peptostreptococcaceae","Micrococcaceae","Lactobacillaceae","Erysipelotrichaceae","Erysipelatoclostridiaceae","Enterococcaceae","Enterobacteriaceae","Eggerthellaceae","Clostridiales_Incertae_Sedis_XIII","Corynebacteriaceae","Coriobacteriaceae","Bifidobacteriaceae","Bacillaceae_1","Aerococcaceae","Ruminococcaceae","Erysipelotrichaceae","Desulfovibrionaceae","Synergistaceae","Bacteroidaceae", "Lachnospiraceae","Eubacteriaceae")

fams <- c("Ruminococcaceae","Lactobacillaceae","Erysipelotrichaceae","Enterobacteriaceae","Bifidobacteriaceae","Ruminococcaceae","Erysipelotrichaceae","Bacteroidaceae", "Lachnospiraceae","Eubacteriaceae")

LVL <- "Family"
  # Filter samples based on variables given
  phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Family, Abundance+pseudocount, color = fct_rev(treatment))) + 
    geom_boxplot() + 
    scale_y_log10() + 
    coord_flip() + 
    scale_color_manual(values = params$COL) + 
    facet_grid(Phylum ~ ., scales = "free_y", space = "free")
  
  p.dT <-  p + ggtitle(paste0("Abundance of significant ",LVL," taxa in all treatment groups for ",MTRL," ",DAY))
  p.dT
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 350, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 350, dpi = 300))
  
  LVL <- "Genus"
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Family~., scales = "free_y", space = "free")
  p.dT <-  p + ggtitle(paste0("Abundance of significant ",LVL," taxa in all treatment groups for ",MTRL," ",DAY))
  p.dT
  psuppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 350, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 350, dpi = 300))



# Create sub-folder
if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
# Filter samples based on vancomycin expsoure
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)
# Filter samples based on feces and day 8
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)
PDI <- "treatment"
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)
head(filt.DA)

filt.DA[filt.DA$Family == "Enterobacteriaceae",]
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.ent <- subset_taxa(filt.ra, Family == "Enterobacteriaceae")

# melt the data
ent.melt <- psmelt(filt.ent)

pseudocount <- min(ent.melt$Abundance[ent.melt$Abundance != 0])


ggplot(ent.melt, aes(Genus, Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip()


```


``` {r , eval=TRUE, error=TRUE}
# Create sub-folder
if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
# Filter samples based on vancomycin expsoure
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)
# Filter samples based on feces and day 8
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)
PDI <- "treatment"
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)
head(filt.DA)

filt.DA[filt.DA$Family == "Enterobacteriaceae",]
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.ent <- subset_taxa(filt.ra, Family == "Enterobacteriaceae")
filt.egg <- subset_taxa(filt.ra, Family == "Eggerthellaceae")

# melt the data
ent.melt <- psmelt(filt.ent)
egg.melt <- psmelt(filt.egg)


pseudocount <- min(ent.melt$Abundance[ent.melt$Abundance != 0])


ggplot(ent.melt, aes(Genus, Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip()



cor


```

## DAY CTRL TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "CTRL"
PDI <- "day"
SUBNAME <- "Feces_CTRL_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "CTRL"
PDI <- "day"
SUBNAME <- "Feces_CTRL_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "CTRL"
PDI <- "day"
SUBNAME <- "Feces_CTRL_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```












## DAY PFOS TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "PFOS"
PDI <- "day"
SUBNAME <- "Feces_PFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "PFOS"
PDI <- "day"
SUBNAME <- "Feces_PFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "PFOS"
PDI <- "day"
SUBNAME <- "Feces_PFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```













## DAY VAN TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN"
PDI <- "day"
SUBNAME <- "Feces_VAN_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lmc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN"
PDI <- "day"
SUBNAME <- "Feces_VAN_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN"
PDI <- "day"
SUBNAME <- "Feces_VAN_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aov(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## DAY VAN+PFOS TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
PDI <- "day"
SUBNAME <- "Feces_VANPFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lao(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
PDI <- "day"
SUBNAME <- "Feces_VANPFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lao2(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
PDI <- "day"
SUBNAME <- "Feces_VANPFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aov(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



## FECES d08 CTRL TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "CTRL"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "CTRL"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "CTRL"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```






## FECES d08 PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```





## FECES d08 VAN TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.per(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.qpo(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```





## FECES d08 VAN+PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```






## FECES d28 VAN TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```






## FECES d28 VAN+PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## Correlation FECES d8 PFOS treated {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
PFOS <- "pfos"
VAN <- "van"            # Also tesing "ctrl"
DAY <- "d8"
PDI <- "PFOS_serum8_ug" # Also testing "PFOS_liver_mg"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))


# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & pfos == PFOS & day == DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
filt <- transform_sample_counts(filt, function(x) x/sum(x))
# Run the selected analysis
filt.DA <- DA.spe(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/corr/stats_",MTRL,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Add pseudo count
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])

p.cor <- ggplot(DAm, aes(x = PFOS_liver_mg, y = Abundance+pseudocount, color = Genus)) + 
  geom_jitter() + 
  geom_smooth(method=lm) + 
  facet_wrap("Genus", scales = "free_y") + 
  scale_y_log10() +
  scale_x_continuous(breaks = seq(1.717, 2.931, 0.25)) + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
  theme_pubr()

p.cor

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/corr/Correlation_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".pdf"), p.cor, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/corr/Correlation_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".png"), p.cor, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# DAm$sampleID <- with(DAm, reorder(sampleID, PFOS_liver_mg, median)) 

# Create plot

# p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = rat_name)) + geom_boxplot() + scale_y_log10() + coord_flip()
# p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",PFOS," ",VAN," ",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
# p.dT
# 
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
# h.plot <- plot_heatmap(DA.sig, sample.label = PDI ,sample.order = PDI, taxa.label = LVL, taxa.order = "Phylum", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",PFOS," ",VAN," ",DAY))
# h.plot
# 
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_corr_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_corr_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```





## SETTINGS {.tabset .tabset-fade .tabset-pills}

### EXPLANATION

Overview of the packages that was used for this analysis

### PACKAGES

The analysis was run in R version `getRversion()` using the following packages:

```{r packages, eval=TRUE}
pack <- data.frame(Package = (.packages()))

for (i in seq(nrow(pack))) pack$Version[i] <- as.character(packageVersion(pack$Package[i]))

kbl(pack[order(pack$Package),], row.names = F) %>% kable_classic(lightable_options = "striped")   
     
```

### PARAMETERS
```{r parameters, eval=TRUE}
params <- readRDS("R_objects/comp_params.RDS")

dat <- as.data.frame(params)

```
## References
