---
title: "Fibrex Compositional Analysis"
author: "Claus Lykkebo"
date: "`r Sys.Date()`"
output:
  rmdformats::readthedown:
    toc_depth: 4
    collapsed: false
    code_folding: hide
    number_sections: true
knit: (function(inputFile, encoding) { 
    rmarkdown::render(
        inputFile, encoding = encoding,
        output_dir = file.path(dirname(inputFile),"output"),
        output_file = paste0("Fibrex_", Sys.Date(), "_differential_abundance.html")) 
    })
bibliography: references.bib
link-citations: true
params:
    input: "R_objects/Phyloseq.Rdata"
    meta: "input/metadata_fibrex.csv"
    batch: "Run"
    indeces: "Observed|Shannon|FaithPD|Chao1"
    COL0: !r c("CTRL" = "#eeeeee","PFOS" = "#666666")
    COL1: !r c("LF_CTRL" = "#a5cee3","LF_PFOS" = "#1778b6","HF_CTRL" = "#b4d88a","HF_PFOS" = "#30a148")
    COL2: !r c("LF_CTRL_d8" = "#a5cee3","LF_CTRL_d21" = "#a5cee3","LF_PFOS_d8" = "#1778b6","LF_PFOS_d21" = "#1778b6","HF_CTRL_d8" = "#b4d88a","HF_CTRL_d21" = "#b4d88a","HF_PFOS_d8" = "#30a148","HF_PFOS_d21" = "#30a148")
    COLFEED: !r c("LF" = "#1778b6","HF" = "#30a148")
---

# GMH ASV ANALYSIS PIPELINE

This Rmarkdown contains the commands necessary to perform initial analyses of the output from the DF_GMH_PIPELINE There will be things that should be updated to fit the exact project. It is important that metadata is loaded correctly and it should contain a variable identifying negative controls and mock samples. I recommend visiting the ["Analysis of community ecology data in R"](https://www.davidzeleny.net/anadat-r/doku.php/en:start) to read about the theory behind the alpha and beta diversity and examples of the necessary R-code to execute them. For analyses of differential abundance I will use the DAtest package [(Russel et al., 2018)](https://www.biorxiv.org/content/10.1101/241802v1). Another excellent source of help is the R [cheat-sheets](https://www.rstudio.com/resources/cheatsheets/).

```{r setup, eval=TRUE, echo=TRUE, message=FALSE,warning = FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Load libraries
library(tidyverse)
library(phyloseq)
library(decontam)
library(pals)
library(ggpubr)
library(rstatix)
library(vegan)
library(reshape2)
library(DAtest)
library(ape)
library(kableExtra)
library(plotly)
library(magick)
library(forcats)

# Create used folders if missing
if (!file.exists("R_objects")) dir.create(file.path(getwd(), "R_objects"))
if (!file.exists("plots")) dir.create(file.path(getwd(), "plots"))
if (!file.exists("plots/differential_abundance")) dir.create(file.path(getwd(), "plots/differential_abundance"))
if (!file.exists("tables")) dir.create(file.path(getwd(), "tables"))
if (!file.exists("scripts")) dir.create(file.path(getwd(), "scripts"))

saveRDS(params, "R_objects/comp_params.RDS")
```

## SCRIPTS {.tabset .tabset-fade .tabset-pills}

### INFO

This section contains the scripts for the custom functions used in this pipeline

### FILTER TAXA

These functions are used to merge ASV that are not one of the most abundant, or below a defined threshold, into "Other"

```{r merge_taxa_scripts, eval=TRUE, echo=TRUE}

# This function merges anything not in the top (10 by default) most abundant ASVs
merge_less_than_top <- function(pobject, top=10){
  # transform to sample counts to relative values
  transformed <- transform_sample_counts(pobject, function(x) x/sum(x))

    # Orientate and export otu table
  if (taxa_are_rows(transformed)) otu.table <- as.data.frame(otu_table(transformed)) else otu.table <- as.data.frame(t(otu_table(transformed)))
  
  # sort otu table by decreasing abundance
  otu.sort <- otu.table[order(rowMeans(otu.table), decreasing = TRUE),]
  
  # Create list of ASVs to merge
  otu.list <- row.names(otu.sort[(top+1):nrow(otu.sort),])
  if(any("Others" %in% rownames(otu.table))) otu.list <- c(otu.list,"Others")
  
  # perform the merger with the original sample counts
  merged <- merge_taxa(pobject, otu.list, 1)
  
  # change the taxa name
  taxa_names(merged)[taxa_names(merged) %in% otu.list[1]] <- "Others"
  
  # change taxa to "Other" for all levels not being NAs
  for (i in 1:ncol(tax_table(merged))){
    if (sum(!is.na(tax_table(merged)[,i]))){
      tax_table(merged)["Others",i] <- "Others"
    }
  }
  return(merged)
}

# This function merges ASVs that are below a defined ratio, with the default being 0.001 (0.1%)
merge_low_abundance <- function(pobject, threshold=0.001){
  
  # transform to sample counts to relative values
  transformed <- transform_sample_counts(pobject, function(x) x/sum(x))
  
  # Orientate and export otu table
  if (taxa_are_rows(transformed)) otu.table <- as.data.frame(otu_table(transformed)) else otu.table <- as.data.frame(t(otu_table(transformed)))
  
  # Create list of ASVs to merge
  otu.list <- row.names(otu.table[rowMeans(otu.table) < threshold,])
  if(any("Others" %in% rownames(otu.table))) otu.list <- c(otu.list,"Others")

  # perform the merger with the original sample counts
  merged <- merge_taxa(pobject, otu.list, 1)
  
  # change the taxa name
  taxa_names(merged)[taxa_names(merged) %in% otu.list[1]] <- "Others"
  
  # change taxa to "Other" for all levels not being NAs
  for (i in 1:ncol(tax_table(merged))){
    if (sum(!is.na(tax_table(merged)[,i]))){
      tax_table(merged)["Others",i] <- "Others"
    }
  }
  return(merged)
}

merge_prevalence <- function(pobject, variable, min.samples){
  # transform to sample counts to relative values
  if (taxa_are_rows(transformed)) otu.table <- as.data.frame(otu_table(transformed)) else otu.table <- as.data.frame(t(otu_table(transformed)))
  
  # make binary
  otu.table[otu.table != 0] <- 1
  
  # extract metadata
  dat <- data.frame(sample_data(pobject))
  
  # set groups
  vgroup <- unique(dat[,variable])
  
  # create count table
  counts <- data.frame(row.names = row.names(otu.table))
  for (i in seq(length(vgroup))) counts[,vgroup[i]] <- rowSums(otu.table[,dat[,variable]==vgroup[i]])
  
  counts$keep <- ifelse(rowSums(counts) > min.sample*length(vgroup), TRUE, FALSE)
  
  # Create list of ASVs to merge
  otu.list <- row.names(counts[!counts$keep,])
  if(any("Others" %in% rownames(otu.table))) otu.list <- c(otu.list,"Others")

  # perform the merger with the original sample counts
  merged <- merge_taxa(pobject, otu.list, 1)
  
  # change the taxa name
  taxa_names(merged)[taxa_names(merged) %in% otu.list[1]] <- "Others"
  
  # change taxa to "Other" for all levels not being NAs
  for (i in 1:ncol(tax_table(merged))){
    if (sum(!is.na(tax_table(merged)[,i]))){
      tax_table(merged)["Others",i] <- "Others"
    }
  }
  return(merged)
  
}
# save functions
save(merge_less_than_top, merge_low_abundance, file = "scripts/merge_abundance.Rdata")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)) #will clear all objects includes hidden objects.
invisible(gc()) #free up memory and report the memory usage.

```

# COMPOSITIONAL DIFFERENCES

The last part of the initial analyses is to compare the composition of the samples and test the differential abundance of individual ASVs or taxa.

## AGGLOMERATE DATA - All data

Depending on the project and data it might be relevant to agglomerate data at species or higher taxonomic level for this type of analysis.

```{r agglomerate, eval=TRUE, echo=TRUE}
params <- readRDS("R_objects/comp_params.RDS")
# load
load(params$input)

# Remove day 20
phy <- subset_samples(physeq = phy, day %in% c("d0","d08","d12","d16","d21"))

# agglomerate at species and genus level
phy.sp <- tax_glom(phy, taxrank = "Species")
phy.ge <- tax_glom(phy, taxrank = "Genus")
phy.fa <- tax_glom(phy, taxrank = "Family")
phy.or <- tax_glom(phy, taxrank = "Order")
phy.cl <- tax_glom(phy, taxrank = "Class")
phy.ph <- tax_glom(phy, taxrank = "Phylum")

# save agglomerated phyloseq objects
save(phy.sp, phy.ge, phy.fa, phy.or, phy.cl, phy.ph, file = "R_objects/Agglomerated_all.RData")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## AGGLOMERATE DATA - ITERATIONS

The following section agglomerates the data on taxonomic levels based on subsets of Material (Feces, Ileum, or Cecum), Day of sampling (Day 0, 8, 12, 16, 20, and 21), Feed type (LF = Low fibre, HF = High fibre), and Treatment (CTRL = Control, PFOS = PFOS). Agglomerations are saved in files with a specific naming convention for easy calling in further analysis.

```{r agglomerate_day, eval=TRUE, echo=TRUE}
params <- readRDS("R_objects/comp_params.RDS")
# load
load(params$input)

# Remove day 20
phy <- subset_samples(physeq = phy, day %in% c("d0","d08","d12","d16","d21"))

str.agg <- c("Feces","Feces_LF","Feces_HF","Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS","Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d21","Feces_d0d08","Feces_d08d21") #,"Feces_d20"
SUBNAME <- "Cecum"  
for (SUBNAME in str.agg) {
  # Determine material parameter
  if (grepl("Feces",SUBNAME) == TRUE) {
    MTRL <- "Feces"
    mn <- "MTRL"
  } else if (grepl("Cecum", SUBNAME) == TRUE) {
    MTRL <- "Cecum"
    mn <- "MTRL"
  } else if (grepl("Ileum", SUBNAME) == TRUE) {
    MTRL <- "Ileum"
    mn <- "MTRL"
  }
  # Determine parameters for differential abundance on specific days
  if (grepl("_d0d08", SUBNAME) ==TRUE) {
    DAY <- c("d0","d08")
    dn <- "day0+8"
  } else if (grepl("_d08",SUBNAME) == TRUE) {
    DAY <- c("d08")
    dn <- "day8"
  } else if (grepl("_d08d21",SUBNAME) == TRUE) {
    DAY <- c("d08","d21")
    dn <- "day8+21"
  } else if (grepl("_d0",SUBNAME) == TRUE) {
    DAY <- "d0"
    dn <- "day0"
  } else if (grepl("_d12",SUBNAME) == TRUE) {
    DAY <- "d12"
    dn <- "day12"
  } else if (grepl("_d16",SUBNAME) == TRUE) {
    DAY <- "d16"
    dn <- "day16"
  # } else if (grepl("_d20",SUBNAME) == TRUE) {
  #   DAY <- "d20"
  #   dn <- "day20"
  } else if (grepl("_d21",SUBNAME) == TRUE) {
    DAY <- "d21"
    dn <- "day21"
  } else {
    dn <- "all_days"
  }
  # Determine parameters for differential abundance in feed and treatment groups
  if (grepl(paste0(MTRL,"_CTRL"),SUBNAME) == TRUE) {
    TREAT <- "CTRL"
    FEED <- c("LF","HF")
  } else if (grepl(paste0(MTRL,"_PFOS"),SUBNAME) == TRUE) {
    TREAT <- "PFOS"
    FEED <- c("LF","HF")
  } else if (grepl("_LF_CTRL",SUBNAME) == TRUE) {
    TREAT <- "CTRL"
    FEED <- "LF"
  } else if (grepl("_LF_PFOS",SUBNAME) == TRUE) {
    TREAT <- "PFOS"
    FEED <- "LF"
  } else if (grepl("_HF_CTRL",SUBNAME) == TRUE) {
    TREAT <- "CTRL"
    FEED <- "HF"
  } else if (grepl("_HF_PFOS",SUBNAME) == TRUE) {
    TREAT <- "PFOS"
    FEED <- "HF"
  } else if (grepl(paste0(MTRL,"_LF"),SUBNAME) == TRUE) {
    TREAT <- c("CTRL","PFOS")
    FEED <- "LF"
  } else if (grepl(paste0(MTRL,"_HF"),SUBNAME) == TRUE) {
    TREAT <- c("CTRL","PFOS")
    FEED <- "HF"
  } else if (dn == "all_days") {
    TREAT <- c("CTRL","PFOS")
    FEED <- c("LF","HF")
  } else if (exists("dn") == TRUE) {
  } else {
    print("Something went wrong...")
  }

  #Subset data
  if (exists("DAY") == TRUE) {
    phy.sub <- subset_samples(phy, material == MTRL & day %in% DAY) 
  } else if (exists("DAY") == FALSE) {
    phy.sub <- subset_samples(phy, material == MTRL & treatment %in% TREAT & feed %in% FEED) 
  }

# Agglomerate at species and genus level
phy.sp <- tax_glom(phy.sub, taxrank = "Species")
phy.ge <- tax_glom(phy.sub, taxrank = "Genus")
phy.fa <- tax_glom(phy.sub, taxrank = "Family")
phy.or <- tax_glom(phy.sub, taxrank = "Order")
phy.cl <- tax_glom(phy.sub, taxrank = "Class")
phy.ph <- tax_glom(phy.sub, taxrank = "Phylum")

# save agglomerated phyloseq objects

save(phy.sp, phy.ge, phy.fa, phy.or, phy.cl, phy.ph, file = paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
print(paste0("Processed ",SUBNAME," correctly!"))
suppressWarnings(rm(MTRL,mn,DAY,dn,TREAT,FEED,phy.sp, phy.ge, phy.fa, phy.or, phy.cl, phy.ph,phy.sub), classes = "warning")
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# DIFFERENTIAL ABUNDANCE
Differential abundance is tested here based on significant differences in relative abundance in specific taxonomic entries based on groupings of different feed and treatment.

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

## FECES
In the following we take a look at differential abundance in feces samples, dividing the analyses into Days and type of test:
  - Test of PFOS effect on taxa abundance (calling SUBNAME = _LF & _HF)
  - Test of FEED effect on taxa abundance (calling SUBNAME = _CTRL & _PFOS)
  
We test all on three taxonomic levels:
  - Species
  - Genus
  - Family

### DAY 0
Day 0 samples should have no taxonomic differences between Treatment groups while a difference is expected in Feed. Therefore we test only these two variables on the entire dataset (cross analysis should not be necessary).

#### FEED
Testing differential abundance between feeding groups LF = Low fibre, HF = High fibre.

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d0" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed" #c("treatment","feed")
SUBNAME <- "Feces_d0" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.sp <- subset_samples(phy.sp, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Calculate pseudocount
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])

# Create plot
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed))) +
  geom_boxplot() + 
  scale_y_log10() + 
  coord_flip() + 
  scale_fill_manual(name = "Feed", values = params$COLFEED, breaks = c("HF","LF")) + 
  facet_grid(Phylum ~ . , scales = "free_y", space = "free") + 
  theme_bw() + 
  labs(fill = "Feed")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 250, height = 250, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 250, height = 250, dpi = 300))

# # Plot on heatmap
# h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," FEED-group ",FEED))
# h.plot
# 
# suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d0" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed" #c("treatment","feed")
SUBNAME <- "Feces_d0" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Choosing to use t-test - ALR (tta) as the test to calculate differential abundance.

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COLFEED, breaks = c("LF","HF")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Feed")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 150, height = 250, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 150, height = 250, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d0" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed" #c("treatment","feed")
SUBNAME <- "Feces_d0" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - Rank (ttr) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COLFEED, breaks = c("LF","HF")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Feed")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 150, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 150, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


#### TREATMENT
##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d0" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "treatment" #c("treatment","feed")
SUBNAME <- "Feces_d0" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.sp <- subset_samples(phy.sp, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL0, breaks = c("CTRL","PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# # Plot on heatmap
# h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," FEED-group ",FEED))
# h.plot
# 
# suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d0" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "treatment" #c("treatment","feed")
SUBNAME <- "Feces_d0" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL0, breaks = c("CTRL","PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 400, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 400, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d0" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "treatment" #c("treatment","feed")
SUBNAME <- "Feces_d0" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - Rank (ttr) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

################################
# NOTHING SIGNIFICANT


# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL0, breaks = c("CTRL","PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



### DAY 8
Samples at day 8 are taken 24 hours after last gavage of corn oil (+/- PFOS) and prior to dissection of half the animals.

#### NO FIBRE (test pfos)
Here we take a look at the differencial abundance driven by PFOS treatment (CTRL vs. PFOS) in the No fibre feeding group (A).

##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)


################################
# NOTHING SIGNIFICANT


# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

################################
# NOTHING SIGNIFICANT


# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

################################
# NOTHING SIGNIFICANT


# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



#### FIBRE (test pfos)
Here we take a look at the differential abundance driven by PFOS treatment (CTRL vs. PFOS) in the Fibre feeding group (B).
##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

################################
# NOTHING SIGNIFICANT


# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# # Plot on heatmap
# h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," FEED-group ",FEED))
# h.plot
# 
# suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

################################
# NOTHING SIGNIFICANT


# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

################################
# NOTHING SIGNIFICANT


# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw() + labs(fill = "Treatment")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




#### CTRL (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the control treatment group (CTRL).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))


# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified Log t-test (ltt) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

#### PFOS (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the PFOS treatment group (PFOS).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d08" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d08" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
p.dT

suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### DAY 12

#### NO FIBRE (test pfos)
Here we take a look at the differencial abundance driven by PFOS treatment (CTRL vs. PFOS) in the No fibre feeding group (A).

##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - RANK (ttr) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



#### FIBRE (test pfos)
Here we take a look at the differential abundance driven by PFOS treatment (CTRL vs. PFOS) in the Fibre feeding group (B).
##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc)  as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




#### CTRL (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the control treatment group (CTRL).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d21","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified Log t-test (ltt) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

#### PFOS (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the PFOS treatment group (PFOS).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d12" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d12" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




### DAY 16

#### NO FIBRE (test pfos)
Here we take a look at the differencial abundance driven by PFOS treatment (CTRL vs. PFOS) in the No fibre feeding group (A).

##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d16" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d16" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d16" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d16" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - RANK (ttr) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d16" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



#### FIBRE (test pfos)
Here we take a look at the differential abundance driven by PFOS treatment (CTRL vs. PFOS) in the Fibre feeding group (B).
##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc)  as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




#### CTRL (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the control treatment group (CTRL).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d21","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified Log t-test (ltt) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

#### PFOS (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the PFOS treatment group (PFOS).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




### DAY 20

#### NO FIBRE (test pfos)
Here we take a look at the differencial abundance driven by PFOS treatment (CTRL vs. PFOS) in the No fibre feeding group (A).

##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - RANK (ttr) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



#### FIBRE (test pfos)
Here we take a look at the differential abundance driven by PFOS treatment (CTRL vs. PFOS) in the Fibre feeding group (B).
##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc)  as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




#### CTRL (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the control treatment group (CTRL).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d21","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified Log t-test (ltt) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

#### PFOS (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the PFOS treatment group (PFOS).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




### DAY 21

Samples at day 8 are taken 24 hours after last gavage of corn oil (+/- PFOS) and prior to dissection of half the animals.

#### NO FIBRE (test pfos)
Here we take a look at the differencial abundance driven by PFOS treatment (CTRL vs. PFOS) in the No fibre feeding group (A).

##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - RANK (ttr) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "LF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("LF_CTRL","LF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



#### FIBRE (test pfos)
Here we take a look at the differential abundance driven by PFOS treatment (CTRL vs. PFOS) in the Fibre feeding group (B).
##### SPECIES (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc)  as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY (NO SIGNIFICANCE)
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
FEED <- "HF" #c("LF","HF")
#TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",FEED,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("Control","PFOS"), breaks = c("HF_CTRL","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",FEED))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",FEED,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```




#### CTRL (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the control treatment group (CTRL).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d21","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified Log t-test (ltt) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "CTRL" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - CLR (ttc) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_CTRL","HF_CTRL")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

#### PFOS (test feed)
Here we investigate the differential abundance driven by feed (No fibre vs. Fibre) in the PFOS treatment group (PFOS).

##### SPECIES
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
if (exists("FEED") == TRUE) {
  phy.sp <- subset_samples(phy.sp, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.sp <- subset_samples(phy.sp, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.sp <- subset_samples(phy.sp, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


##### GENUS
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.ge <- subset_samples(phy.ge, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.ge <- subset_samples(phy.ge, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.ge <- subset_samples(phy.ge, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.ge <- subset_samples(phy.ge, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

##### FAMILY
**TEST METHOD**

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family" #c("Species","Genus","Family")
MTRL <- "Feces" #c("Feces","Ileum","Cecum")
DAY <- "d21" #c("d0","d08","d12","d16","d20","d21")
#FEED <- "LF" #c("LF","HF")
TREAT <- "PFOS" #c("CTRL","PFOS")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Feces_d21" #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Filter samples (if needed)
# phy.fa <- subset_samples(phy.fa, material == MTRL)
if (exists("FEED") == TRUE) {
  phy.fa <- subset_samples(phy.fa, feed == FEED)
  print(paste0("Data contains type ",FEED," feed."))
} else {
  print("Data contains both feed LF and HF.")
}
if (exists("DAY") == TRUE) {
  phy.fa <- subset_samples(phy.fa, day %in% DAY)
  print(paste0("Data contains samples from day ",DAY,"."))
} else {
    print("Data contains all days.")
}
if (exists("TREAT") == TRUE) {
  phy.fa <- subset_samples(phy.fa, treatment == TREAT)
  print(paste0("Data contains only ",TREAT," treatment groups."))
} else {
  print("Data contains all treatment groups.")
}

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/differential_abundance/",SUBNAME,"/filter_summary_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/differential_abundance/",SUBNAME,"/filter_plot_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

**RUN DAtest**

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
write.csv(filt.p5,paste0("plots/differential_abundance/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",TREAT,"_",PDI,"_",LVL,".csv"))

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, fill = fct_rev(feed_treat))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_fill_manual(values = params$COL1, name = "Feed", labels = c("No fibre","Fibre"), breaks = c("LF_PFOS","HF_PFOS")) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + theme_bw()
  p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," ",TREAT))
  p.dT
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

# ILEUM & CAECUM
Here we take a look at the differencial abundance driven by PFOS treatment (CTRL vs. PFOS) in the No fibre feeding group (A).

##### Check for PFOS-driven DA

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Family"
LVLS <- c("Species","Genus","Family","Order","Class","Phylum")
MTRL <- "Ileum" #c("Feces","Ileum","Cecum")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Ileum_LF"
SUBNAMES <- c("Cecum_HF","Cecum_LF","Ileum_HF","Ileum_LF") #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")"Ileum","Ileum_HF","Ileum_LF","Cecum",
for (SUBNAME in SUBNAMES) {
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  
  # Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))
  
  for (LVL in LVLS) {
    
    if (LVL == "Species") {
      phy.used <- phy.sp
    } else if (LVL == "Genus") {
      phy.used <- phy.ge
    } else if (LVL == "Family") {
      phy.used <- phy.fa
    } else if (LVL == "Order") {
      phy.used <- phy.or
    } else if (LVL == "Class") {
      phy.used <- phy.cl
    } else if (LVL == "Phylum") {
      phy.used <- phy.ph
    }
    
    # Filter data
    filt <- preDA(data = phy.used, min.reads = 20, min.samples = 4)
    
    # Run the selected analysis
    filt.DA <- DA.kru(filt, predictor = PDI)
    
    # Evaluate the plot and summary table
    table(filt.DA$pval.adj < 0.05)
    filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
  
    if (any(filt.p5$pval.adj > 0)) {
      print(paste0("Something significant was found in ",SUBNAME," at ",LVL," level."))
      # Create a subset of the samples
      filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
      DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
      
      # melt the data
      DAm <- psmelt(DA.sig)
      
      # Create plot
      pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
      
      # Statistics
      stat.in <- DAm %>%
        group_by(.data[[LVL]], feed, day) %>%
        wilcox_test(as.formula("Abundance ~ feed_treat")) %>%
        adjust_pvalue(method = "BH") %>%
        add_significance("p.adj") %>% 
        add_xy_position(x = "feed", dodge = 0.8) %>%
        p_format("p.adj", accuracy = 0.0001, trailing.zero = TRUE, new.col = TRUE)
      stat.in$y.position <- log10(stat.in$y.position)
      stat.in
      
      stat.out <- DAm %>%
        group_by(.data[[LVL]], day) %>%
        wilcox_test(as.formula("Abundance ~ feed")) %>%
        adjust_pvalue(method = "BH") %>%
        add_significance("p.adj") %>% 
        add_xy_position(x = "feed", dodge = 0.8) %>%
        p_format("p.adj", accuracy = 0.0001, trailing.zero = TRUE, new.col = TRUE)
      stat.out$y.position <- log10(stat.out$y.position)
      stat.out
      
      p <- ggplot(DAm, aes(x = feed, y = Abundance+pseudocount)) +
        geom_boxplot(aes(fill = fct_rev(feed_treat))) + 
        scale_y_log10() + 
    #    coord_flip() + 
        scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("HF_CTRL" ="HF-CTRL","HF_PFOS" = "HF-PFOS","LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF_PFOS"), breaks = c("HF_CTRL","HF_PFOS","LF_CTRL","LF_PFOS")) + 
        facet_grid(day ~ .data[[LVL]] , scales = "free_y", space = "free", labeller = labeller(day = c("d08" = "Day 8","d21" = "Day 21"))) + 
        theme_pubr(border = TRUE)
      p
      
      p.stat <- p + stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red") +
      stat_pvalue_manual(stat.out, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE)
      p.stat
      
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.stat, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.stat, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
      } else {
        print(paste0("No significant results when testing ",PDI," in ",SUBNAME," at ",LVL," level."))
      }
    }
  }


# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Family"
LVLS <- c("Species","Genus","Family")
MTRL <- "Ileum_LF" #c("Feces","Ileum","Cecum")
PDI <- "feed_treat" #c("treatment","feed")
SUBNAME <- "Ileum_LF"
SUBNAMES <- c("Ileum_HF","Ileum_LF","Cecum_HF","Cecum_LF") #c("Feces","Feces_LF","Feces_HF","Feces_CTRL","Feces_PFOS","Feces_LF_CTRL","Feces_LF_PFOS","Feces_HF_CTRL","Feces_HF_PFOS",             "Feces_d0","Feces_d08","Feces_d12","Feces_d16","Feces_d20","Feces_d21","Feces_d0d08","Feces_d08d21",             "Ileum","Ileum_LF","Ileum_HF","Cecum","Cecum_LF","Cecum_HF")
for (SUBNAME in SUBNAMES) {
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  
  # Create sub-folder
  if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))
  
  for (LVL in LVLS) {
    
  if (LVL == "Species") {
    phy.used <- phy.sp
  } else if (LVL == "Genus") {
    phy.used <- phy.ge
  } else if (LVL == "Family") {
    phy.used <- phy.fa
  }
  
# Filter data
filt <- preDA(data = phy.used, min.reads = 20, min.samples = 4)

# Run the selected analysis
filt.DA <- DA.kru(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval.adj < 0.05)
filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]

if (any(filt.p5$pval.adj > 0)) {
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
  
  # melt the data
  DAm <- psmelt(DA.sig)
  
  # Create plot
  pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
  
  # Statistics
stat.in <- DAm %>%
  group_by(.data[[LVL]], feed, day) %>%
  wilcox_test(as.formula("Abundance ~ feed_treat")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance("p.adj") %>% 
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.0001, trailing.zero = TRUE, new.col = TRUE)
stat.in$y.position <- log10(stat.in$y.position)
stat.in

stat.out <- DAm %>%
  group_by(.data[[LVL]], day) %>%
  wilcox_test(as.formula("Abundance ~ feed")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance("p.adj") %>% 
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.0001, trailing.zero = TRUE, new.col = TRUE)
stat.out$y.position <- log10(stat.out$y.position)
stat.out
  
  p <- ggplot(DAm, aes(x = feed, y = Abundance+pseudocount)) +
    geom_boxplot(aes(fill = fct_rev(feed_treat))) + 
    scale_y_log10() + 
#    coord_flip() + 
    scale_fill_manual(values = params$COL1, name = "Treatment", labels = c("HF_CTRL" ="HF-CTRL","HF_PFOS" = "HF-PFOS","LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF_PFOS"), breaks = c("HF_CTRL","HF_PFOS","LF_CTRL","LF_PFOS")) + 
    facet_grid(day ~ .data[[LVL]] , scales = "free_y", space = "free", labeller = labeller(day = c("d08" = "Day 8","d21" = "Day 21"))) + 
    theme_pubr(border = TRUE)
  p
  
  p.stat <- p + stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red") +
  stat_pvalue_manual(stat.out, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE)
  p.stat
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.stat, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.stat, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
} else {
  if (exists("FEED") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for feed ",FEED,"."))
  } else if (exists("TREAT") == TRUE) {
  message(paste0("No significant results when testing ",PDI," in ",SUBNAME," subset for treatment ",TREAT,"."))
  }
}

  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

# CAECUM


# ENTEROBACTERIACEAE > GENERA
## ABUNDANCE + PSEUDOCOUNT
Significant difference in abundance was found for unclassified genera "Family_Enterobacteriaceae" above. Here we investigate and plot all genera present on day 8 in the family "Enterobacteriaceae".
``` {r eval=TRUE, echo=TRUE}
# Collected figures for Families showing significant differences
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Genus"
LVLS <- c("Genus","Family")
SUBNAME <- "Feces"
PDI <- "feed_treat"
TX <- "Enterobacteriaceae" 

# Create sub-folder
if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))

# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

for (LVL in LVLS) {
  # Filter data
  if (LVL == "Genus") {
    filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  } else if (LVL == "Family") {
    filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  }

  # Run tha selected analysis Wilcoxon
  filt.DA <- DA.wil(filt, predictor = "feed")
  
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family == TX)
  
  # Melt the data
  t.melt <- psmelt(filt.t)
  
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  
  # Sort samples to phylogenetic levels (alphabetically)
  if (LVL == "Genus") {
    t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
    genus.sorted <- as.character(unique(t.sorted$Genus))
    t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)
    
    levels(t.melt$genus_new) %>% head
  } else if (LVL == "Family") {
    t.sorted <- t.melt %>% select(Phylum:Family) %>% arrange(Phylum, Class, Order, Family)
    fam.sorted <- as.character(unique(t.sorted$Family))
    t.melt$fam_new <- factor(t.melt$Family, levels = fam.sorted)
    
    levels(t.melt$genus_new) %>% head
  }

  # Statistics
  stat.in <- t.melt %>%
    group_by(.data[[LVL]], feed, day) %>%
    wilcox_test(as.formula("Abundance ~ feed_treat")) %>%
    adjust_pvalue(method = "BH") %>%
    add_significance("p.adj") %>% 
    add_xy_position(x = "feed", dodge = 0.8) %>%
    p_format("p.adj", accuracy = 0.0001, trailing.zero = TRUE, new.col = TRUE)
  stat.in$y.position <- log10(stat.in$y.position)
  stat.in
  
  stat.out <- t.melt %>%
    group_by(.data[[LVL]], day) %>%
    wilcox_test(as.formula("Abundance ~ feed")) %>%
    adjust_pvalue(method = "BH") %>%
    add_significance("p.adj") %>% 
    add_xy_position(x = "feed", dodge = 0.8) %>%
    p_format("p.adj", accuracy = 0.0001, trailing.zero = TRUE, new.col = TRUE)
  stat.out$y.position <- log10(stat.out$y.position)
  stat.out
  
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(feed, Abundance+pseudocount)) +
    geom_boxplot(aes(fill = feed_treat), color = "black") +
    scale_y_log10(name = "Abundance+pseudocount (log10)", expand = expansion(mult = c(0, 0.075)), labels =function(x) paste0(x, "%")) +
    scale_fill_manual(name = "Group", values = params$COL1,
                      breaks = c("HF_CTRL","HF_PFOS","LF_CTRL","LF_PFOS"),
                      labels = c("HF-CTRL","HF-PFOS","LF-CTRL","LF-PFOS")) +
    facet_grid(.data[[LVL]] ~ day, space = "free", 
               labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
    theme_pubr(border = TRUE) +
    guides(color = "none") +
    theme(axis.title.x = element_blank())
  p
  
  if (SUBNAME == "Feces") {
      if (LVL == "Genus") {
        p.stat <- p + stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red") +
    stat_pvalue_manual(stat.out, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, y.position = c(1.1,1.4,1.2,1.3,1.3,0.01,0.01))
      } else {
        p.stat <- p + stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red") +
    stat_pvalue_manual(stat.out, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE)
      }
  } else if (SUBNAME %in% c("Ileum","Cecum")) {
    p.stat <- p + #stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red") +
    stat_pvalue_manual(stat.out, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE)
  }
  print(p.stat)
  
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",LVL,"_",TX,".pdf"), p.stat, device = "pdf", units = "mm", width = 200, height = 100, dpi = 300))
  suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/diffplot_",SUBNAME,"_",LVL,"_",TX,".png"), p.stat, device = "png", units = "mm", width = 200, height = 100, dpi = 300))
  
  save(stat.in, stat.out, p.stat, file = paste0("plots/differential_abundance/",SUBNAME,"/plot_stat_",SUBNAME,"_",LVL,".Rdata"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## CORRELATIONAL ANALYSIS
```{r , eval=FALSE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

#Set parameters
PDI <- "samplePFOS_ug"
SUBNAMES <- c("Feces","Cecum") #,"Ileum","Ileum_HF","Ileum_LF","Cecum","Cecum_HF","Cecum_LF"
TX <- "Enterobacteriaceae" 


  for (SUBNAME in SUBNAMES) {
        # Create sub-folder
    if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))
    
    # Load data
    load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
    
    #Subset and specify level
    LVL <- "Genus"
      phy.ge <- subset_samples(phy.ge, treatment == "PFOS" & day != "d0")
      phy.used <- phy.ge
      F.LVL <- "Genus"

    # Subset taxa
    phy.used <- subset_taxa(phy.used, Family == TX)

    # Filter data
    #filt <- preDA(data = phy.used, min.reads = 1, min.samples = 4)
    
    filt <- transform_sample_counts(phy.used, function(x) x/sum(x))
    
    # Run spearman selected analysis
    filt.DA <- DA.spe(filt, predictor = PDI, exact = FALSE)
    
    # Evaluate the plot and summary table
    table(filt.DA$pval.adj < 0.05)

      # Create a subset of the samples
      filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
      DA.sig <- prune_taxa(filt.DA$Feature, x = filt.ra)
      
      # melt the data
      DAm <- psmelt(DA.sig)
      
      # Add pseudo count
      pseudocount <- min(filt.DA$Abundance[filt.DA$Abundance != 0])
      
      # Create plot
      p.cor <- ggplot(DAm, aes(x = .data[[PDI]], y = Abundance, color = feed)) + 
        geom_jitter(aes(alpha = day)) + 
        geom_smooth(aes(color = feed), method=lm) + 
        geom_line(aes(color = rat_name)) +
        facet_grid("Genus ~ feed", scales = "free") + 
        scale_y_log10(name = "Abundance (log10)") +
        scale_x_continuous("PFOS g/g in sample") + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
        theme_pubr(border = TRUE) +
        scale_color_manual(name = "Feed", values = params$COLFEED)
      p.cor
      print(p.cor)
      # Save output
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/",TX,"_correlation_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.cor, device = "pdf", units = "mm", width = 150, height = 110, dpi = 300))
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/",TX,"_correlation_",SUBNAME,"_",PDI,"_",LVL,".png"), p.cor, device = "png", units = "mm", width = 150, height = 110, dpi = 300))

}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

#CORRELATIONAL ANALYSIS TO QUANTITATIVE PFOS (Feces)
Also run one with day as PDI and split by treatment (CTRL-PFOS) to tell if there are timeline differences.
Note that this following analysis should be run with pval.adj, and not pval.
```{r , eval=FALSE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

#Set parameters
LVLS <- c("Genus","Family","Order","Class")
PDI <- "samplePFOS_ug"#"serumPFOS_ug"
SUBNAMES <- c("Feces","Feces_HF","Feces_LF","Feces_d08","Feces_d12","Feces_d16","Feces_d21") #,"Ileum","Ileum_HF","Ileum_LF","Cecum","Cecum_HF","Cecum_LF"

for (LVL in LVLS) {
  for (SUBNAME in SUBNAMES) {
        # Create sub-folder
    if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))
    
    # Load data
    load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
    
    #Subset and specify level
    if (LVL == "Genus") {
      phy.ge <- subset_samples(phy.ge, treatment == "PFOS" & day != "d0")
      phy.used <- phy.ge
      F.LVL <- "Family"
    } else if (LVL == "Family") {
      phy.fa <- subset_samples(phy.fa, treatment == "PFOS" & day != "d0")
      phy.used <- phy.fa
      F.LVL <- "Family"
    } else if (LVL == "Order") {
      phy.or <- subset_samples(phy.or, treatment == "PFOS" & day != "d0")
      phy.used <- phy.or
      F.LVL <- "Order"
    } else if (LVL == "Class") {
      phy.cl <- subset_samples(phy.cl, treatment == "PFOS" & day != "d0")
      phy.used <- phy.cl
      F.LVL <- "Class"
    }
    
    if (grepl("Feces_d",SUBNAME) == TRUE) {
      phy.hf <- subset_samples(phy.used, feed == "HF")
      phy.lf <- subset_samples(phy.used, feed == "LF")
      FEEDS <- c("HF","LF")
      
      for (FEED in FEEDS) {
        
        if (FEED == "HF") {
          phy.used <- phy.hf
        } else if (FEED == "LF") {
          phy.used <- phy.lf
        }
        
        # Filter data
        filt <- preDA(data = phy.used, min.reads = 20, min.samples = 4)
        filt <- transform_sample_counts(filt, function(x) x/sum(x))
        
        # Run spearman selected analysis
        filt.DA <- DA.spe(filt, predictor = PDI, exact = FALSE)
        
        # Evaluate the plot and summary table
        table(filt.DA$pval.adj < 0.05)
        filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
        filt.p5 <- filter(filt.p5, rowSums(is.na(filt.p5)) != ncol(filt.p5)) 
        filt.p5
        
        if (any(filt.p5$pval.adj > 0)) {
          # Create a subset of the samples
          filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
          DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
          
          # melt the data
          DAm <- psmelt(DA.sig)
          
          # Add pseudo count
          pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
          
          # Create plot
          p.cor <- ggplot(DAm, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = .data[[LVL]])) + 
            geom_jitter() + 
            geom_smooth(method=lm) + 
            facet_wrap(F.LVL, scales = "free_y") + 
            scale_y_log10() +
            scale_x_continuous() + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
            theme_pubr()
          p.cor
          print(p.cor)
          # Save output
          suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_",FEED,".pdf"), p.cor, device = "pdf", units = "mm", width = 400, height = 300, dpi = 300))
          suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_",FEED,".png"), p.cor, device = "png", units = "mm", width = 400, height = 300, dpi = 300))
    
          if (any(filt.p5$rho > 0)) {
            DA.sigpos <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05 & filt.DA$rho > 0], x = filt.ra)
            
            # melt the data
            DAmpos <- psmelt(DA.sigpos)
            
            # Add pseudo count
            pseudocount <- min(DAmpos$Abundance[DAmpos$Abundance != 0])
            
            # Create positive corr plots
            p.pos <- ggplot(DAmpos, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = .data[[LVL]])) + 
              geom_jitter() + 
              geom_smooth(method=lm) + 
              facet_wrap(F.LVL, scales = "free_y") + 
              scale_y_log10() +
              scale_x_continuous() + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
              theme_pubr()
            p.pos
            
            # Print output
            print(p.pos)
            
            # Save output
            suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_",FEED,"_positive.pdf"), p.pos, device = "pdf", units = "mm", width = 400, height = 300, dpi = 300))
            suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_",FEED,"_positive.png"), p.pos, device = "png", units = "mm", width = 400, height = 300, dpi = 300))
          } else {print(paste0("No positive correlations for ",LVL," in ",SUBNAME,"."))}
          
          if (any(filt.p5$rho < 0)) {
            DA.signeg <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05 & filt.DA$rho < 0], x = filt.ra)
            
            # melt the data
            DAmneg <- psmelt(DA.signeg)
            
            # Add pseudo count
            pseudocount <- min(DAmneg$Abundance[DAmneg$Abundance != 0])
            
            # Create negative corr plots
            p.neg <- ggplot(DAmneg, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = .data[[LVL]])) + 
              geom_jitter() + 
              geom_smooth(method=lm) + 
              facet_wrap(F.LVL, scales = "free_y") + 
              scale_y_log10() +
              scale_x_continuous() + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
              theme_pubr()
            p.neg
            
            # Print output
            print(p.neg)
            
            # Save output
            suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_",FEED,"_negative.pdf"), p.neg, device = "pdf", units = "mm", width = 400, height = 300, dpi = 300))
            suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_",FEED,"_negative.png"), p.neg, device = "png", units = "mm", width = 400, height = 300, dpi = 300))
            
            } else {print(paste0("No negative correlations for ",LVL," in ",SUBNAME,"."))}
          }
        }
#############################
      
    } else {
      
    # Filter data
    filt <- preDA(data = phy.used, min.reads = 20, min.samples = 4)
    filt <- transform_sample_counts(filt, function(x) x/sum(x))
    
    # Run spearman selected analysis
    filt.DA <- DA.spe(filt, predictor = PDI, exact = FALSE)
    
    # Evaluate the plot and summary table
    table(filt.DA$pval.adj < 0.05)
    filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
    filt.p5 <- filter(filt.p5, rowSums(is.na(filt.p5)) != ncol(filt.p5)) 
    filt.p5
    
    if (any(filt.p5$pval.adj > 0)) {
      # Create a subset of the samples
      filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
      DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
      
      # melt the data
      DAm <- psmelt(DA.sig)
      
      # Add pseudo count
      pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
      
      # Create plot
      p.cor <- ggplot(DAm, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = .data[[LVL]])) + 
        geom_jitter() + 
        geom_smooth(method=lm) + 
        facet_wrap(F.LVL, scales = "free_y") + 
        scale_y_log10() +
        scale_x_continuous() + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
        theme_pubr()
      p.cor
      print(p.cor)
      # Save output
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.cor, device = "pdf", units = "mm", width = 400, height = 300, dpi = 300))
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,".png"), p.cor, device = "png", units = "mm", width = 400, height = 300, dpi = 300))

      if (any(filt.p5$rho > 0)) {
      DA.sigpos <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05 & filt.DA$rho > 0], x = filt.ra)
      
      # melt the data
      DAmpos <- psmelt(DA.sigpos)
      
      # Add pseudo count
      pseudocount <- min(DAmpos$Abundance[DAmpos$Abundance != 0])
      
      # Create positive corr plots
      p.pos <- ggplot(DAmpos, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = .data[[LVL]])) + 
        geom_jitter() + 
        geom_smooth(method=lm) + 
        facet_wrap(F.LVL, scales = "free_y") + 
        scale_y_log10() +
        scale_x_continuous() + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
        theme_pubr()
      p.pos
      
      # Print output
      print(p.pos)
      
      # Save output
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_positive.pdf"), p.pos, device = "pdf", units = "mm", width = 400, height = 300, dpi = 300))
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_positive.png"), p.pos, device = "png", units = "mm", width = 400, height = 300, dpi = 300))
      } else {print(paste0("No positive correlations for ",LVL," in ",SUBNAME,"."))}
      
      if (any(filt.p5$rho < 0)) {
      DA.signeg <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05 & filt.DA$rho < 0], x = filt.ra)
      
      # melt the data
      DAmneg <- psmelt(DA.signeg)
      
      # Add pseudo count
      pseudocount <- min(DAmneg$Abundance[DAmneg$Abundance != 0])
      
      # Create negative corr plots
      p.neg <- ggplot(DAmneg, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = .data[[LVL]])) + 
        geom_jitter() + 
        geom_smooth(method=lm) + 
        facet_wrap(F.LVL, scales = "free_y") + 
        scale_y_log10() +
        scale_x_continuous() + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
        theme_pubr()
      p.neg
      
      # Print output
      print(p.neg)
      
      # Save output
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_negative.pdf"), p.neg, device = "pdf", units = "mm", width = 400, height = 300, dpi = 300))
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,"_negative.png"), p.neg, device = "png", units = "mm", width = 400, height = 300, dpi = 300))
      } else {print(paste0("No negative correlations for ",LVL," in ",SUBNAME,"."))}
      
    } else {
      print(paste0("No signficance found in filt.p5 in ",PDI," for ",SUBNAME," at ",LVL,"."))
    }
    }
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```
#CORRELATIONAL ANALYSIS TO DAY (Feces)
Also run one with day as PDI and split by treatment (CTRL-PFOS) to tell if there are timeline differences.
Note that this following analysis should be run with pval.adj, and not pval.
```{r , eval=FALSE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

#Set parameters
LVLS <- c("Genus","Family","Order","Class")
PDI <- "day_no"
SUBNAMES <- c("Feces","Feces_","Ileum","Cecum") #,"Ileum","Ileum_HF","Ileum_LF","Cecum","Cecum_HF","Cecum_LF"
SUBNAME <- "Feces"
for (LVL in LVLS) {
  for (SUBNAME in SUBNAMES) {
        # Create sub-folder
    if (!file.exists(paste0("plots/differential_abundance/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/differential_abundance/",SUBNAME)))
    
    # Load data
    load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
    
    #Subset and specify level
    if (LVL == "Genus") {
      # phy.ge <- subset_samples(phy.ge, treatment == "PFOS" & day != "d0")
      phy.used <- phy.ge
      F.LVL <- "Genus"
    } else if (LVL == "Family") {
      # phy.fa <- subset_samples(phy.fa, treatment == "PFOS" & day != "d0")
      phy.used <- phy.fa
      F.LVL <- "Family"
    } else if (LVL == "Order") {
      # phy.or <- subset_samples(phy.or, treatment == "PFOS" & day != "d0")
      phy.used <- phy.or
      F.LVL <- "Order"
    } else if (LVL == "Class") {
      # phy.cl <- subset_samples(phy.cl, treatment == "PFOS" & day != "d0")
      phy.used <- phy.cl
      F.LVL <- "Class"
    }
      
    # Filter data
    filt <- preDA(data = phy.used, min.reads = 20, min.samples = 4)
    filt <- transform_sample_counts(filt, function(x) x/sum(x))
    
    # Run spearman selected analysis
    filt.DA <- DA.spe(filt, predictor = PDI, exact = FALSE)
    
    # Evaluate the plot and summary table
    table(filt.DA$pval.adj < 0.05)
    filt.p5 <- filt.DA[filt.DA$pval.adj < 0.05,]
    filt.p5 <- filter(filt.p5, rowSums(is.na(filt.p5)) != ncol(filt.p5)) 
    filt.p5
    
    if (any(filt.p5$pval.adj > 0)) {
      # Create a subset of the samples
      filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
      DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval.adj < 0.05], x = filt.ra)
      
      # melt the data
      DAm <- psmelt(DA.sig)
      
      # Add pseudo count
      pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
      
      # Create plot
      p.cor <- ggplot(DAm, aes(x = .data[[PDI]], y = Abundance+pseudocount, color = feed_treat)) + 
        geom_jitter() +
        geom_smooth(method=lm) +
        facet_grid(as.formula(paste0("treatment ~",F.LVL)), scales = "free_y", switch = "y") + 
        scale_y_log10() +
        scale_x_continuous(name = "Day", breaks = c(0,8,12,16,21)) + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
        theme_pubr(border = TRUE) +
        scale_color_manual(values = params$COL1) +
        theme(strip.text.x.top = element_text(angle = 90))
      p.cor
      print(p.cor)
      
      # Save output
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.cor, device = "pdf", units = "mm", width = 800, height = 300, dpi = 300))
      suppressMessages(ggsave(paste0("plots/differential_abundance/",SUBNAME,"/Correlation_",SUBNAME,"_",PDI,"_",LVL,".png"), p.cor, device = "png", units = "mm", width = 800, height = 300, dpi = 300))

    } else {
      print(paste0("No signficance found in filt.p5 in ",PDI," for ",SUBNAME," at ",LVL,"."))
    }
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



## ILEUM
In the following we take a look at differential abundance in ileum samples, dividing the analyses into Days and type of test:
  - Test of PFOS effect on taxa abundance (calling SUBNAME = _A & _B)
  - Test of FEED effect on taxa abundance (calling SUBNAME = _CTRL & _PFOS)
  
We test all on three taxonomic levels:
  - Species
  - Genus
  - Family
  
## CECUM





#------------------------------
#** OLD SCRIPTS **

## FECES DAY 8 TREATMENT - DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "all"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
#phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "all"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
#phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 400, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 400, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```







### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "all"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
#phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, ,effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lao(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

## FECES DAY 8 CTRL vs. VAN - DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified t-test - ALR (tta) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```







### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, ,effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### ORDER
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Order"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.or <- subset_samples(phy.or, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.or, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Order, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### CLASS
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Class"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.cl <- subset_samples(phy.cl, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.cl, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Class, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### PHYLUM
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Phylum"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "ctrl"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ph <- subset_samples(phy.ph, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ph, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Phylum, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## FECES DAY 8 VAN vs. VAN+PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0("SPECIES: Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### ORDER
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Order"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.or <- subset_samples(phy.or, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.or, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Order, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### CLASS
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Class"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.cl <- subset_samples(phy.cl, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.cl, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Class, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

### PHYLUM
#### TEST METHOD

```{r testDA_order, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Phylum"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
PDI <- "treatment"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ph <- subset_samples(phy.ph, material == MTRL & day %in% DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ph, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",DAY,"_",VAN,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Phylum, y = Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free")
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",DAY," VAN: ",VAN))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = "Sample", taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," - DAY ",DAY," VAN-group ",VAN))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",VAN,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

## Figure for F8 PFOS effect
``` {r dabu_f8_pfos_effect, eval=TRUE, echo=TRUE, error=TRUE}
# Using Welch t-test (TTA) as test for all days on Genus level based on significant divergences from Family level
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
SUBNAME <- "Feces_d8"
PDI <- "treatment"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# "fams" is a manual list of all families generating significant results from differential abundance tests above on either family or lower taxonomical levels.
fams <- c("Veillonellaceae",
"Streptococcaceae",
"Staphylococcaceae",
"Ruminococcaceae",
"Phylum_Firmicutes",
"Phylum_Actinobacteria",
"Peptostreptococcaceae",
"Micrococcaceae",
"Lactobacillaceae",
"Erysipelotrichaceae",
"Erysipelatoclostridiaceae",
"Enterococcaceae",
"Enterobacteriaceae",
"Eggerthellaceae",
"Clostridiales_Incertae_Sedis_XIII",
"Corynebacteriaceae",
"Coriobacteriaceae",
"Bifidobacteriaceae",
"Bacillaceae_1",
"Aerococcaceae",
"Ruminococcaceae",
"Erysipelotrichaceae",
"Desulfovibrionaceae",
"Synergistaceae")

# Genera in Families showing significant differences in either CTRL <> PFOS and VAN <> VAN+PFOS
LVL <- "Genus"
for (TX in fams) {
  print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
  p.dT <-  p + ggtitle(paste0("Abundance of ",LVL," taxa in the ",TX," family for ",MTRL," ",DAY))
  p.dT
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
}

# Species in Families showing significant differences in either CTRL <> PFOS and VAN <> VAN+PFOS
LVL <- "Species"
for (TX in fams) {
  print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Species, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL)
  p.dT <-  p + ggtitle(paste0("Abundance of ",LVL," taxa in the ",TX," family for ",MTRL," ",DAY))
  p.dT
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PDI,"_",LVL,"_",TX,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
}


```
## Figure for specific families + genera
``` {r eval=TRUE, echo=TRUE}
# Collected figures for Families showing significant differences
params <- readRDS("R_objects/comp_params.RDS")
LVL <- "Genus"
MTRL <- "Feces"
DAY <- "d8"
VAN <- "van"
SUBNAME <- "Feces_d8"
PDI <- "treatment"
fams <- c("Erysipelotrichaceae",
"Enterobacteriaceae")

#for (TX in fams) {
  LVL <- "Genus"
  #print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.aoa(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Sort samples to phylogenetic levels (alphabetically)
t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
genus.sorted <- as.character(unique(t.sorted$Genus))
t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)

levels(t.melt$genus_new) %>% head
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + 
    geom_boxplot() + 
    scale_y_log10() + 
    coord_flip() +
    scale_color_manual(values = params$COL) + 
    theme_bw() +
    facet_grid(Family ~ ., scales = "free_y", space = "free") +
    labs(color = "Treatment")
  p.ge <-  p #+ ggtitle(paste0("Abundance of ",LVL," taxa"))
  p.ge
  
    suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.ge, device = "pdf", units = "mm", width = 200, height = 250, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_",SUBNAME,"_",PDI,"_",LVL,".png"), p.ge, device = "png", units = "mm", width = 200, height = 250, dpi = 300))
  # # Species
  # LVL <- "Species"
  # phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY)
  # # Filter data
  # filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)
  # # Run tha selected analysis TTA
  # filt.DA <- DA.tta(filt, predictor = PDI)
  # # Create a subset of the samples
  # filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  # filt.t <- subset_taxa(filt.ra, Family == TX)
  # # Melt the data
  # t.melt <- psmelt(filt.t)
  # # Create pseudocounts
  # pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # # Plot data and save as output formats
  # p <- ggplot(t.melt, aes(Species, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + theme(axis.text.y = element_text(angle = 90))
  # p.sp <-  p + ggtitle(paste0("Abundance of ",LVL," taxa in the ",TX," family for ",MTRL," ",DAY))
  ####
  # CLEAN UP UNDER THIS LINE
  # Family
  LVL <- "Family"
  phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.aoa(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  t.sorted <- t.melt %>% select(Phylum:Family) %>% arrange(Phylum, Class, Order, Family)
family.sorted <- as.character(unique(t.sorted$Family))
t.melt$family_new <- factor(t.melt$Family, levels = family.sorted)

levels(t.melt$family_new) %>% head
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(family_new, Abundance+pseudocount, color = fct_rev(treatment))) + 
    geom_boxplot() + 
    scale_y_log10() + 
    coord_flip() + 
    scale_color_manual(values = params$COL) + 
    facet_grid(. ~ Family , scales = "free_y", space = "free") +
        labs(color = "Treatment")
    #theme(axis.text.y = element_text(angle = 90))
  p.fa <-  p #+ ggtitle(paste0("Abundance of ",LVL," taxa"))
  p.legend <- get_legend(p.fa)
  # p.done <- ggdraw() +
  #     draw_plot(p.fa, 0, 0.1, 0.5, 0.9) + 
  #     draw_plot(p.ge, 0.5, 0.1, 0.5, 0.9) +
  #     draw_grob(p.legend, 0, 0, 1, .1) +
  #     draw_plot_label("Family") + 
  #     draw_plot_label("Genus", x = 0.5) + 
  #     theme(plot.background = element_rect(fill="white", color = NA))
  # p.done
  done <- plot_grid(p.fa, p.ge, ncol = 1, align = "v", axis = "l", rel_heights = c(1,5))
  done
  # p.all <- ggarrange(p.fa, p.ge, labels = c("LF","HF"), ncol = 1, nrow = 2, common.legend = TRUE)
  # p.all
  
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_done_",SUBNAME,"_",PDI,".pdf"), done, device = "pdf", units = "mm", width = 300, height = 300, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotarrange_done_",SUBNAME,"_",PDI,".png"), done, device = "png", units = "mm", width = 300, height = 300, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
#}
```


``` {r , eval=TRUE, echo=TRUE}

# Plot for all taxonomy in phyla Actinobacteria and Firmicutes  
phs <- c("Firmicutes","Actinobacteria")  

for (TX in phs) {
  LVL <- "Family"
  print(paste0("Running ",TX,"..."))
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Family, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.fa <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  
  LVL <- "Genus"
  # Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.ge <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  
  # Species
  LVL <- "Species"
  phy.sp <- subset_samples(phy.sp, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Species, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.sp <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  
  p.all <- ggarrange(p.fa, p.ge, p.sp, labels = c("LF","HF","C"), ncol = 3, nrow = 1, common.legend = TRUE)
  p.allT <- annotate_figure(p.all, top = text_grob(paste0("Significant differential abundance for ",TX," in ", MTRL," ",DAY)))
  
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotphylum_",SUBNAME,"_",PDI,"_",TX,".pdf"), p.allT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplotphylum_",SUBNAME,"_",PDI,"_",TX,".png"), p.allT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))
  print(paste0("Done processing ",TX," and moving on to the next"))
}

# Plot for all phyla
  LVL <- "Phylum"
  # Load agglomorated subset
  load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
  # Filter samples based on variables given
  phy.ph <- subset_samples(phy.ph, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ph, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  #filt.t <- subset_taxa(filt.ra, Phylum == TX)
  # Melt the data
  t.melt <- psmelt(filt.ra)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Phylum, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) #+ theme(axis.text.y = element_text(angle = 90))
  p.ph <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
  p.ph

  # clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## Plot for significant genera in all groups
``` {r sig_genera, eval=TRUE, echo=TRUE}
SUBNAME <- "Feces_d8"  
MTRL <- "Feces"
DAY <- "d8"
PDI <- "treatment"


#gen.s <- c("Acetatifactor","Family_Clostridiales_Incertae_Sedis_XIII","Family_Desulfovibrionaceae","Family_Enterobacteriaceae","Family_Erysipelotrichaceae","Holdemania","Mucinivorans","Intestinimonas","Anaerotruncus","Breznakia","Clostridium_XIVa","Dysosmobacter","Enterococcus","Flavonifractor","Lachnospira","Marvinbryantia","Neglecta","Oscillibacter","Pseudoflavonifractor")

gen.s <- c("Pseudoflavonifractor","Oscillibacter","Neglecta","Marvinbryantia","Lachnospira","Flavonifractor","Enterococcus","Dysosmobacter","Clostridium_XIVa","Breznakia","Anaerotruncus","Intestinimonas","Mucinivorans","Holdemania","Family_Erysipelotrichaceae","Family_Enterobacteriaceae","Family_Desulfovibrionaceae","Family_Clostridiales_Incertae_Sedis_XIII","Acetatifactor")

LVL <- "Genus"
# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# Filter samples based on variables given
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
# Run tha selected analysis TTA
filt.DA <- DA.tta(filt, predictor = PDI)
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.t <- subset_taxa(filt.ra, Genus %in% gen.s)
# Melt the data
t.melt <- psmelt(filt.t)
# Create pseudocounts
pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
# Reorder to sample list above
t.melt$Genus <- factor(t.melt$Genus, levels = gen.s)
# Sort samples to phylogenetic levels (alphabetically)
# t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
# genus.sorted <- as.character(unique(t.sorted$Genus))
# t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)
# 
# levels(t.melt$genus_new) %>% head

# Plot data and save as output formats
p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Phylum ~ . , scales = "free_y", space = "free") + guides(color = guide_legend(title = "Treatment")) + theme_bw()
            
p.ph <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))

p.ph
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_allgenus_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.ph, device = "pdf", units = "mm", width = 200, height = 300, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_allgenus_",SUBNAME,"_",PDI,"_",LVL,".png"), p.ph, device = "png", units = "mm", width = 200, height = 300, dpi = 300))

  # clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```



## Plot for high relative abundance taxa (GENUS)
``` {r sig_genera, eval=TRUE, echo=TRUE}
SUBNAME <- "Feces_d8"  
MTRL <- "Feces"
DAY <- "d8"
PDI <- "treatment"

GENS <- c("Bifidobacterium",
"Bacteroides",
"Parabacteroides",
"Prevotellamassilia",
"Lactobacillus",
"Ligilactobacillus",
"Family_Lachnospiraceae",
"Order_Clostridiales",
"Family_Ruminococcaceae",
"Intestinimonas",
"Holdemania",
"Turicibacter",
"Escherichia/Shigella",
"Blautia")

LVL <- "Genus"
# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# Filter samples based on variables given
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
#filt.test <- testDA(filt, predictor = PDI, effectSize = 10)
# Run tha selected analysis TTA
filt.DA <- DA.aoa(filt, predictor = PDI)
filt.DA %>% select(Genus,pval.adj) %>% head(100)
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.t <- subset_taxa(filt.ra, Genus %in% GENS)
# Melt the data
t.melt <- psmelt(filt.t)
# Create pseudocounts
pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
# Reorder to sample list above
#t.melt$Genus <- factor(t.melt$Genus, levels = gen.s)
# Sort samples to phylogenetic levels (alphabetically)
t.sorted <- t.melt %>% select(Phylum:Genus) %>% arrange(Phylum, Class, Order, Family, Genus)
genus.sorted <- as.character(unique(t.sorted$Genus))
t.melt$genus_new <- factor(t.melt$Genus, levels = genus.sorted)

#levels(t.melt$genus_new) %>% head

# Plot data and save as output formats
p <- ggplot(t.melt, aes(genus_new, Abundance+pseudocount, color = fct_rev(treatment))) + 
  geom_boxplot() + 
  scale_y_log10() + 
  coord_flip() + 
  scale_color_manual(values = params$COL) + 
  facet_grid(Phylum ~ . , scales = "free_y", space = "free") + 
  guides(color = guide_legend(title = "Treatment")) +
  theme_bw()
            
#p.ph <-  p + ggtitle(paste0("Abundance of ",LVL," taxa"))
p

  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_hi_relabundance_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p, device = "pdf", units = "mm", width = 200, height = 300, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_ahi_relabundance_",SUBNAME,"_",PDI,"_",LVL,".png"), p, device = "png", units = "mm", width = 200, height = 300, dpi = 300))

  # clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```



## Figure for all significant families in all four treatment groups


``` {r , eval=TRUE, error=TRUE}
params <- readRDS("R_objects/comp_params.RDS")
MTRL <- "Feces"
DAY <- "d8"
SUBNAME <- "Feces_d8"
PDI <- "treatment"
# Load agglomorated subset
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))
# Make sure there is a folder
if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))


fams <- c("Veillonellaceae","Streptococcaceae","Staphylococcaceae","Ruminococcaceae","Phylum_Firmicutes","Phylum_Actinobacteria","Peptostreptococcaceae","Micrococcaceae","Lactobacillaceae","Erysipelotrichaceae","Erysipelatoclostridiaceae","Enterococcaceae","Enterobacteriaceae","Eggerthellaceae","Clostridiales_Incertae_Sedis_XIII","Corynebacteriaceae","Coriobacteriaceae","Bifidobacteriaceae","Bacillaceae_1","Aerococcaceae","Ruminococcaceae","Erysipelotrichaceae","Desulfovibrionaceae","Synergistaceae","Bacteroidaceae", "Lachnospiraceae","Eubacteriaceae")

fams <- c("Ruminococcaceae","Lactobacillaceae","Erysipelotrichaceae","Enterobacteriaceae","Bifidobacteriaceae","Ruminococcaceae","Erysipelotrichaceae","Bacteroidaceae", "Lachnospiraceae","Eubacteriaceae")

LVL <- "Family"
  # Filter samples based on variables given
  phy.fa <- subset_samples(phy.fa, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Family, Abundance+pseudocount, color = fct_rev(treatment))) + 
    geom_boxplot() + 
    scale_y_log10() + 
    coord_flip() + 
    scale_color_manual(values = params$COL) + 
    facet_grid(Phylum ~ ., scales = "free_y", space = "free")
  
  p.dT <-  p + ggtitle(paste0("Abundance of significant ",LVL," taxa in all treatment groups for ",MTRL," ",DAY))
  p.dT
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 350, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 350, dpi = 300))
  
  LVL <- "Genus"
  # Filter samples based on variables given
  phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
  # Filter data
  filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
  # Run tha selected analysis TTA
  filt.DA <- DA.tta(filt, predictor = PDI)
  # Create a subset of the samples
  filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
  filt.t <- subset_taxa(filt.ra, Family %in% fams)
  # Melt the data
  t.melt <- psmelt(filt.t)
  # Create pseudocounts
  pseudocount <- min(t.melt$Abundance[t.melt$Abundance != 0])
  # Plot data and save as output formats
  p <- ggplot(t.melt, aes(Genus, Abundance+pseudocount, color = fct_rev(treatment))) + geom_boxplot() + scale_y_log10() + coord_flip() + scale_color_manual(values = params$COL) + facet_grid(Family~., scales = "free_y", space = "free")
  p.dT <-  p + ggtitle(paste0("Abundance of significant ",LVL," taxa in all treatment groups for ",MTRL," ",DAY))
  p.dT
  psuppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 200, height = 350, dpi = 300))
  suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_all_",SUBNAME,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 200, height = 350, dpi = 300))



# Create sub-folder
if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
# Filter samples based on vancomycin expsoure
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)
# Filter samples based on feces and day 8
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)
PDI <- "treatment"
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)
head(filt.DA)

filt.DA[filt.DA$Family == "Enterobacteriaceae",]
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.ent <- subset_taxa(filt.ra, Family == "Enterobacteriaceae")

# melt the data
ent.melt <- psmelt(filt.ent)

pseudocount <- min(ent.melt$Abundance[ent.melt$Abundance != 0])


ggplot(ent.melt, aes(Genus, Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip()


```


``` {r , eval=TRUE, error=TRUE}
# Create sub-folder
if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))
# Filter samples based on vancomycin expsoure
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY & van == VAN)
# Filter samples based on feces and day 8
phy.ge <- subset_samples(phy.ge, material == MTRL & day %in% DAY)
# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)
PDI <- "treatment"
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)
head(filt.DA)

filt.DA[filt.DA$Family == "Enterobacteriaceae",]
# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
filt.ent <- subset_taxa(filt.ra, Family == "Enterobacteriaceae")
filt.egg <- subset_taxa(filt.ra, Family == "Eggerthellaceae")

# melt the data
ent.melt <- psmelt(filt.ent)
egg.melt <- psmelt(filt.egg)


pseudocount <- min(ent.melt$Abundance[ent.melt$Abundance != 0])


ggplot(ent.melt, aes(Genus, Abundance+pseudocount, color = treatment)) + geom_boxplot() + scale_y_log10() + coord_flip()



cor


```

## DAY CTRL TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "CTRL"
PDI <- "day"
SUBNAME <- "Feces_CTRL_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "CTRL"
PDI <- "day"
SUBNAME <- "Feces_CTRL_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "CTRL"
PDI <- "day"
SUBNAME <- "Feces_CTRL_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```












## DAY PFOS TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "PFOS"
PDI <- "day"
SUBNAME <- "Feces_PFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "PFOS"
PDI <- "day"
SUBNAME <- "Feces_PFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoa(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "PFOS"
PDI <- "day"
SUBNAME <- "Feces_PFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```













## DAY VAN TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN"
PDI <- "day"
SUBNAME <- "Feces_VAN_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lmc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN"
PDI <- "day"
SUBNAME <- "Feces_VAN_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aoc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN"
PDI <- "day"
SUBNAME <- "Feces_VAN_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aov(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## DAY VAN+PFOS TIME TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
PDI <- "day"
SUBNAME <- "Feces_VANPFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lao(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
PDI <- "day"
SUBNAME <- "Feces_VANPFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.lao2(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
PDI <- "day"
SUBNAME <- "Feces_VANPFOS_all"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.aov(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," all days"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```



## FECES d08 CTRL TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "CTRL"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "CTRL"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "CTRL"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```






## FECES d08 PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttr(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```





## FECES d08 VAN TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.per(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.qpo(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```





## FECES d08 VAN+PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ltt(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d0d8"
PDI <- "day"
SUBNAME <- "Feces_d0d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 0 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```






## FECES d28 VAN TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```






## FECES d28 VAN+PFOS TEST DIFFERENTIAL ABUNDANCE {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### SPECIES
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Species"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.sp <- subset_samples(phy.sp, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.sp, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Species, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT,"_",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.tta(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


### FAMILY
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Family"
MTRL <- "Feces"
TREAT <- "VAN+PFOS"
DAY <- "d2d8"
PDI <- "day"
SUBNAME <- "Feces_d2d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))

# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.fa <- subset_samples(phy.fa, material == MTRL & treatment == TREAT)

# Filter data
filt <- preDA(data = phy.fa, min.reads = 20, min.samples = 4)

# Test best method 
filt.test <- testDA(filt, predictor = PDI, effectSize = 10)

# Evaluate the plot and summary table
sum.fil <- summary(filt.test)
write.csv(sum.fil, file = paste0("plots/dabu/",SUBNAME,"/filtersummary_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))
p.fil <- plot(filt.test)
p.fil
pdf(paste0("plots/dabu/",SUBNAME,"/filterplot_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".pdf"))
    p.fil
    dev.off()

```

#### RUN DAtest

Having now identified LIMMA - CLR (lic) as the optimal test I will use it to calculate differential abundance

```{r DAtest_species, eval=TRUE, echo=TRUE, fig.width=10, fig.height=5}
# Run the selected analysis
filt.DA <- DA.ttc(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/stats_",MTRL,"_",TREAT,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Create plot
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])
p.dif <- ggplot(DAm, aes(x = Family, y = Abundance+pseudocount, color = day)) + geom_boxplot() + scale_y_log10() + coord_flip()
p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",TREAT))# + add_pvalue(filt.p5, xmin = "" label = "pval")
p.dT

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
h.plot <- plot_heatmap(DA.sig, sample.label = "Sample" ,sample.order = PDI, taxa.label = LVL, taxa.order = "Order", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",TREAT," day 2 and 8"))
h.plot

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_",SUBNAME,"_",TREAT,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```


## Correlation FECES d8 PFOS treated {.tabset .tabset-fade .tabset-pills}

There are many different methods that can be used to calculate differential abundance, all with their own advantages and disadvantages. Which method to use depends on the data and therefore I will be using DAtest [@DAtest] as described by [Russel et al. (2018)](https://doi.org/10.1101/241802).

### GENUS
#### TEST METHOD

```{r testDA_species, eval=TRUE, echo=TRUE,fig.width=10, fig.height=5}
params <- readRDS("R_objects/comp_params.RDS")

LVL <- "Genus"
MTRL <- "Feces"
PFOS <- "pfos"
VAN <- "van"            # Also tesing "ctrl"
DAY <- "d8"
PDI <- "PFOS_serum8_ug" # Also testing "PFOS_liver_mg"
SUBNAME <- "Feces_d8"
load(paste0("R_objects/Agglomerated_",SUBNAME,".RData"))


# Create sub-folder
  if (!file.exists(paste0("plots/dabu/",SUBNAME))) dir.create(file.path(getwd(), paste0("plots/dabu/",SUBNAME)))

# Filter samples (if needed)
phy.ge <- subset_samples(phy.ge, material == MTRL & pfos == PFOS & day == DAY & van == VAN)

# Filter data
filt <- preDA(data = phy.ge, min.reads = 20, min.samples = 4)
filt <- transform_sample_counts(filt, function(x) x/sum(x))
# Run the selected analysis
filt.DA <- DA.spe(filt, predictor = PDI)

# Evaluate the plot and summary table
table(filt.DA$pval < 0.05)
filt.p5 <- filt.DA[filt.DA$pval < 0.05,]
filt.p5
write.csv(filt.p5,paste0("plots/dabu/",SUBNAME,"/corr/stats_",MTRL,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".csv"))

# Create a subset of the samples
filt.ra <- transform_sample_counts(filt, function(x) x/sum(x)*100)
DA.sig <- prune_taxa(filt.DA$Feature[filt.DA$pval < 0.05], x = filt.ra)

# melt the data
DAm <- psmelt(DA.sig)

# Add pseudo count
pseudocount <- min(DAm$Abundance[DAm$Abundance != 0])

p.cor <- ggplot(DAm, aes(x = PFOS_liver_mg, y = Abundance+pseudocount, color = Genus)) + 
  geom_jitter() + 
  geom_smooth(method=lm) + 
  facet_wrap("Genus", scales = "free_y") + 
  scale_y_log10() +
  scale_x_continuous(breaks = seq(1.717, 2.931, 0.25)) + #liver: 1.717, 2.931, 0.25   Serum: 21.84, 69.92, 10
  theme_pubr()

p.cor

suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/corr/Correlation_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".pdf"), p.cor, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/corr/Correlation_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".png"), p.cor, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# DAm$sampleID <- with(DAm, reorder(sampleID, PFOS_liver_mg, median)) 

# Create plot

# p.dif <- ggplot(DAm, aes(x = Genus, y = Abundance+pseudocount, color = rat_name)) + geom_boxplot() + scale_y_log10() + coord_flip()
# p.dT <- p.dif + ggtitle(paste0("Differential abundance on ",LVL," level for ",MTRL," ",PFOS," ",VAN," ",DAY))# + add_pvalue(filt.p5, xmin = "" label = "pval")
# p.dT
# 
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".pdf"), p.dT, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/diffplot_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".png"), p.dT, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# Plot on heatmap
# h.plot <- plot_heatmap(DA.sig, sample.label = PDI ,sample.order = PDI, taxa.label = LVL, taxa.order = "Phylum", title = paste0(LVL," - Heatmap of ",MTRL," with treatment ",PFOS," ",VAN," ",DAY))
# h.plot
# 
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_corr_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".pdf"), h.plot, device = "pdf", units = "mm", width = 350, height = 200, dpi = 300))
# suppressMessages(ggsave(paste0("plots/dabu/",SUBNAME,"/heatmap_corr_",SUBNAME,"_",PFOS,"_",VAN,"_",DAY,"_",PDI,"_",LVL,".png"), h.plot, device = "png", units = "mm", width = 350, height = 200, dpi = 300))

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```





## SETTINGS {.tabset .tabset-fade .tabset-pills}

### EXPLANATION

Overview of the packages that was used for this analysis

### PACKAGES

The analysis was run in R version `getRversion()` using the following packages:

```{r packages, eval=TRUE}
pack <- data.frame(Package = (.packages()))

for (i in seq(nrow(pack))) pack$Version[i] <- as.character(packageVersion(pack$Package[i]))

kbl(pack[order(pack$Package),], row.names = F) %>% kable_classic(lightable_options = "striped")   
     
```

### PARAMETERS
```{r parameters, eval=TRUE}
params <- readRDS("R_objects/comp_params.RDS")

dat <- as.data.frame(params)

```
## References
