---
title: "3. Microbiome description (relative abundance)"
author: "Claus Lykkebo"
date: "`r Sys.Date()`"
output:
  rmdformats::readthedown:
    toc_depth: 4
    collapsed: false
    code_folding: hide
    number_sections: false
knit: (function(inputFile, encoding) { 
    rmarkdown::render(
        inputFile, encoding = encoding,
        output_dir = file.path(dirname(inputFile),"output"),
        output_file = paste0("Fibrex_", Sys.Date(), "_3_Description.html"))
        })
params:
    input: "R_objects/Phyloseq.Rdata"
    group_var: "feed_treat"
    subject_var: "rat_name"
    time_var: "day"
    COL: !r c("LF_CTRL" = "#a5cee3","LF_PFOS" = "#1778b6","HF_CTRL" = "#b4d88a","HF_PFOS" = "#30a148")
    COLFEED: !r c("LF" = "#1778b6","HF" = "#30a148")
---

# INFO

This document is build to use the output from **2_Import.Rmd** as input.
It will generate a general summary of the microbiome for the project, on
all taxonomic levels, followed by a detailed summary where the group and
subject variables is considered in the description.

The output will be observational, containing tables and plots, without
any statistical analyses.

At the end, code for generation of figures for the final article are included.

# SETUP
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(GMHmicrobiome)
library(ggpubr)
library(kableExtra)
library(phyloseq)
library(tidyverse)
library(decontam)
library(pals)
library(rstatix)
library(vegan)
library(reshape2)
library(ape)

# save parameters
saveRDS(params, "R_objects/params_description.RDS")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## VARIABLES

Group variable is set for **"feed_treat"** which is a combined variable in the format e.g. "LF_CTRL" or "HF_PFOS" grouping based on:
- type of feed (LF = low dietary fibre and HF = high dietary fibre)
- type of treatment given by oral gavage daily for 7 days (CTRL = control with pure corn oil; PFOS = 1.5mg/mL PFOS in corn oil)

Subject variable is set to **"rat_name"** in the format R01 - R48.

Time variable is set to **"day"** which includes days of sampling (day 0 = feces before treatment; day 8 = feces directly after treatment from all rats and ileum+cecum from half the rats; day 12 and 16 = 21)

# FORMAT DATA

To describe the data we will have to agglommerate the data to each taxonomic level. As this is time consuming, the default is to skip this step if the file "R_objects/Agglomerated.Rdata" does not exist.

```{r format, include=FALSE, eval=FALSE}
# Load data
MTRLS <- c("Feces","Ileum","Cecum")

# Subset
for (MTRL in MTRLS) {
  load(params$input)
  phy <- subset_samples(phy, material == MTRL & day != "d20")
  
  # Update taxa_names
  ### SPECIES
  phy.sp <- tax_glom(phy, taxrank = "Species")
  taxnames <- as.vector(tax_table(phy.sp)[,7])
  taxdub <- taxnames[duplicated(taxnames)]
  for (tax in taxdub){
    taxnames[taxnames == tax] <- paste(tax, seq(length(taxnames[taxnames == tax])), sep = "_")
  }
  taxa_names(phy.sp) <- taxnames
  
  ### GENUS
  phy.ge <- tax_glom(phy, taxrank = "Genus")
  taxnames <- as.vector(tax_table(phy.ge)[,6])
  taxdub <- taxnames[duplicated(taxnames)]
  for (tax in taxdub){
    taxnames[taxnames == tax] <- paste(tax, seq(length(taxnames[taxnames == tax])), sep = "_")
  }
  taxa_names(phy.ge) <- taxnames
  
  ### FAMILY
  phy.fa <- tax_glom(phy, taxrank = "Family")
  taxnames <- as.vector(tax_table(phy.fa)[,5])
  taxdub <- taxnames[duplicated(taxnames)]
  for (tax in taxdub){
    taxnames[taxnames == tax] <- paste(tax, seq(length(taxnames[taxnames == tax])), sep = "_")
  }
  taxa_names(phy.fa) <- taxnames
  
  ### ORDER
  phy.or <- tax_glom(phy, taxrank = "Order")
  taxnames <- as.vector(tax_table(phy.or)[,4])
  taxdub <- taxnames[duplicated(taxnames)]
  for (tax in taxdub){
    taxnames[taxnames == tax] <- paste(tax, seq(length(taxnames[taxnames == tax])), sep = "_")
  }
  taxa_names(phy.or) <- taxnames
  
  ### CLASS
  phy.cl <- tax_glom(phy, taxrank = "Class")
  taxnames <- as.vector(tax_table(phy.cl)[,3])
  taxdub <- taxnames[duplicated(taxnames)]
  for (tax in taxdub){
    taxnames[taxnames == tax] <- paste(tax, seq(length(taxnames[taxnames == tax])), sep = "_")
  }
  taxa_names(phy.cl) <- taxnames
  
  ### PHYLUM
  phy.ph <- tax_glom(phy, taxrank = "Phylum")
  taxnames <- as.vector(tax_table(phy.ph)[,2])
  taxdub <- taxnames[duplicated(taxnames)]
  for (tax in taxdub){
    taxnames[taxnames == tax] <- paste(tax, seq(length(taxnames[taxnames == tax])), sep = "_")
  }
  taxa_names(phy.ph) <- taxnames
  
  # save agglomerated phyloseq objects
  save(phy.sp, phy.ge, phy.fa, phy.or, phy.cl, phy.ph, file = paste0("R_objects/Agglomerated_",MTRL,".RData"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# FECES

## OVERALL {.tabset .tabset-dropdown}

This section produces a general description of the data with the plots
grouped by the group_var, but without using any other metadata.

If there are no relevant groups in this part of the analysis it can also
be used to indicate batches. If no groups or batches exist, then please
create a variable that just indicate the project name for all samples.

### PHYLUM {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At phylum level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.ph, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of phyla in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### CLASS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At class level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.cl, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of classes in general and per sample') %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### ORDER {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At order level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.or, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of orders in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### FAMILY {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At family level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.fa, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of families in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### GENUS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At genus level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.ge, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of genera in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SPECIES {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At species level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.sp, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of species in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus, Species) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## DETAILED {.tabset .tabset-fade .tabset-dropdown}

Many projects contain multiple timepoints and for those it can be
relevant to see data organised by group and time. For the plots in this
part of the script it is required that all parameters set in the header
matches a sample variable. Also, the plots will be created with the
subject_var on the x-axis instead of the sample var, so it is important
that there are no duplicates when combining subject_var and time_var.

### PHYLUM {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At phylum level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.ph)[!is.na(sample_data(phy.ph)[,params$group_var]) & !is.na(sample_data(phy.ph)[,params$time_var])], phy.ph)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of phyla in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa, .keep_all = T)%>% select(Taxa, Kingdom, Phylum) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### CLASS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At class level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.cl)[!is.na(sample_data(phy.cl)[,params$group_var]) & !is.na(sample_data(phy.cl)[,params$time_var])], phy.cl)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of classes in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### ORDER {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At order level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.or)[!is.na(sample_data(phy.or)[,params$group_var]) & !is.na(sample_data(phy.or)[,params$time_var])], phy.or)


# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of orders in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### FAMILY {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At family level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.fa)[!is.na(sample_data(phy.fa)[,params$group_var]) & !is.na(sample_data(phy.fa)[,params$time_var])], phy.fa)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of families in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### GENUS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At genus level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.ge)[!is.na(sample_data(phy.ge)[,params$group_var]) & !is.na(sample_data(phy.ge)[,params$time_var])], phy.ge)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of genera in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(day ~ feed_treat,
             space = "free_x",scales = "free_x",
             labeller = labeller(day = c("d0" = "Day 0", "d08" = "Day 8", "d12" = "Day 12", "d16" = "Day 16", "d20" = "Day 20", "d21" = "Day 21"),
                                 feed_treat = c("HF_CTRL" = "HF-CTRL","HF_PFOS"="HF-PFOS","LF_CTRL"="LF-CTRL","LF_PFOS"="LF-PFOS"))) +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

p.fig <- p + theme_pubr(legend = "right", border = TRUE) + theme(axis.text.x = element_text(angle = 90))

if (!file.exists(paste0("plots/relative_abundance"))) dir.create(file.path(getwd(), paste0("plots/relative_abundance")))
ggsave(filename = "plots/figures/Genus_abundance_faeces.png", plot = p.fig, device = "png", units = "mm", width = 300, height = 185, dpi = 300)
ggsave(filename = "plots/figures/Genus_abundance_faeces.pdf", plot = p.fig, device = "pdf", units = "mm", width = 300, height = 185, dpi = 300)
ggsave(filename = "plots/figures/final_figures_Appendix/Figure A.6.png", plot = p.fig, device = "png", units = "mm", width = 300, height = 185, dpi = 300)
ggsave(filename = "plots/figures/final_figures_Appendix/Figure A.6.pdf", plot = p.fig, device = "pdf", units = "mm", width = 300, height = 185, dpi = 300)

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family,Genus)  %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SPECIES {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At species level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Feces.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.sp)[!is.na(sample_data(phy.sp)[,params$group_var]) & !is.na(sample_data(phy.sp)[,params$time_var])], phy.sp)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of species in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family,Genus,Species) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# ILEUM

## OVERALL {.tabset .tabset-dropdown}

This section produces a general description of the data with the plots
grouped by the group_var, but without using any other metadata.

If there are no relevant groups in this part of the analysis it can also
be used to indicate batches. If no groups or batches exist, then please
create a variable that just indicate the project name for all samples.

### PHYLUM {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At phylum level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.ph, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of phyla in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### CLASS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At class level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.cl, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of classes in general and per sample') %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### ORDER {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At order level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.or, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of orders in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### FAMILY {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At family level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.fa, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of families in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### GENUS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At genus level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.ge, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of genera in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SPECIES {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At species level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.sp, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of species in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus, Species) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## DETAILED {.tabset .tabset-fade .tabset-dropdown}

Many projects contain multiple timepoints and for those it can be
relevant to see data organised by group and time. For the plots in this
part of the script it is required that all parameters set in the header
matches a sample variable. Also, the plots will be created with the
subject_var on the x-axis instead of the sample var, so it is important
that there are no duplicates when combining subject_var and time_var.

### PHYLUM {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At phylum level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.ph)[!is.na(sample_data(phy.ph)[,params$group_var]) & !is.na(sample_data(phy.ph)[,params$time_var])], phy.ph)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of phyla in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa, .keep_all = T)%>% select(Taxa, Kingdom, Phylum) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### CLASS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At class level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.cl)[!is.na(sample_data(phy.cl)[,params$group_var]) & !is.na(sample_data(phy.cl)[,params$time_var])], phy.cl)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of classes in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### ORDER {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At order level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.or)[!is.na(sample_data(phy.or)[,params$group_var]) & !is.na(sample_data(phy.or)[,params$time_var])], phy.or)


# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of orders in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### FAMILY {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At family level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.fa)[!is.na(sample_data(phy.fa)[,params$group_var]) & !is.na(sample_data(phy.fa)[,params$time_var])], phy.fa)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of families in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### GENUS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At genus level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.ge)[!is.na(sample_data(phy.ge)[,params$group_var]) & !is.na(sample_data(phy.ge)[,params$time_var])], phy.ge)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of genera in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(day ~ feed_treat,
             space = "free_x",scales = "free_x",
             labeller = labeller(day = c("d0" = "Day 0", "d08" = "Day 8", "d12" = "Day 12", "d16" = "Day 16", "d20" = "Day 20", "d21" = "Day 21"),
                                 feed_treat = c("HF_CTRL" = "HF-CTRL","HF_PFOS"="HF-PFOS","LF_CTRL"="LF-CTRL","LF_PFOS"="LF-PFOS"))) +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

p.fig <- p + theme_pubr(legend = "right", border = TRUE) + theme(axis.text.x = element_text(angle = 90))

if (!file.exists(paste0("plots/relative_abundance"))) dir.create(file.path(getwd(), paste0("plots/relative_abundance")))
ggsave(filename = "plots/figures/Genus_abundance_ileum.png", plot = p.fig, device = "png", units = "mm", width = 300, height = 185, dpi = 300)
ggsave(filename = "plots/figures/Genus_abundance_ileum.pdf", plot = p.fig, device = "pdf", units = "mm", width = 300, height = 185, dpi = 300)

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family,Genus)  %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SPECIES {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At species level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Ileum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.sp)[!is.na(sample_data(phy.sp)[,params$group_var]) & !is.na(sample_data(phy.sp)[,params$time_var])], phy.sp)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of species in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family,Genus,Species) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# CAECUM

## OVERALL {.tabset .tabset-dropdown}

This section produces a general description of the data with the plots
grouped by the group_var, but without using any other metadata.

If there are no relevant groups in this part of the analysis it can also
be used to indicate batches. If no groups or batches exist, then please
create a variable that just indicate the project name for all samples.

### PHYLUM {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At phylum level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.ph, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of phyla in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### CLASS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At class level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.cl, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of classes in general and per sample') %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### ORDER {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At order level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.or, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of orders in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### FAMILY {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At family level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.fa, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of families in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### GENUS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At genus level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.ge, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of genera in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SPECIES {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At species level the microbiome counts are as follows

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Transform data
phy.rel <- transform_sample_counts(phy.sp, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(-Sample, 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff) %>% 
  summarise(sample_mean = mean(Count), 
            sample_sd = sd(Count))

# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU) %>% 
  summarise(Abundance = mean(Abundance)) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>% 
  pivot_longer(cols = everything(), 
               names_to = c("Cutoff"), 
               values_to = "Count")

# Combine
output <- full_join(sumall, sumsample)
output$Cutoff <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of species in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# Create plot
p <- ggplot(dat.sort, aes(x = SampleID,
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(. ~ get(params$group_var),space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, Class, Order, Family, Genus, Species) %>% summarise(Abundance = mean(Abundance))


kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

## DETAILED {.tabset .tabset-fade .tabset-dropdown}

Many projects contain multiple timepoints and for those it can be
relevant to see data organised by group and time. For the plots in this
part of the script it is required that all parameters set in the header
matches a sample variable. Also, the plots will be created with the
subject_var on the x-axis instead of the sample var, so it is important
that there are no duplicates when combining subject_var and time_var.

### PHYLUM {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At phylum level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.ph)[!is.na(sample_data(phy.ph)[,params$group_var]) & !is.na(sample_data(phy.ph)[,params$time_var])], phy.ph)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of phyla in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa, .keep_all = T)%>% select(Taxa, Kingdom, Phylum) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### CLASS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At class level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.cl)[!is.na(sample_data(phy.cl)[,params$group_var]) & !is.na(sample_data(phy.cl)[,params$time_var])], phy.cl)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of classes in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### ORDER {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At order level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.or)[!is.na(sample_data(phy.or)[,params$group_var]) & !is.na(sample_data(phy.or)[,params$time_var])], phy.or)


# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of orders in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var,min.rank = 6,includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### FAMILY {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At family level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.fa)[!is.na(sample_data(phy.fa)[,params$group_var]) & !is.na(sample_data(phy.fa)[,params$time_var])], phy.fa)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of families in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### GENUS {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At genus level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.ge)[!is.na(sample_data(phy.ge)[,params$group_var]) & !is.na(sample_data(phy.ge)[,params$time_var])], phy.ge)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of genera in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(day ~ feed_treat,
             space = "free_x",scales = "free_x",
             labeller = labeller(day = c("d0" = "Day 0", "d08" = "Day 8", "d12" = "Day 12", "d16" = "Day 16", "d20" = "Day 20", "d21" = "Day 21"),
                                 feed_treat = c("HF_CTRL" = "HF-CTRL","HF_PFOS"="HF-PFOS","LF_CTRL"="LF-CTRL","LF_PFOS"="LF-PFOS"))) +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

p.fig <- p + theme_pubr(legend = "right", border = TRUE) + theme(axis.text.x = element_text(angle = 90))

if (!file.exists(paste0("plots/relative_abundance"))) dir.create(file.path(getwd(), paste0("plots/relative_abundance")))
ggsave(filename = "plots/figures/Genus_abundance_caecum.png", plot = p.fig, device = "png", units = "mm", width = 300, height = 185, dpi = 300)
ggsave(filename = "plots/figures/Genus_abundance_caecum.pdf", plot = p.fig, device = "pdf", units = "mm", width = 300, height = 185, dpi = 300)

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family,Genus)  %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SPECIES {.tabset .tabset-fade .tabset-pills}

#### COUNTS

At species level the microbiome counts are as follows for each group and
timepoint

```{r, warning=FALSE, message=FALSE}
# load data 
load("R_objects/Agglomerated_Cecum.RData")
params <- readRDS("R_objects/params_description.RDS")

# Remove samples with missing data
phy.use <- prune_samples(sample_names(phy.sp)[!is.na(sample_data(phy.sp)[,params$group_var]) & !is.na(sample_data(phy.sp)[,params$time_var])], phy.sp)

# Transform data
phy.rel <- transform_sample_counts(phy.use, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# summarise per sample
sumsample <- 
  dat %>% 
  filter(Abundance > 0) %>% 
  group_by(Sample, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "Count") %>% 
  group_by(Cutoff, .data[[params$group_var]], .data[[params$time_var]]) %>% 
  summarise(mean = round(mean(Count),2), 
            sd = round(sd(Count),2)) 
  
# summarise total
sumall <- dat %>% 
  filter(Abundance > 0) %>% 
  group_by(OTU, .data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(.data[[params$group_var]], .data[[params$time_var]]) %>%
  summarise(pct_0 = n(),
            pct_0.1 = sum(Abundance < 0.1),
            pct_0_1 = sum(Abundance >= 0.1 & Abundance < 1),
            pct_1_10 = sum(Abundance >= 1 & Abundance < 10),
            pct_10 = sum(Abundance > 10)) %>%
  pivot_longer(starts_with("pct"), 
               names_to = c("Cutoff"), 
               values_to = "n")

# Combine
output <- full_join(sumall, sumsample) %>% 
  pivot_wider(id_cols = c(Cutoff, .data[[params$time_var]]), 
              names_from = .data[[params$group_var]],
              values_from = c(n, mean, sd), 
              names_vary = "slowest") %>%
  mutate(Cutoff=factor(Cutoff))


levels(output$Cutoff) <- c("All", "n < 0.1%", "0.1% < n < 1.0%", "1.0% < n < 10%", "10%  < n")

# Create output table
kable(output, row.names = F,digits = 2, caption = 'Count of species in general and per sample',align = "r") %>% 
  kable_classic(full_width = F, position = "left")
                                              
```

#### PLOTS

To improve the interpretation of the compositional plots I will pretreat
the data with three steps:

1.  Filter taxa by average abundance (minimum 1 %) and rank (max 19
    features + "Others")

2.  Sort the plotted taxa by full phylogeny

3.  Change taxa names in plot to a letter and include table of full
    taxonomy

Following this I will incorporate both time_var and subject_var in the
plot

```{r, warning=FALSE, message=FALSE}
# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel, group = params$group_var,includes = "any") %>% filter_rank(group = params$time_var, min.rank = 6, includes = "any")

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# ensure that the variables are factors
dat.sort[,params$time_var] <- factor(dat.sort[,params$time_var])
dat.sort[,params$subject_var] <- factor(dat.sort[,params$subject_var])
dat.sort[,params$group_var] <- factor(dat.sort[,params$group_var])

# Create plot
p <- ggplot(dat.sort, aes(x = .data[[params$subject_var]],
                          y = Abundance,
                          fill = Taxa,
                          color = Taxa)) +
  geom_col() +
  xlab("") + 
  facet_grid(get(params$time_var) ~ get(params$group_var),
             space = "free_x",scales = "free_x") +
  ggsci::scale_color_d3(palette = "category20") + 
  ggsci::scale_fill_d3(palette = "category20") + 
  guides(fill=guide_legend(ncol=2), color=guide_legend(ncol=2)) +
  clean_theme()
p

# print tax table
tax <- dat.sort %>% distinct(Taxa,.keep_all = T) %>% select(Taxa, Kingdom, Phylum, Class, Order,Family,Genus,Species) %>% arrange(Taxa)

kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
  kable_classic(full_width = F, position = "left")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# SPECIAL FIGURES FOR ARTICLE
## Figure A.5: PHYLA {.tabset .tabset-dropdown}
### Prep data
```{r}
# load data 
load("R_objects/Agglomerated.RData")
rm(phy.sp, phy.ge, phy.fa, phy.or, phy.cl)
params <- readRDS("R_objects/params_description.RDS")
LIST.DAY <- c("d0","d08","d12","d16","d21", "all")
DIETS <- c("HF","LF")
LIST.TR <- c("CTRL","PFOS", "both")
LVL <- "Phylum"

# Prepare dataframe
df <- data.frame()

# Run calculations
suppressMessages({suppressWarnings({
for (DIET in DIETS) {
  #print(DIET)
  load("R_objects/Agglomerated.RData")
  phy.feed <- phy.ph
  phy.feed <- subset_samples(phy.feed, feed == DIET)
  
  for (DAY in LIST.DAY) {
    if (DAY == "all") {
      print("## All days selected")
      phy.day <- phy.feed
    } else {
      print(paste0("## Day ",DAY," selected"))
      phy.day <- subset_samples(phy.feed, day == DAY)
    }
    
    for (TR in LIST.TR) {
      if (TR == "both") {
        # print("## Both treatment groups selected")
        phy.tr <- phy.day
      } else {
        # print(paste0("## Treatment ",TR," selected"))
        phy.tr <- subset_samples(phy.day, treatment == TR)
      }
      # Transform data
      phy.rel <- transform_sample_counts(phy.tr, fun = function(x) x/sum(x)*100)
      dat <- psmelt(phy.rel)
      
      # Filter by abundance, then rank
      phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)
      
      #Melt data
      dat <- suppressWarnings(psmelt(phy.top))
      
      # Prevent duplicate column names
      colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")
      
      # Rename relevant columns
      colnames(dat)[c(1,4)] <- c("Taxa","SampleID")
      
      # Sort taxa
      dat.sort <- sort_taxa(dat) 
      
      # rename taxa
      levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]
      
      # print tax table
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))
      
      # Bacteroidetes-Firmicutes ratio
      Ba <- subset(tax, Phylum == "Bacteroidetes")
      Fi <- subset(tax, Phylum == "Firmicutes")
      Pb <- subset(tax, Phylum == "Proteobacteria")
      if (nrow(Pb) == 0) {
        print(paste0("Zero Preoteobacteria in ",DIET," ",DAY," ",TR))
        df_0 <- c("D","Bacteria","Proteobacteria",0)
        Pb <- data.frame()
        Pb <- rbind(Pb,df_0)
        colnames(Pb) <- c("Taxa","Kingdom","Phylum","Abundance")
        Pb <- transform(Pb, Abundance = as.numeric(Abundance))
        } else {}
      Ac <- subset(tax, Phylum == "Actinobacteria")
      if (nrow(Ac) == 0) {
        print(paste0("Zero Actinobacteria in ",DIET," ",DAY," ",TR))
        df_0 <- c("A","Bacteria","Actinobacteria",0)
        Ac <- data.frame()
        Ac <- rbind(Ac,df_0)
        colnames(Ac) <- c("Taxa","Kingdom","Phylum","Abundance")
        Ac <- transform(Ac, Abundance = as.numeric(Abundance))
        } else {}
      OT_FB <- 100 - (round(Fi$Abundance,3) + round(Ba$Abundance,3))
      OT_ALL <- 100 - (round(Fi$Abundance,3) + round(Ba$Abundance,3) + round(Pb$Abundance,3) + round(Ac$Abundance,3))
      FBR <- Fi$Abundance / Ba$Abundance 
      
      output <- c(DIET, DAY, TR, paste0(DIET,"_",TR), round(Fi$Abundance,3), round(Ba$Abundance,3),round(Pb$Abundance,3),round(Ac$Abundance,3), OT_FB, OT_ALL, FBR) #round(Ab$Abundance,3),
      
      df <- rbind(df, output)
      
    }
  }
}
 })})

# Names columns
colnames(df) <- c("feed","day","treatment","feed_treat",
                  "Firmicutes","Bacteroidetes","Proteobacteria","Actinobacteria","Others_FB","Others_all","FB_ratio")
df
df <- transform(df, Firmicutes = as.numeric(Firmicutes),
                Bacteroidetes = as.numeric(Bacteroidetes),
                Proteobacteria = as.numeric(Proteobacteria),
                Actinobacteria = as.numeric(Actinobacteria),
                Others_FB = as.numeric(Others_FB),
                Others_all = as.numeric(Others_all),
                FB_ratio = as.numeric(FB_ratio))
df
write.csv(df, file = "plots/relative_abundance/Phyla_relative_abundance.csv")

# Prepare Firmicutes and Bacteroidetes for plotting
df.fb <- pivot_longer(df, cols = c(Firmicutes, Bacteroidetes, Others_FB), names_to = "Phylum", values_to = "Abundance")
df.fb <- subset(df.fb, treatment != "both" & day != "all")
df.fb

# Plot Firmicutes + Bacteroidetes
p <- ggplot(df.fb, aes(fill = Phylum, y = Abundance, x = feed)) +
  geom_bar(position = "fill", stat = "identity") +
  facet_grid(treatment ~ day, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  geom_text(aes(label = paste0(round(Abundance,2),"%")), position = position_fill(vjust = 0.5), size = 3, color = "#222222") +
  ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others_FB" = "Others")) +
  scale_y_continuous(labels = function(x) paste0(x*100, "%"), name = "Relative Abundance") +
  theme_pubr(legend = "top", border = TRUE) +
  theme(axis.title.x = element_blank())

p

ggsave(filename = "plots/relative_abundance/Firmicutes-Bacteroidetes-ratios.png", plot = p, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Firmicutes-Bacteroidetes-ratios.pdf", plot = p, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# Prepare All for plotting
df.all <- pivot_longer(df, cols = c(Firmicutes, Bacteroidetes, Proteobacteria, Actinobacteria, Others_all), names_to = "Phylum", values_to = "Abundance")
df.both <- subset(df.all, treatment == "both" & day != "all")
df.both
df.all <- subset(df.all, treatment != "both" & day != "all")
df.all

l <- c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Others_all")

df.all$Phylum <- factor(df.all$Phylum, levels = c(l))

if (!file.exists("plots/figures/objects")) dir.create(file.path(getwd(), "plots/figures/objects"))
save(df,df.fb,df.all,df.both, file = "plots/figures/objects/phyla_means.Rdata")

```

The following section compare relative abundances and the ratio between Firmicutes and Bacteroidetes in data daily and in respect to factors such as HF/LF feed and PFOS exposure. 

### Create figure A.5
```{r}
# load data
load("R_objects/Agglomerated.RData")
load("plots/figures/objects/phyla_means.Rdata")
params <- readRDS("R_objects/params_description.RDS")
LVL <- "Phylum"

load("R_objects/Agglomerated.RData")
phy.used <- phy.ph

  # Transform data
phy.rel <- transform_sample_counts(phy.used, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat)

# # rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))

# Subset to phyla and order
dat.used <- subset(dat.sort, Phylum %in% c("Firmicutes", "Bacteroidetes"))
df.all <- subset(df.all, Phylum %in% c("Firmicutes", "Bacteroidetes"))
l <- c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria","Others")
d <- c("d21","d16","d12","d08","d0")

dat.used$Phylum <- factor(dat.used$Phylum, levels = c(l))
dat.used$day <- factor(dat.used$day, levels = c(d))

df.all$Phylum <- factor(df.all$Phylum, levels = c(l))
df.all$day <- factor(df.all$day, levels = c(d))

# Set parameters
PREDICTOR <- c("feed","day","Phylum")
OUTCOME <- "Abundance"
SUBJECT <- "rat_name"

# Create formula
PREDICTOR.F <- ifelse(length(PREDICTOR) > 1, paste(PREDICTOR, collapse = "*"), PREDICTOR)
FORMULA <- as.formula(paste(OUTCOME,PREDICTOR.F, sep = " ~ "))

# Summary samples in groups
dat.used %>% group_by(across(all_of(PREDICTOR))) %>% get_summary_stats(!!sym(OUTCOME), type = "mean_sd")

# Test for outliers
dat.used %>%
  group_by(across(all_of(PREDICTOR))) %>%
  identify_outliers(!!sym(OUTCOME))

# Check normality
# Build the linear model
model  <- lm(FORMULA, data = dat.used)
# Create a QQ plot of residuals
ggqqplot(residuals(model))

#Check for variance homogeneity
# Compute Shapiro-Wilk test of normality
shapiro_test(residuals(model))
dat.used %>% levene_test(FORMULA)

# Save result
EQUAL.VAR <- dat.used %>% levene_test(FORMULA) %>% pull(p) > 0.05

## Abundance data is not normally distributed and does not have equal variance
## We use Wilcoxon and Kruskal-Wallis

# Plot for comparing PFOS influence
# Statistics costumed for facet plotting
stat.in <- dat.used %>%
  group_by(Phylum,feed,day) %>%
  wilcox_test(as.formula("Abundance ~ treatment")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance() %>%
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.in

stat.out <- dat.used %>%
  group_by(feed, Phylum) %>%
  kruskal_test(as.formula("Abundance ~ day")) %>%
  add_significance() %>%
  p_format("p", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.out

stat.out2 <- dat.used %>%
  group_by(feed, Phylum) %>%
  dunn_test(as.formula("Abundance ~ day"), p.adjust.method = "fdr") %>%
  add_significance() %>%
  add_xy_position(x = "day") %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.out2

dat.used$feed_treat <- factor(as.character(dat.used$feed_treat))

# Plot check PFOS
p.pfos <- ggplot(dat.used) +
  geom_bar(data = df.all, aes(fill = fct_rev(feed_treat), y = Abundance, x = day), stat = "identity", position = position_dodge()) +
  facet_grid(Phylum ~ feed, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  #ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others" = "Others")) +
  geom_text(data = df.all, aes(label = paste0(round(Abundance,1),"%"), x = day, y = Abundance/2, group = fct_rev(feed_treat)), position = position_dodge(width = 0.9), size = 3, color = "black") +
  coord_flip() +
  scale_fill_manual(values = params$COL, name = "Group", labels = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS"), breaks = c("HF_CTRL","HF_PFOS","LF_CTRL","LF_PFOS"))+
  scale_color_manual(values = c("black","black","black","black")) +
  theme_pubr(legend = "top", border = TRUE) +
  scale_x_discrete(name = "Day", labels = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21")) +
  theme(axis.title.y = element_blank())
p.pfos

p.done <- p.pfos + stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red", coord.flip = TRUE) +
  stat_pvalue_manual(stat.out2, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, coord.flip = TRUE) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance", limits = c(0,100))
p.done

ggsave(filename = "plots/relative_abundance/Firmicutes_Bacteroidetes_relative_abundance.png", plot = p.done, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Firmicutes_Bacteroidetes_relative_abundance.pdf", plot = p.done, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# Plot for comparing group comparison
# Statistics costumed for facet plotting
stat.in1 <- dat.used %>%
  group_by(feed_treat, Phylum) %>%
  kruskal_test(as.formula("Abundance ~ day")) %>%
  add_significance() %>%
  p_format("p", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.in1

stat.in2 <- dat.used %>%
  group_by(feed_treat, Phylum) %>%
  dunn_test(as.formula("Abundance ~ day"), p.adjust.method = "fdr") %>%
  add_significance() %>%
  add_xy_position(x = "day") %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.in2

dat.used$feed_treat <- factor(as.character(dat.used$feed_treat))

# Plot check PFOS
p.group <- ggplot(dat.used) +
  geom_bar(data = df.all, aes(fill = fct_rev(feed_treat), y = Abundance, x = day), stat = "identity", position = position_dodge()) +
  geom_point(data = dat.used, aes(x = day, y = Abundance, color = feed_treat), position = position_jitter(width = 0.2), size = 0.5, alpha = 0.8) +
  facet_grid(Phylum ~ feed_treat, labeller = labeller(feed_treat = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS"))) +
  #ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others" = "Others")) +
  geom_text(data = df.all, aes(label = paste0(round(Abundance,1),"%"), x = day, y = Abundance/2, group = fct_rev(feed_treat), color = treatment), size = 3) +
  coord_flip() +
  scale_fill_manual(values = params$COL, name = "Group", labels = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS"), breaks = c("HF_CTRL","HF_PFOS","LF_CTRL","LF_PFOS"))+
  scale_color_manual(values = c("LF_CTRL" = "#1778b6","LF_PFOS" = "#1748b6","HF_CTRL" = "#84b559","HF_PFOS" = "#306148", "CTRL" = "black","PFOS" = "white")) +
  theme_pubr(legend = "top", border = TRUE) +
  scale_x_discrete(name = "Day", labels = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21")) +
  theme(axis.title.y = element_blank(), axis.text.x = element_text(angle = 90)) +
  guides(color = "none", alpha = "none")
p.group

p.stat <- p.group + stat_pvalue_manual(stat.in2, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, coord.flip = TRUE) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance", limits = c(0,100))
p.stat

ggsave(filename = "plots/figures/SUP_Figure A4 - Firmicutes_Bacteroidetes_relative_abundance_group.png", plot = p.stat, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/figures/SUP_Figure A4 - Firmicutes_Bacteroidetes_relative_abundance_group.pdf", plot = p.stat, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
#   kable_classic(full_width = F, position = "left")
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```
### Create Figure A.5 with supplements
```{r}
load("R_objects/Agglomerated.RData")
load("plots/figures/objects/phyla_means.Rdata")
params <- readRDS("R_objects/params_description.RDS")
LVL <- "Phylum"
phy.used <- phy.ph

# Transform data
phy.rel <- transform_sample_counts(phy.used, fun = function(x) x/sum(x)*100)
dat <- psmelt(phy.rel)

# Filter by abundance, then rank
phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)

#Melt data
dat <- suppressWarnings(psmelt(phy.top))

# Prevent duplicate column names
colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")

# Rename relevant columns
colnames(dat)[c(1,4)] <- c("Taxa","SampleID")

# Sort taxa
dat.sort <- sort_taxa(dat) 

# # rename taxa
levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]

# print tax table
tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum, day, feed) %>% summarise(Abundance = mean(Abundance))

tax <- subset(tax, Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"))

# Subset to phyla and order
dat.used <- subset(dat.sort, Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"))
df.all <- subset(df.all, Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"))
l <- c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria","Others")
d <- c("d21","d16","d12","d08","d0")

dat.used$Phylum <- factor(dat.used$Phylum, levels = c(l))
dat.used$day <- factor(dat.used$day, levels = c(d))

df.all$Phylum <- factor(df.all$Phylum, levels = c(l))
df.all$day <- factor(df.all$day, levels = c(d))

tax$Phylum <- factor(tax$Phylum, levels = c(l))
tax$day <- factor(tax$day, levels = c(d))

# Set parameters
PREDICTOR <- c("feed","day","Phylum")
OUTCOME <- "Abundance"
SUBJECT <- "rat_name"

# Create formula
PREDICTOR.F <- ifelse(length(PREDICTOR) > 1, paste(PREDICTOR, collapse = "*"), PREDICTOR)
FORMULA <- as.formula(paste(OUTCOME,PREDICTOR.F, sep = " ~ "))

# Summary samples in groups
dat.used %>% group_by(across(all_of(PREDICTOR))) %>% get_summary_stats(!!sym(OUTCOME), type = "mean_sd")

# Test for outliers
dat.used %>%
  group_by(across(all_of(PREDICTOR))) %>%
  identify_outliers(!!sym(OUTCOME))

# Check normality
# Build the linear model
model  <- lm(FORMULA, data = dat.used)
# Create a QQ plot of residuals
ggqqplot(residuals(model))

#Check for variance homogeneity
# Compute Shapiro-Wilk test of normality
shapiro_test(residuals(model))
dat.used %>% levene_test(FORMULA)

# Save result
EQUAL.VAR <- dat.used %>% levene_test(FORMULA) %>% pull(p) > 0.05

## Abundance data is not normally distributed and does not have equal variance
## We use Wilcoxon and Kruskal-Wallis

# Plot for comparing diet influence
# Statistics costumed for facet plotting
stat.in <- dat.used %>%
  group_by(day,Phylum) %>%
  wilcox_test(as.formula("Abundance ~ feed")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance() %>%
  add_xy_position(x = "day", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.in

stat.out <- dat.used %>%
  group_by(Phylum) %>%
  kruskal_test(as.formula("Abundance ~ day")) %>%
  add_significance() %>%
  p_format("p", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.out

stat.out2 <- dat.used %>%
  group_by(Phylum) %>%
  dunn_test(as.formula("Abundance ~ day"), p.adjust.method = "fdr") %>%
  add_significance() %>%
  add_xy_position(x = "day") %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.out2

dat.used$feed <- factor(as.character(dat.used$feed))

df.both <- subset(df.both, Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"))
df.both$Phylum <- factor(df.both$Phylum, levels = c(l))
df.both$day <- factor(df.both$day, levels = c(d))

df.both

# Plot check PFOS
p.feed <- ggplot(dat.used, aes()) +
  geom_bar(data = df.both, aes(fill = fct_rev(feed), y = Abundance, x = day), stat = "identity", position = position_dodge()) +
  geom_point(data = dat.used, aes(x = day, y = Abundance, color = fct_rev(feed)), position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), size = 0.5, alpha = 0.5) +
  facet_grid(Phylum ~ .) +
  geom_text(data = df.both, aes(label = paste0(round(Abundance,1),"%"), x = day, y = Abundance+1, group = fct_rev(feed)), position = position_dodge(width = 0.9), size = 3, hjust = 0) +
  coord_flip() +
  scale_fill_manual(values = params$COLFEED, name = "Group")+
  scale_color_manual(values = c("LF" = "#1748b6","HF" = "#306148")) +
  theme_pubr(legend = "top", border = TRUE) +
  scale_x_discrete(name = "Day", labels = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21")) +
  theme(axis.title.y = element_blank()) +
  guides(color = "none")
p.feed

p.done <- p.feed + stat_pvalue_manual(stat.in, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red", coord.flip = TRUE, y.position = c(80.6,55.4,14.1,86.3,57.4,13.5,14.7,74.5,61.7,15.9,75.9,76.1,12.8)) +
  stat_pvalue_manual(stat.out2, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, coord.flip = TRUE, y.position = c(20)) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance", limits = c(0,100), expand = c(0.01,0.01))
p.done

ggsave(filename = "plots/relative_abundance/Major_phyla_abundance.png", plot = p.done, device = "png", unit = "mm", dpi = 300, height = 200, width = 200)
ggsave(filename = "plots/relative_abundance/Major_phyla_abundance.pdf", plot = p.done, device = "pdf", unit = "mm", dpi = 300, height = 200, width = 200)

# Plot for comparing treatment influence
# Statistics costumed for facet plotting
df.all <- subset(df.all, Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"))
df.all$Phylum <- factor(df.all$Phylum, levels = c(l))
df.all$day <- factor(df.all$day, levels = c(d))
dat.used$day <- factor(dat.used$day, levels = c(d))
dat.used$feed <- factor(dat.used$feed, levels = c("LF","HF"))
df.all$feed <- factor(df.all$feed, levels = c("LF","HF"))

stat.in_2 <- dat.used %>%
  group_by(feed,day,Phylum) %>%
  wilcox_test(as.formula("Abundance ~ feed_treat")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance() %>%
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.in_2

stat.out_2 <- dat.used %>%
  group_by(day,Phylum) %>%
  wilcox_test(as.formula("Abundance ~ feed")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance() %>%
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.out_2

dat.used$feed_treat <- factor(as.character(dat.used$feed_treat))

d <- c("d0","d08","d12","d16","d21")

df.all <- subset(df.all, Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"))
df.all$Phylum <- factor(df.all$Phylum, levels = c(l))
df.all$day <- factor(df.all$day, levels = c(d))
dat.used$day <- factor(dat.used$day, levels = c(d))
dat.used$feed <- factor(dat.used$feed, levels = c("LF","HF"))
df.all$feed <- factor(df.all$feed, levels = c("LF","HF"))

p.fig <- ggplot(dat.used) +
  geom_bar(data = df.all, aes(fill = fct_rev(feed_treat), y = Abundance, x = feed), stat = "identity", position = position_dodge()) +
  geom_point(data = dat.used, aes(x = feed, y = Abundance, color = fct_rev(feed_treat)), position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), size = 0.5, alpha = 0.5) +
  geom_text(data = df.all, aes(label = paste0(round(Abundance,1),"%"), x = feed, y = ifelse(Abundance<25, Abundance+15, Abundance/2), group = fct_rev(feed_treat)), position = position_dodge(width = 0.9), size = 3, hjust = 0.5) +
  facet_grid(Phylum ~ day, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  coord_flip() +
  scale_fill_manual(values = params$COL, name = "", labels = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS"), breaks = c("HF_CTRL", "HF_PFOS","LF_CTRL","LF_PFOS")) +
  scale_color_manual(values = c("LF_CTRL" = "#1778b6","LF_PFOS" = "#1748b6","HF_CTRL" = "#84b559","HF_PFOS" = "#306148")) +
  theme_pubr(legend = "top", border = TRUE) +
  scale_x_discrete(name = "Feed") +
  theme(axis.title.y = element_blank(),
        axis.text.x = element_text(size = 10, angle = 90, vjust = 0.5, hjust = 1)) +
  guides(color = "none")
p.fig

p.fig1 <- p.fig + stat_pvalue_manual(stat.in_2, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, color = "red", coord.flip = TRUE) +
  stat_pvalue_manual(stat.out_2, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, coord.flip = TRUE, y.position = c(80,80,45,90,70,45,45,90,70,45,90,75,45)) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance", limits = c(0,100), expand = c(0.025,0.025))
p.fig1

ggsave(filename = "plots/relative_abundance/SUP_-_Figure_A4_Major_phyla_abundance_by_day.png", plot = p.fig1, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/SUP_-_Figure_A4_Major_phyla_abundance_by_day.pdf", plot = p.fig1, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)


## SUPPLEMENTING FIGURE FOR PFOS EFFECT ON ABUNDANCES FOPR ALL DATA EXCLUDING DAY 0

dat.1 <- subset(dat.used, day != "d0")
df.1 <- subset(df.all, day != "d0")
dat.1$feed <- factor(dat.1$feed, levels = c("LF","HF"))
df.1$feed <- factor(df.1$feed, levels = c("LF","HF"))

stat.in_3 <- dat.used %>%
  group_by(feed,Phylum) %>%
  wilcox_test(as.formula("Abundance ~ feed_treat")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance() %>%
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.in_3

stat.out_3 <- dat.used %>%
  group_by(Phylum) %>%
  wilcox_test(as.formula("Abundance ~ feed")) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance() %>%
  add_xy_position(x = "feed", dodge = 0.8) %>%
  p_format("p.adj", accuracy = 0.001, trailing.zero = TRUE, new.col = TRUE)
stat.out_3

p.all <- ggplot(dat.1) +
  geom_bar(data = df.1, aes(fill = fct_rev(feed_treat), y = Abundance, x = feed), stat = "identity", position = position_dodge()) +
  geom_point(data = dat.1, aes(x = feed, y = Abundance, color = fct_rev(feed_treat)), position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), size = 0.5, alpha = 0.5) +
  # geom_text(data = df.all, aes(label = paste0(round(Abundance,1),"%"), x = feed, y = ifelse(Abundance<25, Abundance+15, Abundance/2), group = fct_rev(feed_treat)), position = position_dodge(width = 0.9), size = 3, hjust = 0.5) +
  facet_grid(Phylum ~ .) +
  coord_flip() +
  scale_fill_manual(values = params$COL, name = "All data (Day 8 - 21)", labels = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS"), breaks = c("HF_CTRL", "HF_PFOS","LF_CTRL","LF_PFOS"))+
  scale_color_manual(values = c("LF_CTRL" = "#1778b6","LF_PFOS" = "#1748b6","HF_CTRL" = "#84b559","HF_PFOS" = "#306148")) +
  theme_pubr(legend = "top", border = TRUE) +
  scale_x_discrete(name = "Feed") +
  theme(axis.title.y = element_blank()) +
  guides(color = "none")

p.fig2 <- p.all+ stat_pvalue_manual(stat.in_3, label = "p.adj.format", tip.length = 0, hide.ns = FALSE, color = "red", coord.flip = TRUE, y.position = c(85,60,15,80,65,10)) +
  stat_pvalue_manual(stat.out_3, label = "p.adj.signif", tip.length = 0, hide.ns = TRUE, coord.flip = TRUE, y.position = c(90,70,20)) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance", limits = c(0,100), expand = c(0.01,0.01))
p.fig2

ggsave(filename = "plots/relative_abundance/Excluding_D0_phyla_abundance_all.png", plot = p.fig2, device = "png", unit = "mm", dpi = 300, height = 100, width = 200)
ggsave(filename = "plots/relative_abundance/Excluding_D0_phyla_abundance_all.pdf", plot = p.fig2, device = "pdf", unit = "mm", dpi = 300, height = 100, width = 200)

p.final <- ggarrange(p.fig1, p.fig2, 
                     ncol = 1, nrow = 2, 
                     common.legend = TRUE, 
                     heights = c(4,3), 
                     labels = c("A","B"))
p.final

ggsave(filename = "plots/figures/final_figures_Appendix/Figure A.5.png", plot = p.final, device = "png", unit = "mm", dpi = 300, height = 220, width = 200)
ggsave(filename = "plots/figures/final_figures_Appendix/Figure A.5.pdf", plot = p.final, device = "pdf", unit = "mm", dpi = 300, height = 220, width = 200)


```




## Figure A.X: PHYLA {.tabset .tabset-dropdown}

The following section summarises relative abundances and the ratio between Firmicutes and Bacteroidetes in data daily and in respect to factors such as HF/LF feed and PFOS exposure. 

```{r}
# load data 
load("R_objects/Agglomerated.RData")
params <- readRDS("R_objects/params_description.RDS")
LIST.DAY <- c("d0","d08","d12","d16","d21", "all")
DIETS <- c("HF","LF")
LIST.TR <- c("CTRL","PFOS", "both")
LVL <- "Phylum"

# Prepare dataframe
df <- data.frame()

# Run calcualtions
suppressMessages({suppressWarnings({
for (DIET in DIETS) {
  #print(DIET)
  load("R_objects/Agglomerated.RData")
  phy.feed <- phy.ph
  phy.feed <- subset_samples(phy.feed, feed == DIET)
  
  for (DAY in LIST.DAY) {
    if (DAY == "all") {
      print("## All days selected")
      phy.day <- phy.feed
    } else {
      print(paste0("## Day ",DAY," selected"))
      phy.day <- subset_samples(phy.feed, day == DAY)
    }
    
    for (TR in LIST.TR) {
      if (TR == "both") {
        # print("## Both treatment groups selected")
        phy.tr <- phy.day
      } else {
        # print(paste0("## Treatment ",TR," selected"))
        phy.tr <- subset_samples(phy.day, treatment == TR)
      }
      # Transform data
      phy.rel <- transform_sample_counts(phy.tr, fun = function(x) x/sum(x)*100)
      dat <- psmelt(phy.rel)
      
      # Filter by abundance, then rank
      phy.top <- filter_abundance(phy.rel) %>% filter_rank(min.rank = 19)
      
      #Melt data
      dat <- suppressWarnings(psmelt(phy.top))
      
      # Prevent duplicate column names
      colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")] <- paste(colnames(dat)[colnames(dat) %in% c("Taxa","SampleID")], "old", sep = "_")
      
      # Rename relevant columns
      colnames(dat)[c(1,4)] <- c("Taxa","SampleID")
      
      # Sort taxa
      dat.sort <- sort_taxa(dat) 
      
      # rename taxa
      levels(dat.sort$Taxa) <- LETTERS[seq(length(levels(dat.sort$Taxa)))]
      
      # print tax table
      tax <- dat.sort %>% group_by(Taxa, Kingdom, Phylum) %>% summarise(Abundance = mean(Abundance))
      
      # Bacteroidetes-Firmicutes ratio
      Ba <- subset(tax, Phylum == "Bacteroidetes")
      Fi <- subset(tax, Phylum == "Firmicutes")
      Pb <- subset(tax, Phylum == "Proteobacteria")
      if (nrow(Pb) == 0) {
        print(paste0("Zero Preoteobacteria in ",DIET," ",DAY," ",TR))
        df_0 <- c("D","Bacteria","Proteobacteria",0)
        Pb <- data.frame()
        Pb <- rbind(Pb,df_0)
        colnames(Pb) <- c("Taxa","Kingdom","Phylum","Abundance")
        Pb <- transform(Pb, Abundance = as.numeric(Abundance))
        } else {}
      Ac <- subset(tax, Phylum == "Actinobacteria")
      if (nrow(Ac) == 0) {
        print(paste0("Zero Actinobacteria in ",DIET," ",DAY," ",TR))
        df_0 <- c("A","Bacteria","Actinobacteria",0)
        Ac <- data.frame()
        Ac <- rbind(Ac,df_0)
        colnames(Ac) <- c("Taxa","Kingdom","Phylum","Abundance")
        Ac <- transform(Ac, Abundance = as.numeric(Abundance))
        } else {}
      OT_FB <- 100 - (round(Fi$Abundance,3) + round(Ba$Abundance,3))
      OT_ALL <- 100 - (round(Fi$Abundance,3) + round(Ba$Abundance,3) + round(Pb$Abundance,3) + round(Ac$Abundance,3))
      FBR <- Fi$Abundance / Ba$Abundance 
      
      output <- c(DIET, DAY, TR, paste0(DIET,"_",TR), round(Fi$Abundance,3), round(Ba$Abundance,3),round(Pb$Abundance,3),round(Ac$Abundance,3), OT_FB, OT_ALL, FBR) #round(Ab$Abundance,3),
      
      df <- rbind(df, output)
      
    }
  }
}
 })})

# Names columns
colnames(df) <- c("feed","day","treatment","feed_treat",
                  "Firmicutes","Bacteroidetes","Proteobacteria","Actinobacteria","Others_FB","Others_all","FB_ratio")
df
df <- transform(df, Firmicutes = as.numeric(Firmicutes),
                Bacteroidetes = as.numeric(Bacteroidetes),
                Proteobacteria = as.numeric(Proteobacteria),
                Actinobacteria = as.numeric(Actinobacteria),
                Others_FB = as.numeric(Others_FB),
                Others_all = as.numeric(Others_all),
                FB_ratio = as.numeric(FB_ratio))
df
write.csv(df, file = "plots/relative_abundance/Phyla_relative_abundance.csv")

# Prepare Firmicutes and Bacteroidetes for plotting
df.fb <- pivot_longer(df, cols = c(Firmicutes, Bacteroidetes, Others_FB), names_to = "Phylum", values_to = "Abundance")
df.fb <- subset(df.fb, treatment != "both" & day != "all")
df.fb

# Plot Firmicutes + Bacteroidetes
p <- ggplot(df.fb, aes(fill = Phylum, y = Abundance, x = feed)) +
  geom_bar(position = "fill", stat = "identity") +
  facet_grid(treatment ~ day, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  geom_text(aes(label = paste0(round(Abundance,2),"%")), position = position_fill(vjust = 0.5), size = 3, color = "#222222") +
  ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others_FB" = "Others")) +
  scale_y_continuous(labels = function(x) paste0(x*100, "%"), name = "Relative Abundance") +
  theme_pubr(legend = "top", border = TRUE) +
  theme(axis.title.x = element_blank())

p

ggsave(filename = "plots/relative_abundance/Firmicutes-Bacteroidetes-ratios.png", plot = p, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Firmicutes-Bacteroidetes-ratios.pdf", plot = p, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# Prepare All for plotting
df.all <- pivot_longer(df, cols = c(Firmicutes, Bacteroidetes, Proteobacteria, Actinobacteria, Others_all), names_to = "Phylum", values_to = "Abundance")
df.all <- subset(df.all, treatment != "both" & day != "all")
df.all

l <- c("Firmicutes", "Bacteroidetes", "Proteobacteria", "Actinobacteria", "Others_all")

df.all$Phylum <- factor(df.all$Phylum, levels = c(l))

if (!file.exists("plots/figures/objects")) dir.create(file.path(getwd(), "plots/figures/objects"))
save(df,df.fb,df.all, file = "plots/figures/objects/phyla_means.Rdata")

# Plot All
p.all <- ggplot(df.all, aes(fill = Phylum, y = Abundance, x = feed)) +
  geom_bar(position = "fill", stat = "identity") +
  facet_grid(treatment ~ day, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  geom_text(data = df.all[df.all$Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"),], aes(label = paste0(round(Abundance,2),"%")), position = position_fill(vjust = 0.5), size = 3) +
  ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others_all" = "Others")) +
  scale_y_continuous(labels = function(x) paste0(x*100, "%"), name = "Relative Abundance") +
  theme_pubr(legend = "top", border = TRUE) +
  theme(axis.title.x = element_blank())

p.all

ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance.png", plot = p.all, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance.pdf", plot = p.all, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# Plot check PFOS
p.pfos <- ggplot(df.all, aes(fill = treatment, y = Abundance, x = day)) +
  geom_bar(position = position_dodge(), stat = "identity") +
  geom_point() +
  facet_grid(feed ~ Phylum, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  #geom_text(data = df.all[df.all$Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"),], aes(label = paste0(round(Abundance,2),"%")), position = position_fill(vjust = 0.5), size = 3) +
  ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others_all" = "Others")) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance") +
  theme_pubr(legend = "top", border = TRUE) +
  theme(axis.title.x = element_blank())

p.pfos

ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance_pfos.png", plot = p.pfos, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance_pfos.pdf", plot = p.pfos, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)


# Plot check Diet
p.feed <- ggplot(df.all, aes(fill = feed, y = Abundance, x = day)) +
  geom_bar(position = position_dodge(), stat = "identity") +
  facet_grid(treatment ~ Phylum, labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  #geom_text(data = df.all[df.all$Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"),], aes(label = paste0(round(Abundance,2),"%")), position = position_fill(vjust = 0.5), size = 3) +
  ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others_all" = "Others")) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance") +
  theme_pubr(legend = "top", border = TRUE) +
  theme(axis.title.x = element_blank()) +
  scale_x_discrete(labels = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))

p.feed

ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance_diet.png", plot = p.feed, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance_diet.pdf", plot = p.feed, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# Plot check Day
p.day <- ggplot(df.all, aes(fill = day, y = Abundance, x = feed)) +
  geom_bar(position = position_dodge(), stat = "identity") +
  facet_grid(Phylum ~ treatment, scales = "free_y", labeller = labeller(day = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))) +
  #geom_text(data = df.all[df.all$Phylum %in% c("Firmicutes", "Bacteroidetes", "Proteobacteria"),], aes(label = paste0(round(Abundance,2),"%")), position = position_fill(vjust = 0.5), size = 3) +
  ggsci::scale_fill_d3(palette = "category20", name = "Phylum", labels = c("Others_all" = "Others", "d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21")) +
  scale_y_continuous(labels = function(x) paste0(x, "%"), name = "Relative Abundance") +
  theme_pubr(legend = "top", border = TRUE) +
  theme(axis.title.x = element_blank()) +
  scale_x_discrete(labels = c("d0" = "Day 0","d08" = "Day 8","d12" = "Day 12","d16" = "Day 16","d21" = "Day 21"))

p.day

ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance_day.png", plot = p.day, device = "png", unit = "mm", dpi = 300, height = 150, width = 200)
ggsave(filename = "plots/relative_abundance/Phyla_relative_mean_abundance_day.pdf", plot = p.day, device = "pdf", unit = "mm", dpi = 300, height = 150, width = 200)

# kable(tax,digits = 2,caption = paste("Taxa plotted", sep = " ")) %>%
#   kable_classic(full_width = F, position = "left")
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())
```

# FINAL COMMENT

This completes the microbiome description.

To begin the statistical analysis of the data you are now ready for the
following steps:

| Analysis               | Template                   | Note                                                               |
|------------------|---------------------|---------------------------------|
| Statistical testing    | GMH_test_variables         | Statistical test and visualization of alpha diversity and metadata |
| Beta diversity         | GMH_beta_diveristy         | Statistical test and visualization of beta diversity               |
| Differential abundance | GMH_differential_abundance | Test differential abundance of taxa against sample variables       |

# SETTINGS {.tabset .tabset-fade .tabset-pills}

Overview of the parameters and packages that were used for this
Rmarkdown.

## PARAMETERS

The following paramenters were set in for this analysis:

```{r parameters, eval=TRUE}
params <- readRDS("R_objects/params_description.RDS")


tmp <- unlist(params)
dat <- data.frame(Parameter = names(tmp), Value = unname(tmp))


kbl(dat, row.names = F) %>% kable_classic(lightable_options = "striped")

```

## SESSION INFO

The analysis was run in the following environment:

```{r packages, eval=TRUE}
sessionInfo()
```
