---
title: "5. Beta diversity"
author: "Claus Lykkebo"
date: "`r Sys.Date()`"
output:
  rmdformats::readthedown:
    toc_depth: 4
    collapsed: false
    code_folding: hide
    number_sections: false
knit: (function(inputFile, encoding) { 
    rmarkdown::render(
        inputFile, encoding = encoding,
        output_dir = file.path(dirname(inputFile),"output"),
        output_file = paste0("Fibrex_", Sys.Date(), "_5_Beta_diversity.html"))
        })
params:
    input: "R_objects/Phyloseq.Rdata"
    metrics: "bray|jac|unif|wuf|ait"
    COL: !r c("LF_CTRL" = "#a5cee3","LF_PFOS" = "#1778b6","HF_CTRL" = "#b4d88a","HF_PFOS" = "#30a148")
    COLFEED: !r c("LF" = "#1778b6","HF" = "#30a148")

---

# INFO {.tabset .tabset-fade .tabset-pills}

This Rmarkdown contains the commands necessary to perform beta diversity analysis of the output from the [DF_GMH_PIPELINE](https://github.com/MSMortensen/DF_GMH_pipeline). It is expected that the data has been imported, cleaned, and saved following the script **1_Import.Rmd** prior to using this script.

Beta diversity, also called "between sample diversity" is calculated as a dissimilarity distance between individual samples.
Here we calculate metrices:
 -    Bray-Curtis
 -    Jaccard
 -    UniFrac
 -    Unweighted UniFrac
 -    Aitchison (not used in final presentation)

# SETUP

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(GMHmicrobiome)
library(ggpubr)
library(kableExtra)
library(phyloseq)
library(cowplot)
library(ggExtra)
library(vegan)
library(ggrepel)

# save parameters
saveRDS(params, "R_objects/params_betadiv.RDS")

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# CALCULATE DISTANCES

## PREPARE DATA

First step is to create a clean phyloseq object (remove samples missing relevant data and/or subset by a relevant variable).

Beta diversity are affected by all samples included, so if some samples are removed, this should preferably be done before the beta diversity is calculated. \> qualitative metrics are sensitive to sequencing depth, so rarefaction should be done for those metrics as well.

```{r, include=FALSE, eval=FALSE}
# Load data
params <- readRDS("R_objects/params_betadiv.RDS")

MTRLS <- c("Feces","Cecum","Ileum","Cecum_HF","Ileum_HF","Cecum_LF","Ileum_LF","Feces_no20","Feces_no020","Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")

load(params$input)

#Extract metadata
dat <- data.frame(sample_data(phy))

#Add metadata for direct PFOS exposure
for (i in dat$SampleID) {
 dat$PFOS <- ifelse(dat$day == "d0", "ctrl", ifelse(dat$treatment == "PFOS","PFOS","ctrl"))
 dat$feed_day <- paste0(dat$feed,"_",dat$day)
 }

#Merge addition to phyloseq element
dat <- sample_data(dat)
phy <- merge_phyloseq(phy, dat)

# Save phyloseq with additional metadata
save(phy, file = "R_objects/Phyloseq_beta.Rdata")

#Calculate iterations of data for analysis
for (MTRL in MTRLS) {
  load("R_objects/Phyloseq_beta.Rdata")
  message(paste0("Calcualting distances for ",MTRL,"..."))
  # Create subset
  if (MTRL == "Feces_no20") {
    phy.clean <- subset_samples(phy, material == "Feces" & day != "d20")
  } else if (MTRL == "Feces_no020") {
    phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d08","d12","d16","d21"))
  } else if (MTRL == "Feces_HF") {
      phy.clean <- subset_samples(phy, material == "Feces" & day != "d20" & feed == "HF")
  } else if (MTRL == "Feces_LF") {
      phy.clean <- subset_samples(phy, material == "Feces" & day != "d20" & feed == "LF")
  } else if (MTRL == "Feces_HF_no0") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d08","d12","d16","d21") & feed == "HF")
  } else if (MTRL == "Feces_LF_no0") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d08","d12","d16","d21") & feed == "LF")
  } else if (MTRL == "Feces_HF_d08") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d0","d08") & feed == "HF")
  } else if (MTRL == "Feces_LF_d08") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d0","d08") & feed == "LF")
  } else if (MTRL == "Feces_HF-CTRL_d08") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d0","d08") & feed_treat == "HF_CTRL")
  } else if (MTRL == "Feces_HF-PFOS_d08") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d0","d08") & feed_treat == "HF_PFOS")
  } else if (MTRL == "Feces_LF-CTRL_d08") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d0","d08") & feed_treat == "LF_CTRL")
  } else if (MTRL == "Feces_LF-PFOS_d08") {
      phy.clean <- subset_samples(phy, material == "Feces" & day %in% c("d0","d08") & feed_treat == "LF_PFOS")
  } else if (MTRL == "Cecum_HF") {
      phy.clean <- subset_samples(phy, material == "Cecum" & feed == "HF")
  } else if (MTRL == "Cecum_LF") {
      phy.clean <- subset_samples(phy, material == "Cecum" & feed == "LF")
  } else if (MTRL == "Ileum_HF") {
      phy.clean <- subset_samples(phy, material == "Ileum" & feed == "HF")
  } else if (MTRL == "Ileum_LF") {
      phy.clean <- subset_samples(phy, material == "Ileum" & feed == "LF")
  } else {
      phy.clean <- subset_samples(phy, material == MTRL)
  }
  
  # Clean data
  phy.clean <- prune_samples(sample_names(phy.clean)[!is.na(sample_data(phy.clean))], phy.clean)
  
  # Remove empty taxa
  phy.clean <- prune_taxa(taxa_sums(phy.clean) > 0, phy.clean)
  
  # Recalculate midpoint root
  phy_tree(phy.clean) <- phangorn::midpoint(phy_tree(phy.clean))
  
  # Perform multiple rarefactions
  phy.rare <- multiple_rarefy(phy.clean)
  
  # Remove empty taxa
  phy.rare <- prune_taxa(taxa_sums(phy.rare) > 0, phy.rare)
  
  # Recalculate midpoint root
  phy_tree(phy.rare) <- phangorn::midpoint(phy_tree(phy.rare))
  
  # Save object
  save(phy.rare, phy.clean, file = paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE)) #will clear all objects includes hidden objects.
invisible(gc()) #free up memory and report the memory usage.
```

## CALCULATE

There are many different algorithms that can be used to calculate beta diversity. Each metric has its own advantages and many times a comparison of the results from two metrics tells more about the data than each one separately. I will include phylogeny based (UniFrac and weighted UniFrac), qualitative (presence/absence), quantitative (abundance based), and compositional (Aitchison) metrics in this template

### UNWEIGHTED UNIFRAC

The unique fraction metric, or UniFrac, measures the phylogenetic distance between sets of taxa in a phylogenetic tree as the fraction of the branch length of the tree that leads to descendants from either one environment or the other, but not both [(Lozupone & Knight, 2005)](https://doi.org/10.1128/AEM.71.12.8228-8235.2005). This metric is sensitive to sequencing depth, so it is required to use a rarefied phyloseq object The UniFrac algorithm requires a rooted tree, so if ASVs has been removed from the raw da the tree should be rerooted manually, else a random ASV will be chosen as root.

```{r calc-unif, eval=FALSE, echo=TRUE}
MTRLS <- c("Feces","Cecum","Ileum","Cecum_HF","Ileum_HF","Cecum_LF","Ileum_LF","Feces_no20","Feces_no020","Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")
# load
params <- readRDS("R_objects/params_betadiv.RDS")
for (MTRL in MTRLS) {
  load(paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
  message(paste0("Calcualting distances for ",MTRL,"..."))

  # Calculate UniFrac distances
  unif.dist <- UniFrac(phy.rare, weighted = FALSE, parallel = FALSE)
  
  # Calculate PCoA data
  unif.pcoa <- ordinate(phy.rare, method = "PCoA",distance = unif.dist)
  unif.nmds <- metaMDS(unif.dist, k = 5, trymax = 1000)
  
  # Save distance objects
  save(unif.dist, unif.nmds, unif.pcoa, phy.rare, file = paste0("R_objects/bdiv_unif_",MTRL,".RData"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### WEIGHTED UNIFRAC

The unique fraction metric, or UniFrac, measures the phylogenetic distance between sets of taxa in a phylogenetic tree as the fraction of the branch length of the tree that leads to descendants from either one environment or the other, but not both [(Lozupone & Knight, 2005)](https://doi.org/10.1128/AEM.71.12.8228-8235.2005). Weighted UniFrac takes the abundance of each ASV into account instead of just presence/absence, which means that it will not be sensitive to sequencing depth. The UniFrac algorithm requires a rooted tree, so if ASVs has been removed from the raw da the tree should be rerooted manually, else a random ASV will be chosen as root.

```{r calc-wunif, eval=FALSE, echo=TRUE}
MTRLS <- c("Feces","Cecum","Ileum","Cecum_HF","Ileum_HF","Cecum_LF","Ileum_LF","Feces_no20","Feces_no020","Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")
# load
params <- readRDS("R_objects/params_betadiv.RDS")
for (MTRL in MTRLS) {
  load(paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
  message(paste0("Calcualting distances for ",MTRL,"..."))

  # Calculate UniFrac distances
  wuf.dist <- UniFrac(phy.clean, weighted = TRUE, parallel = FALSE)
  
  # Calculate PCoA data
  wuf.pcoa <- ordinate(phy.clean, method = "PCoA",distance = wuf.dist)
  wuf.nmds <- metaMDS(wuf.dist, k = 5, trymax = 1000)
  
  # Save distance objects
  save(wuf.dist, wuf.nmds, wuf.pcoa, phy.clean, file = paste0("R_objects/bdiv_wuf_",MTRL,".RData"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### BRAY-CURTIS

Bray-Curtis dissimilarity index (as implemented by the vegan package) is the sum of abundance difference for each species/ASV, divided by theoretical maximum difference between the samples if no ASV overlapped. The formula used is: $$d_{jk} = \frac{\sum|n_{ij}-n_{ik}|}{\sum(n_{ij}+n_{ik})}$$ Bray-Curtis dissimilarity is not a true distance metric as it does not adhere to the [triangle inequality](https://en.wikipedia.org/wiki/Triangle_inequality), but is often used to compare microbiomes. Bray-Curtis dissimilarities are based on the assumption that measurements are taken from equal areas, so differences in total counts between samples will bias the metric. As differences in sequences depth is due to differences in the lab procedures and not biological differences, we should transform our counts to relative abundances before calculating Bray-Curtis dissimilarities. By transforming the data to abundances no data is lost, but rarefied data can also be used.

```{r calc-bray, eval=FALSE, echo=TRUE}
MTRLS <- c("Feces","Cecum","Ileum","Cecum_HF","Ileum_HF","Cecum_LF","Ileum_LF","Feces_no20","Feces_no020","Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")
# load
params <- readRDS("R_objects/params_betadiv.RDS")
for (MTRL in MTRLS) {
  load(paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
  message(paste0("Calcualting distances for ",MTRL,"..."))
  
  # transform counts
  phy.ra <- transform_sample_counts(phy.clean, function(x) x/sum(x))
  
  # Calculate Bray-Curtis dissimilarities
  bray.dist <- distance(phy.ra, method = "bray",)
  
  # Calculate PCoA data
  bray.pcoa <- ordinate(phy.ra, method = "PCoA",distance = bray.dist)
  bray.nmds <- metaMDS(bray.dist, k = 5, trymax = 1000)
  
  # Save distance objects
  save(bray.dist, bray.nmds, bray.pcoa, phy.ra, file = paste0("R_objects/bdiv_bray_",MTRL,".RData"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### JACCARD

The Jaccard similarity measures the similarity between two sets of data to see which members are shared and distinct. The Jaccard similarity is calculated by dividing the number of observations in both sets by the number of observations in either set. In other words, the Jaccard similarity can be computed as the size of the intersection divided by the size of the union of two sets. This can be written in set notation using intersection $(A \cap B)$ and unions $(A \cup B)$ of two sets: $$J(A,B) = \frac{|A \cap B|}{|A \cup B|}$$ where $|A \cap B|$ gives the number of members shared between both sets and $|A \cup B|$ gives the total number of members in both sets (shared and un-shared). The Jaccard Similarity will be 0 if the two sets don't share any values and 1 if the two sets are identical.

Additionally, this function can be used to find the dissimilarity between two sets by calculating:
$$d(A,B) = 1 - J(A,B)$$
> We will be calculating binary Jaccard dissimilarity

```{r calc-jac, eval=FALSE, echo=TRUE}
MTRLS <- c("Feces","Cecum","Ileum","Cecum_HF","Ileum_HF","Cecum_LF","Ileum_LF","Feces_no20","Feces_no020","Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")
# load
params <- readRDS("R_objects/params_betadiv.RDS")
for (MTRL in MTRLS) {
  load(paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
  message(paste0("Calcualting distances for ",MTRL,"..."))
  
  # Calculate Jaccard binary dissimilarities
  jac.dist <- distance(phy.rare, method = "jaccard", binary = TRUE)
  
  # Calculate PCoA data
  jac.pcoa <- ordinate(phy.rare, method = "PCoA",distance = jac.dist)
  jac.nmds <- metaMDS(jac.dist, k = 5, trymax = 1000)
  
  # Save distance objects
  save(jac.dist, jac.nmds, jac.pcoa, phy.rare, file = paste0("R_objects/bdiv_jac_",MTRL,".RData"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### AITCHISON

Aitchison distance (1986) and robust Aitchison distance [(Martino et al. 2019)](https://journals.asm.org/doi/10.1128/mSystems.00016-19) are metrics that deal with compositional data. The Aitchison distance is a dissimilarity measure calculated as the Euclidean distance between observations (samples) after performing a centered log ratio (“clr”) transformation. Aitchison distance has been said to outperform Jensen-Shannon divergence and Bray-Curtis dissimilarity, due to a better stability to subsetting and aggregation, and it being a proper distance (Aitchison et al., 2000).

```{r calc-ait, eval=FALSE, echo=TRUE}
MTRLS <- c("Feces","Cecum","Ileum","Cecum_HF","Ileum_HF","Cecum_LF","Ileum_LF","Feces_no20","Feces_no020","Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")
# load
params <- readRDS("R_objects/params_betadiv.RDS")
for (MTRL in MTRLS) {
  load(paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
  message(paste0("Calcualting distances for ",MTRL,"..."))
  
  # Calculate Bray-Curtis dissimilarities
  ait.dist <- vegan::vegdist(otu_table(phy.clean), method = "robust.aitchison")
  
  # Calculate PCoA data
  ait.pcoa <- ordinate(phy.clean, method = "PCoA",distance = ait.dist)
  ait.nmds <- metaMDS(ait.dist, k = 5, trymax = 1000)
  
  # Save distance objects
  save(ait.dist, ait.nmds, ait.pcoa, phy.clean, file = paste0("R_objects/bdiv_ait_",MTRL,".RData"))
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

### SUBSETS PER DAY AND FEED

Running subsets for downstream analysis.
```{r subsets, eval=FALSE, echo=TRUE}
METRICES <- c("bray","unif","wuf","jac","ait")
DAYS <- c("d0","d08","d12","d16","d20","d21")
FEEDS <- c("HF","LF","ALL")
MTRLS <- c("Feces","Cecum","Ileum")

# load
params <- readRDS("R_objects/params_betadiv.RDS")
#load("R_objects/Phyloseq_betadiv.Rdata")
for (MTRL in MTRLS) {
message(paste0("Running loop for calculation of beta diversity subsets for ",MTRL,"."))
  if (MTRL == "Cecum" || MTRL == "Ileum") {
    DAYS <- c("d08","d21")
  } else if (MTRL == "Feces") {
    DAYS <- c(,"d0","d08","d12","d16","d20","d21")
  }
  for (DAY in DAYS) {
    print(paste0("Processing day ",DAY,"..."))
    for (FEED in FEEDS) {
      for (METRIC in METRICES) {
    load(paste0("R_objects/Phyloseq_betadiv_",MTRL,".Rdata"))
        if (FEED == "ALL") {
          FILENAME <- paste0("R_objects/bdiv_sub_",MTRL,"_",DAY,"_",METRIC,".Rdata")
        } else {
          FILENAME <- paste0("R_objects/bdiv_sub_",MTRL,"_",DAY,"_",FEED,"_",METRIC,".Rdata")
        }
        if (METRIC == "unif") {
          load(paste0("R_objects/bdiv_unif_",MTRL,".RData"))
          tmp.dist <- as.matrix(unif.dist)
          if (FEED == "ALL") {
            phy.sub <- subset_samples(phy.rare, material == MTRL & day %in% DAY)
          } else {
            phy.sub <- subset_samples(phy.rare, material == MTRL & day %in% DAY & feed %in% FEED)
          }
          SUBS <- sample_names(phy.sub)
          
          tmp.sub.dist <- tmp.dist[SUBS, SUBS]
          unif.sub.dist <- as.dist(tmp.sub.dist, diag = TRUE, upper = TRUE)
          
          unif.sub.pcoa <- ordinate(phy.sub, method = "PCoA",distance = unif.sub.dist)
          unif.sub.nmds <- metaMDS(unif.sub.dist, k = 5, trymax = 1000)
          
          save(phy.sub, unif.sub.dist, unif.sub.nmds, unif.sub.pcoa, file = FILENAME)
        } else if (METRIC == "wuf") {
          load(paste0("R_objects/bdiv_wuf_",MTRL,".RData"))
          tmp.dist <- as.matrix(wuf.dist)
          
          if (FEED == "ALL") {
            phy.sub <- subset_samples(phy.clean, material == MTRL & day %in% DAY)
          } else {
            phy.sub <- subset_samples(phy.clean, material == MTRL & day %in% DAY & feed %in% FEED)
          }
          SUBS <- sample_names(phy.sub)
          
          tmp.sub.dist <- tmp.dist[SUBS, SUBS]
          wuf.sub.dist <- as.dist(tmp.sub.dist, diag = TRUE, upper = TRUE)
          
          wuf.sub.pcoa <- ordinate(phy.sub, method = "PCoA",distance = wuf.sub.dist)
          wuf.sub.nmds <- metaMDS(wuf.sub.dist, k = 5, trymax = 1000)
          
          save(phy.sub, wuf.sub.dist, wuf.sub.nmds, wuf.sub.pcoa, file = FILENAME)
        } else if (METRIC == "bray") {
          load(paste0("R_objects/bdiv_bray_",MTRL,".RData"))
          tmp.dist <- as.matrix(bray.dist)
          
          if (FEED == "ALL") {
            phy.sub <- subset_samples(phy.rare, material == MTRL & day %in% DAY)
          } else {
            phy.sub <- subset_samples(phy.rare, material == MTRL & day %in% DAY & feed %in% FEED)
          }
          SUBS <- sample_names(phy.sub)
          
          tmp.sub.dist <- tmp.dist[SUBS, SUBS]
          bray.sub.dist <- as.dist(tmp.sub.dist, diag = TRUE, upper = TRUE)
          
          bray.sub.pcoa <- ordinate(phy.sub, method = "PCoA", distance = bray.sub.dist)
          bray.sub.nmds <- metaMDS(bray.sub.dist, k = 5, trymax = 1000)
          
          save(phy.sub, bray.sub.dist, bray.sub.pcoa, bray.sub.nmds, file = FILENAME)
        } else if (METRIC == "jac") {
          load(paste0("R_objects/bdiv_jac_",MTRL,".RData"))
          tmp.dist <- as.matrix(jac.dist)
          
          if (FEED == "ALL") {
            phy.sub <- subset_samples(phy.rare, material == MTRL & day %in% DAY)
          } else {
            phy.sub <- subset_samples(phy.rare, material == MTRL & day %in% DAY & feed %in% FEED)
          }
          SUBS <- sample_names(phy.sub)
          
          tmp.sub.dist <- tmp.dist[SUBS, SUBS]
          jac.sub.dist <- as.dist(tmp.sub.dist, diag = TRUE, upper = TRUE)
          
          jac.sub.pcoa <- ordinate(phy.sub, method = "PCoA", distance = jac.sub.dist)
          jac.sub.nmds <- metaMDS(jac.sub.dist, k = 5, trymax = 1000)
          
          save(phy.sub, jac.sub.dist, jac.sub.pcoa, jac.sub.nmds, file = FILENAME)
        } else if (METRIC == "ait") {
          load(paste0("R_objects/bdiv_ait_",MTRL,".RData"))
          tmp.dist <- as.matrix(ait.dist)
          
          if (FEED == "ALL") {
            phy.sub <- subset_samples(phy.clean, material == MTRL & day %in% DAY)
          } else {
            phy.sub <- subset_samples(phy.clean, material == MTRL & day %in% DAY & feed %in% FEED)
          }
          SUBS <- sample_names(phy.sub)
          
          tmp.sub.dist <- tmp.dist[SUBS, SUBS]
          ait.sub.dist <- as.dist(tmp.sub.dist, diag = TRUE, upper = TRUE)
          
          ait.sub.pcoa <- ordinate(phy.sub, method = "PCoA", distance = ait.sub.dist)
          ait.sub.nmds <- metaMDS(ait.sub.dist, k = 5, trymax = 1000)
          
          save(phy.sub, ait.sub.dist, ait.sub.pcoa, ait.sub.nmds, file = FILENAME)
        }
      }
    }
  }
}
# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())

```

# VISUALIZATION (TIMELINE)

Beta diversity is generally visualized by a NMDS (non metric multidimensional scaling) or PCoA (Principal Coordinate Analysis) plots. Both methods creates a representation of the beta diversity on few axes (3D plots should only be used for interactive on screen representations). Here we have chosen to present in PCoA. Distance separation is based on Axis 1 and 2 as these provide the clearest distances of the analyses.

## ANALYSES FOR ALL SUBSETS
The following block runs processing and presentation for all distance metrices subsets of interest along with statistical support through PERMANOVA.
These subsets include the following (Note: day 20 is excluded for all sets):

 - Feces samples
   + All samples
      + Without Day 0
   + Separated by diet (HF and LF)
      + Without Day 0
      + Separated by treatment (CTRL and PFOS)
      + Isolated Day 0 and 8
      + All days excluding Day 0 (due to significant difference in LF DAy 0 data)
 - Ileum samples
   + All samples
   + Separated by diet (HF and LF)
 - Cecum samples
   + All samples
   + Separated by diet (HF and LF)

Analyses includes ordination analysis (PCoA), PERMANOVA and test for effect from dispersion (betadisper), as well as plotting.

```{r, eval=FALSE, echo=TRUE}
params <- readRDS("R_objects/params_betadiv.RDS")

# Define Metrices 
METRICES <- c("bray","jac","unif","wuf")

# Define Subsets to analyse
MTRLS <- c("Feces_no20","Feces_no020","Feces_HF","Feces_HF_no0","Feces_LF","Feces_LF_no0","Cecum","Cecum_HF","Cecum_LF","Ileum","Ileum_HF","Ileum_LF","Feces_HF_d08","Feces_LF_d08","Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08") #not used 

# Run loop
for (MTRL in MTRLS) {
  for (METRIC in METRICES) {
    print(paste0("Processing ",MTRL," ",METRIC," data..."))
    # Choose variable
    VAR <- "feed_treat_day"
  
    # Load data based on metric
    if (METRIC == "unif") {
      load(paste0("R_objects/bdiv_unif_",MTRL,".RData"))
      dist.used <- unif.dist
      nmds.used <- unif.nmds
      pcoa.used <- unif.pcoa
      phy.used <- phy.rare
      rm(unif.dist, unif.nmds, unif.pcoa, phy.rare)
    } else if (METRIC == "wuf") {
      load(paste0("R_objects/bdiv_wuf_",MTRL,".RData"))
      dist.used <- wuf.dist
      nmds.used <- wuf.nmds
      pcoa.used <- wuf.pcoa
      phy.used <- phy.clean
      rm(wuf.dist, wuf.nmds, wuf.pcoa,phy.clean)
    } else if (METRIC == "bray"){
      load(paste0("R_objects/bdiv_bray_",MTRL,".RData"))
      dist.used <- bray.dist
      nmds.used <- bray.nmds
      pcoa.used <- bray.pcoa
      phy.used <- phy.ra
      rm(bray.dist, bray.nmds, bray.pcoa, phy.ra)
    } else if (METRIC == "jac"){
      load(paste0("R_objects/bdiv_jac_",MTRL,".RData"))
      dist.used <- jac.dist
      nmds.used <- jac.nmds
      pcoa.used <- jac.pcoa
      phy.used <- phy.rare
      rm(jac.dist, jac.nmds, jac.pcoa, phy.rare)
    } else if (METRIC == "ait"){
      load(paste0("R_objects/bdiv_ait_",MTRL,".RData"))
      dist.used <- ait.dist
      nmds.used <- ait.nmds
      pcoa.used <- ait.pcoa
      phy.used <- phy.clean
      rm(ait.dist, ait.nmds, ait.pcoa, phy.clean)
    }
  
    # Extract metadata from phyloseq
    mdat <- data.frame(sample_data(phy.used))

    # If a variable consist of numbers, but represent distinct groups remember to make it into a factor
    mdat[,VAR] <- as.factor(mdat[,VAR])
    
    COL_DFT <- c("0_HF_CTRL" = "#e4fadd","8_HF_CTRL" = "#d2e9cc", "12_HF_CTRL" = "#9fd3a3", "16_HF_CTRL" = "#65bf76", "21_HF_CTRL" = "#32a248",
               "0_HF_PFOS" = "#d2e9cc","8_HF_PFOS" = "#9fd3a3", "12_HF_PFOS" = "#65bf76", "16_HF_PFOS" = "#32a248", "21_HF_PFOS" = "#108026",
               "0_LF_CTRL" = "#def8fd",	"8_LF_CTRL" = "#bde7fb", "12_LF_CTRL" = "#86c1e5", "16_LF_CTRL" = "#509cce", "21_LF_CTRL" = "#1879b7",
               "0_LF_PFOS" = "#bde7fb",	"8_LF_PFOS" = "#86c1e5", "12_LF_PFOS" = "#509cce", "16_LF_PFOS" = "#1879b7", "21_LF_PFOS" = "#065795")
    
    COL_DF <- c("HF_d0" = "#d2e9cc","HF_d08" = "#9fd3a3", "HF_d12" = "#65bf76", "HF_d16" = "#32a248", "HF_d21" = "#108026",
               "LF_d0" = "#bde7fb",	"LF_d08" = "#86c1e5", "LF_d12" = "#509cce", "LF_d16" = "#1879b7", "LF_d21" = "#065795")

    # Create plots of eigenvalues for PCoA plots
    pcoa.tab <- plot_ordination(phy.used, pcoa.used,axes = 1:5,justDF = TRUE)
    nmds.tab <- plot_ordination(phy.used, nmds.used,axes = 1:5,justDF = TRUE)
    
    # Reformat tables to create one common table
    colnames(nmds.tab)[1:5] <- c("Axis.1","Axis.2","Axis.3","Axis.4","Axis.5")
    
    nmds.tab$ordination <- "nmds"
    pcoa.tab$ordination <- "pcoa"
    
    # Calculate centroids in PCOA
    centroid.1 <- pcoa.tab %>% group_by(across(all_of("feed_treat_day"))) %>% get_summary_stats("Axis.1", type = "mean")
    centroid.1 <- subset(centroid.1, select = -c(variable,n))
    colnames(centroid.1)[1:2] <- c("feed_treat_day","X")
    centroid.2 <- pcoa.tab %>% group_by(across(all_of("feed_treat_day"))) %>% get_summary_stats("Axis.2", type = "mean")
    centroid.2 <- subset(centroid.2, select = -c(feed_treat_day,variable,n))
    colnames(centroid.2)[1] <- "Y"
    centroid <- cbind(centroid.1,centroid.2)

    centroid <- centroid %>% mutate("day" = case_when(grepl("0_",feed_treat_day) ~ "d0",
                                                    grepl("8_",feed_treat_day) ~ "d08",
                                                    grepl("12_",feed_treat_day) ~ "d12",
                                                    grepl("16_",feed_treat_day) ~ "d16",
                                                    grepl("21_",feed_treat_day) ~ "d21"),
                                  "label" = case_when(grepl("0_",feed_treat_day) ~ "Day 0",
                                                      grepl("8_",feed_treat_day) ~ "Day 8",
                                                      grepl("12_",feed_treat_day) ~ "Day 12",
                                                      grepl("16_",feed_treat_day) ~ "Day 16",
                                                      grepl("21_",feed_treat_day) ~ "Day 21",))
    centroid <- centroid %>% mutate("feed_treat" = case_when(grepl("HF_CTRL",feed_treat_day) ~ "HF_CTRL",
                                                       grepl("HF_PFOS",feed_treat_day) ~ "HF_PFOS",
                                                       grepl("LF_CTRL",feed_treat_day) ~ "LF_CTRL",
                                                       grepl("LF_PFOS",feed_treat_day) ~ "LF_PFOS"))
    centroid <- centroid %>% mutate("treat" = case_when(grepl("_CTRL",feed_treat_day) ~ "CTRL",
                                                       grepl("_PFOS",feed_treat_day) ~ "PFOS"))
    centroid <- centroid %>% mutate("feed" = case_when(grepl("HF_",feed_treat_day) ~ "HF",
                                                       grepl("LF_",feed_treat_day) ~ "LF"))
    centroid <- centroid %>% mutate("coord" = case_when(day == "d0" ~ "x1",
                                                      day == "d08" ~ "x2",
                                                      day == "d12" ~ "x3",
                                                      day == "d16" ~ "x4",
                                                      day == "d21" ~ "x5"))

    centroid <- centroid[order(centroid$day),]

    ord.tab <- rbind(nmds.tab,pcoa.tab)
    ord.tab[,VAR] <- as.factor(ord.tab[,VAR])
  
    # Melt axis to be in one variable
    axis.tab <- pivot_longer(data = ord.tab, cols = c("Axis.1","Axis.2","Axis.3","Axis.4","Axis.5"), names_to = "Axis", values_to = "position")
  
    # Plot PCoA - Split by Grouping
    pcoa.plot <- ggplot(data = ord.tab[ord.tab$ordination == "pcoa",], aes_string(x = "Axis.1", y = "Axis.2", color = "feed_day")) +
      stat_ellipse(aes_string(x = "Axis.1", y = "Axis.2", color = "feed_day", fill = "feed_day"), geom = "polygon", alpha = 0.1, level = 0.9) +
    geom_point() +
    geom_path(data = centroid, aes_string(x = "X", y = "Y", color = "feed_day"), color = "#222222", size = 0.5, alpha = 0.8, arrow = arrow(type = "closed", length = unit(2.5,"mm"))) + 
    geom_point(data = centroid, aes_string(x = "X", y = "Y", color = "feed_day"), color = "#555555", shape = 20, size = 3, alpha = 1) +
    geom_text_repel(data = centroid, mapping = aes(x = X, y = Y), label = centroid$label, size = 4, min.segment.length = 0, segment.alpha = 0.8, box.padding = 1, force = 1, color = "#222222", position = "identity", direction = "both") +
    theme_pubr() +
    scale_color_manual(values = COL_DF) +
    scale_fill_manual(values = COL_DF, name = "Feed + Day", 
                      labels = c("HF_d0" = "HF 0","HF_d08" = "HF 8", "HF_d12" = "HF 12", "HF_d16" = "HF 16", "HF_d21" = "HF 21",
                                 "LF_d0" = "LF 0",	"LF_d08" = "LF 8", "LF_d12" = "LF 12", "LF_d16" = "LF 16", "LF_d21" = "LF 21"),
                      breaks = c("HF_d0","LF_d0","HF_d08","LF_d08","HF_d12","LF_d12","HF_d16","LF_d16","HF_d21","LF_d21")) +
    facet_wrap(.~feed_treat, labeller = labeller(feed_treat = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS"))) +
    scale_y_continuous() +
    labs(x = "PCoA 1", y = "PCoA 2") +
    guides(color = "none", fill = guide_legend(override.aes = list(alpha = 1, byrow = TRUE, direction = "horizontal")))
    pcoa.plot
    
    if (MTRL %in% c("Feces_no20","Feces_no020","Cecum","Ileum")) {
      # Plot PCoA - Split by Treatment
      pcoa.treat <- ggplot(data = ord.tab[ord.tab$ordination == "pcoa",], aes_string(x = "Axis.1", y = "Axis.2")) +
      stat_ellipse(aes_string(x = "Axis.1", y = "Axis.2", color = "feed_treat", fill = "feed_day"), geom = "polygon", alpha = 0.1, level = 0.9) +
      geom_point(aes_string(color = "feed_treat", shape = "day")) +
      theme_pubr() +
      scale_color_manual(values = params$COL, name = "Group", 
                        labels = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS")) +
      scale_fill_manual(values = COL_DF, name = "Feed + Day", 
                        labels = c("HF_d0" = "HF 0","HF_d08" = "HF 8", "HF_d12" = "HF 12", "HF_d16" = "HF 16", "HF_d21" = "HF 21",
                                   "LF_d0" = "LF 0",	"LF_d08" = "LF 8", "LF_d12" = "LF 12", "LF_d16" = "LF 16", "LF_d21" = "LF 21"),
                        breaks = c("HF_d0","LF_d0","HF_d08","LF_d08","HF_d12","LF_d12","HF_d16","LF_d16","HF_d21","LF_d21")) +
      scale_shape_manual(values = c("d0" = 3, "d08" = 19, "d12" = 1, "d16" = 17, "d21" = 2), name = "Day", labels = c("d0" = "0", "d08" = "8", "d12" = "12", "d16" = "16", "d21" = "21")) +
      facet_wrap(.~treatment) +
      scale_y_continuous() +
      labs(x = "PCoA 1", y = "PCoA 2") +
      guides(fill = "none", shape = guide_legend(override.aes = list(size = 3), order = 2), color = guide_legend(override.aes = list(shape = NA, fill = params$COL, alpha = 1, linetype = c(0,0)), order = 1))#guide_legend(override.aes = list(alpha = 1, byrow = TRUE, direction = "horizontal")))
      pcoa.treat
      
      # Plot PCoA - Split by Diet
      pcoa.diet <- ggplot(data = ord.tab[ord.tab$ordination == "pcoa",], aes_string(x = "Axis.1", y = "Axis.2")) +
      stat_ellipse(aes_string(x = "Axis.1", y = "Axis.2", color = "feed_treat", fill = "feed_treat"), geom = "polygon", alpha = 0.1, level = 0.9) +
      geom_point(aes_string(color = "feed_treat", shape = "day")) +
      theme_pubr() +
      scale_color_manual(values = params$COL, name = "Group", 
                        labels = c("LF_CTRL" = "LF-CTRL","LF_PFOS" = "LF-PFOS","HF_CTRL" = "HF-CTRL","HF_PFOS" = "HF-PFOS")) +
      scale_fill_manual(values = params$COL, name = "Group") +
      scale_shape_manual(values = c("d0" = 3, "d08" = 19, "d12" = 1, "d16" = 17, "d21" = 2), name = "Day", labels = c("d0" = "0", "d08" = "8", "d12" = "12", "d16" = "16", "d21" = "21")) +
      facet_wrap(.~feed, scales = "free") +
      scale_y_continuous() +
      labs(x = "PCoA 1", y = "PCoA 2") +
      guides(fill = "none", shape = guide_legend(override.aes = list(size = 3), order = 2), color = guide_legend(override.aes = list(shape = NA, fill = params$COL, alpha = 1, linetype = c(0,0)), order = 1))#guide_legend(override.aes = list(alpha = 1, byrow = TRUE, direction = "horizontal")))
      pcoa.diet
      
      # Rename
      if (METRIC == "bray") {
        bray.treat <- pcoa.treat
        bray.diet <- pcoa.diet
      } else if (METRIC == "jac") {
        jac.treat <- pcoa.treat
        jac.diet <- pcoa.diet
      } else if (METRIC == "unif") {
        unif.treat <- pcoa.treat
        unif.diet <- pcoa.diet
      } else if (METRIC == "wuf") {
        wuf.treat <- pcoa.treat
        wuf.diet <- pcoa.diet
      } else if (METRIC == "ait") {
        ait.treat <- pcoa.treat
        ait.diet <- pcoa.diet
      }
  
      # Save extra plots
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_treat_",MTRL,"_",METRIC,".png"), plot = pcoa.treat, device = "png", dpi = 300, units = "mm", height = 110, width = 200))
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_treat_",MTRL,"_",METRIC,".pdf"), plot = pcoa.treat, device = "pdf", dpi = 300, units = "mm", height = 110, width = 200))
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_diet_",MTRL,"_",METRIC,".png"), plot = pcoa.diet, device = "png", dpi = 300, units = "mm", height = 110, width = 200))
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_diet_",MTRL,"_",METRIC,".pdf"), plot = pcoa.diet, device = "pdf", dpi = 300, units = "mm", height = 110, width = 200))
      
    } else {print("No extra plots...")}

     
    # Calculate PERMANOVA
    ## Note: use of column variable "PFOS" rather than "treatment", as this variable reflects direct PFOS exposure on all days and not just treatment grouping - this is important to detect impact of PFOS longitudinally as HF-PFOS and LF-PFOS has not been exposed on day 0.
    if (MTRL %in% c("Feces_HF","Feces_LF","Feces_HF_no0","Feces_LF_no0","Feces_HF_d08","Feces_LF_d08","Cecum_HF","Cecum_LF","Ileum_HF","Ileum_LF")) {
      perm <- adonis2(dist.used ~ day*PFOS, data = mdat, permutations = 999, na.action = na.omit)
    } else if (MTRL %in% c("Feces_HF-CTRL_d08","Feces_LF-CTRL_d08","Feces_HF-PFOS_d08","Feces_LF-PFOS_d08")) {
      perm <- adonis2(dist.used ~ day, data = mdat, permutations = 999, na.action = na.omit)
    } else if (MTRL == "Feces_no020"){
      perm <- adonis2(dist.used ~ feed*PFOS, data = mdat, permutations = 999, na.action = na.omit)
    } else {
      perm <- adonis2(dist.used ~ day*feed*PFOS, data = mdat, permutations = 999, na.action = na.omit)
    }
    # DISPERTION
    ## First we will test the beta diversity dispertion to determine whether any differences in dispertion might cause any PERMANOVA differences.
    
    # Calculate beta-dispertion
    bdisp <- betadisper(dist.used, mdat[,VAR])
    plot(bdisp)
    # Run statical test
    aov <- anova(bdisp) # not significant
    
    # Run posthoc test if significant and more than two groups
    thsd <- TukeyHSD(bdisp)
    plot(TukeyHSD(bdisp))
    
    # Plot dispertion
    boxplot(bdisp)
    
    ## When plotting the TukeyHSD, the y-axis is terribly formatted, but ordered similarly to the written output.
    
    # Print output for loop
    print(plot(bdisp))
    print(aov)
    print(pcoa.plot)
    print(paste0("PERMANOVA RESULTS FOR: ",METRIC))
    print(perm)
    print("")
    
    if (METRIC == "bray") {
      bray.plot <- pcoa.plot
      bray.perm <- perm
    } else if (METRIC == "jac") {
      jac.plot <- pcoa.plot
      jac.perm <- perm
    } else if (METRIC == "unif") {
      unif.plot <- pcoa.plot
      unif.perm <- perm
    } else if (METRIC == "wuf") {
      wuf.plot <- pcoa.plot
      wuf.perm <- perm
    } else if (METRIC == "ait") {
      ait.plot <- pcoa.plot
      ait.perm <- perm
    }
    
    # Save PERMANOVA as txt file
    if (!file.exists(paste0("plots/bdiv/timeline/",MTRL))) dir.create(file.path(getwd(), paste0("plots/bdiv/timeline/",MTRL)))
    write.table(perm, file = paste0("plots/bdiv/timeline/",MTRL,"/perm_timeline_",MTRL,"_",METRIC,".txt"))
    write.table(aov, file = paste0("plots/bdiv/timeline/",MTRL,"/aov_timeline_",MTRL,"_",METRIC,".txt"))

    # Save plot
    if (MTRL %in% c("Feces_no20","Cecum","Ileum")) {
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_timeline_",MTRL,"_",METRIC,".png"), plot = pcoa.plot, device = "png", dpi = 300, units = "mm", height = 200, width = 200))
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_timeline_",MTRL,"_",METRIC,".pdf"), plot = pcoa.plot, device = "pdf", dpi = 300, units = "mm", height = 200, width = 200))
    } else {
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_timeline_",MTRL,"_",METRIC,".png"), plot = pcoa.plot, device = "png", dpi = 300, units = "mm", height = 170, width = 150))
      suppressMessages(ggsave(filename = paste0("plots/bdiv/timeline/",MTRL,"/bdiv_timeline_",MTRL,"_",METRIC,".pdf"), plot = pcoa.plot, device = "pdf", dpi = 300, units = "mm", height = 170, width = 150))
    }
  }
  if (MTRL %in% c("Feces_no20","Feces_no020","Cecum","Ileum")) {
  save(bray.perm,bray.plot,jac.perm,jac.plot,unif.perm,unif.plot,wuf.perm,wuf.plot,bray.treat,bray.diet,jac.treat,jac.diet,unif.treat,unif.diet,wuf.treat,wuf.diet, file = paste0("plots/bdiv/timeline/",MTRL,"/PCoA_plot_",MTRL,".Rdata"))
  } else {
  save(bray.perm,bray.plot,jac.perm,jac.plot,unif.perm,unif.plot,wuf.perm,wuf.plot, file = paste0("plots/bdiv/timeline/",MTRL,"/PCoA_plot_",MTRL,".Rdata"))
  }
}

# clear the environment and release memory
rm(list = ls(all.names = TRUE))
invisible(gc())


```

# SPECIAL FIGURES (Combined Alpha- and Beta-diversity) {.tabset .tabset-dropdown}
The following contained code and content for creating Figures for the main article and Appendix A.

## Figure 6: Faecal Alpha & Beta Diversity

Figure 6 for the main article. Presents alpha- and beta-diversity development over time.

```{r, eval=TRUE, echo=TRUE}
params <- readRDS("R_objects/Adiv_params.RDS")
load("scripts/adiv.Rdata")

# Load rdata files with scfa plots
div.plots <- c("plots/adiv/timeline/adiv_figures.Rdata","plots/bdiv/timeline/Feces_HF_no0/PCoA_plot_Feces_HF_no0.Rdata")
lapply(div.plots, load,.GlobalEnv)
# rm_legend <- function(p){p + theme(legend.position = "none")}

#Renaming bdiv HF plots
hf.bray.perm <- bray.perm
hf.bray.plot <- rm_legend(bray.plot)
hf.jac.perm <- jac.perm
hf.jac.plot <- rm_legend(jac.plot)
hf.unif.perm <- unif.perm
hf.unif.plot <- rm_legend(unif.plot)
hf.wuf.perm <- wuf.perm
hf.wuf.plot <- rm_legend(wuf.plot)

# Load bdiv LF plots
load("plots/bdiv/timeline/Feces_LF_no0/PCoA_plot_Feces_LF_no0.Rdata")

#Renaming bdiv HF plots
lf.bray.perm <- bray.perm
lf.bray.plot <- rm_legend(bray.plot)
lf.jac.perm <- jac.perm
lf.jac.plot <- rm_legend(jac.plot)
lf.unif.perm <- unif.perm
lf.unif.plot <- rm_legend(unif.plot)
lf.wuf.perm <- wuf.perm
lf.wuf.plot <- rm_legend(wuf.plot)

load("plots/bdiv/timeline/Feces_no20/PCoA_plot_Feces_no20.Rdata")

legend <- get_legend(bray.plot)

#remove leftovers
rm(bray.perm,bray.plot,jac.perm,jac.plot,unif.perm,unif.plot,wuf.perm,wuf.plot)


p.adiv <- ggarrange(p.obs,p.sha,p.fpd, 
                    nrow = 1, 
                    labels = c("A","",""),
                    common.legend = TRUE,
                    legend = "top",
                    font.label = list(size = 24, face = "bold"))
p.adiv

p.bdiv <- ggarrange(hf.bray.plot, hf.jac.plot, lf.bray.plot, lf.jac.plot,
                   ncol = 2, nrow = 2,
                   align = "hv",
                   label.x = 0,
                   font.label = list(size = 24, face = "bold"),
                   labels = c("B","C","D","E"))
p.bdiv

p.all <- ggarrange(p.adiv, legend, p.bdiv,
                   ncol = 1, nrow = 3,
                   heights = c(3,0.5,4))
p.all

# Save combined graphics
ggsave(filename = "plots/figures/Figure 6 - diversity_combined_bray_jac.png", p.all, device = "png", dpi = 300, height = 350, width = 350, units = "mm")
ggsave(filename = "plots/figures/Figure 6 - diversity_combined_bray_jac.pdf", p.all, device = "pdf", dpi = 300, height = 350, width = 350, units = "mm")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)[ls(all.names = TRUE) != "params"])
invisible(gc())
```

## Figure A.1: Day 0

Figure A.1 (Appendix). Presents alpha- and beta-diversity for day 0 samples.

```{r, eval=TRUE, echo=TRUE}
# Load and process Day 0 beta-diversity for both diets
params <- readRDS("R_objects/params_betadiv.RDS")
#load("R_objects/bdiv_bray_Feces.RData")
# Choose metric
METRICES <- c("bray","jac","unif","wuf") #not used for illustration: "unif", "wuf"

# Choose variable 
VAR <- "feed_treat"
MTRL <- "Feces"
DAY <- "d0"
FEEDS <- c("HF","LF")

# Load data
for (FEED in FEEDS) {
  for (METRIC in METRICES) {
    print(paste0("Running ",FEED," and ",METRIC))
    BFILE <- paste0("R_objects/bdiv_sub_",MTRL,"_",DAY,"_",FEED,"_",METRIC,".Rdata")
    if (METRIC == "unif") {
      load(BFILE)
      dist.used <- unif.sub.dist
      nmds.used <- unif.sub.nmds
      pcoa.used <- unif.sub.pcoa
      phy.used <- phy.sub
      rm(unif.sub.dist, unif.sub.nmds, unif.sub.pcoa, phy.sub)
    } else if (METRIC == "wuf") {
      load(BFILE)
      dist.used <- wuf.sub.dist
      nmds.used <- wuf.sub.nmds
      pcoa.used <- wuf.sub.pcoa
      phy.used <- phy.sub
      rm(wuf.sub.dist, wuf.sub.nmds, wuf.sub.pcoa,phy.sub)
    } else if (METRIC == "bray"){
      load(BFILE)
      dist.used <- bray.sub.dist
      nmds.used <- bray.sub.nmds
      pcoa.used <- bray.sub.pcoa
      phy.used <- phy.sub
      rm(bray.sub.dist, bray.sub.nmds, bray.sub.pcoa, phy.sub)
    } else if (METRIC == "jac"){
      load(BFILE)
      dist.used <- jac.sub.dist
      nmds.used <- jac.sub.nmds
      pcoa.used <- jac.sub.pcoa
      phy.used <- phy.sub
      rm(jac.sub.dist, jac.sub.nmds, jac.sub.pcoa, phy.sub)
    }
    
    # Extract metadata from phyloseq
    mdat <- data.frame(sample_data(phy.used))
    
    # If a variable consist of numbers, but represent distinct groups remember to make it into a factor
    mdat[,VAR] <- as.factor(mdat[,VAR])
    
    # Create plots of eigenvalues for PCoA plots
    pcoa.tab <- plot_ordination(phy.used, pcoa.used,axes = 1:5,justDF = TRUE)
    nmds.tab <- plot_ordination(phy.used, nmds.used,axes = 1:5,justDF = TRUE)
    
    # Reformat tables to create one common table
    colnames(nmds.tab)[1:5] <- c("Axis.1","Axis.2","Axis.3","Axis.4","Axis.5")
    
    nmds.tab$ordination <- "nmds"
    pcoa.tab$ordination <- "pcoa"
    
    ord.tab <- rbind(nmds.tab,pcoa.tab)
    ord.tab[,VAR] <- as.factor(ord.tab[,VAR])
    
    # Melt axis to be in one variable
    axis.tab <- pivot_longer(data = ord.tab, cols = c("Axis.1","Axis.2","Axis.3","Axis.4","Axis.5"), names_to = "Axis", values_to = "position")
    
    # Calculate centroids PCOA
    centroid.1 <- pcoa.tab %>% group_by(across(all_of("feed_treat_day"))) %>% get_summary_stats("Axis.1", type = "mean")
    centroid.1 <- subset(centroid.1, select = -c(variable,n))
    colnames(centroid.1)[1:2] <- c("feed_treat_day","X")
    centroid.2 <- pcoa.tab %>% group_by(across(all_of("feed_treat_day"))) %>% get_summary_stats("Axis.2", type = "mean")
    centroid.2 <- subset(centroid.2, select = -c(feed_treat_day,variable,n))
    colnames(centroid.2)[1] <- "Y"
    centroid <- cbind(centroid.1,centroid.2)
  
    centroid <- centroid %>% mutate("day" = case_when(grepl("0_",feed_treat_day) ~ "d0",
                                                    grepl("8_",feed_treat_day) ~ "d08",
                                                    grepl("12_",feed_treat_day) ~ "d12",
                                                    grepl("16_",feed_treat_day) ~ "d16",# grepl("20_",feed_treat_day) ~ "d20",
                                                    grepl("21_",feed_treat_day) ~ "d21"),
                                  "label" = case_when(grepl("0_",feed_treat_day) ~ "Day 0",
                                                      grepl("8_",feed_treat_day) ~ "Day 8",
                                                      grepl("12_",feed_treat_day) ~ "Day 12",
                                                      grepl("16_",feed_treat_day) ~ "Day 16",
                                                      grepl("21_",feed_treat_day) ~ "Day 21",))
    centroid <- centroid %>% mutate("feed_treat" = case_when(grepl("HF_CTRL",feed_treat_day) ~ "HF_CTRL",
                                                       grepl("HF_PFOS",feed_treat_day) ~ "HF_PFOS",
                                                       grepl("LF_CTRL",feed_treat_day) ~ "LF_CTRL",
                                                       grepl("LF_PFOS",feed_treat_day) ~ "LF_PFOS"))
    centroid <- centroid %>% mutate("coord" = case_when(day == "d0" ~ "x1",
                                                      day == "d08" ~ "x2",
                                                      day == "d12" ~ "x3",
                                                      day == "d16" ~ "x4",
                                                      # day == "d20" ~ "x5",
                                                      day == "d21" ~ "x5"))
  
    # centroid <- centroid[order(centroid$day),]
  
    
    
    # Plot PCoA
    pcoa.plot <- ggplot(data = ord.tab[ord.tab$ordination == "pcoa",], aes_string(x = "Axis.1", y = "Axis.2", color = "feed_treat")) +
      stat_ellipse(aes_string(x = "Axis.1", y = "Axis.2", color = "feed_treat", fill = "feed_treat"), geom = "polygon", alpha = 0.1, level = 0.9) +
    geom_point() +
    geom_point(data = centroid, aes_string(x = "X", y = "Y", color = "feed_treat"), shape = 3, size = 4, alpha = 1) +
    theme_pubr() +
    scale_color_manual(values = params$COL) +
    scale_fill_manual(values = params$COL, name = "Feed + Day", 
                      labels = c("HF_d0" = "HF 0","HF_d08" = "HF 8", "HF_d12" = "HF 12", "HF_d16" = "HF 16", "HF_d21" = "HF 21",
                                 "LF_d0" = "LF 0",	"LF_d08" = "LF 8", "LF_d12" = "LF 12", "LF_d16" = "LF 16", "LF_d21" = "LF 21"),
                      breaks = c("HF_d0","LF_d0","HF_d08","LF_d08","HF_d12","LF_d12","HF_d16","LF_d16","HF_d21","LF_d21")) +
    facet_wrap(.~feed, labeller = labeller(feed = c("LF" = "LF","HF" = "HF"))) +
    scale_y_continuous() +
    labs(x = "PCoA 1", y = "PCoA 2") +
    guides(color = "none", fill = guide_legend(override.aes = list(alpha = 1, byrow = TRUE, direction = "horizontal")))
  print(pcoa.plot)
  
    # Calculate betadispertion
    bdisp <- betadisper(dist.used, mdat[,VAR])
    print(bdisp)
    # Run statical test
    ANOVA <- anova(bdisp)

        # Run posthoc test if significant and more than two groups
    TukeyHSD(bdisp)
    plot(TukeyHSD(bdisp))
    
    # Plot dispertion
    boxplot(bdisp)
    
    # Perform test
    PERM <- adonis2(dist.used ~ treatment, data = mdat, strata = mdat$day, permutations = 999, na.action = na.omit)
    print(PERM)
    
      # Save plot
    suppressMessages(ggsave(filename = paste0("plots/bdiv/Feces_d0/bdiv_timeline_",FEED,"_",METRIC,".png"), plot = pcoa.plot, device = "png", dpi = 300, units = "mm", height = 200, width = 200))
    suppressMessages(ggsave(filename = paste0("plots/bdiv/Feces_d0/bdiv_timeline_",FEED,"_",METRIC,".pdf"), plot = pcoa.plot, device = "pdf", dpi = 300, units = "mm", height = 200, width = 200))
    
    if (FEED == "HF") {
      if (METRIC == "bray") {
        hf.bray.plot <- pcoa.plot
        hf.bray.perm <- PERM
      } else if (METRIC == "jac") {
        hf.jac.plot <- pcoa.plot
        hf.jac.perm <- PERM
      } else if (METRIC == "unif") {
        hf.unif.plot <- pcoa.plot
        hf.unif.perm <- PERM
      } else if (METRIC == "wuf") {
        hf.wuf.plot <- pcoa.plot
        hf.wuf.perm <- PERM
      }
    } else if (FEED == "LF") {
      if (METRIC == "bray") {
        lf.bray.plot <- pcoa.plot
        lf.bray.perm <- PERM
      } else if (METRIC == "jac") {
        lf.jac.plot <- pcoa.plot
        lf.jac.perm <- PERM
      } else if (METRIC == "unif") {
        lf.unif.plot <- pcoa.plot
        lf.unif.perm <- PERM
      } else if (METRIC == "wuf") {
        lf.wuf.plot <- pcoa.plot
        lf.wuf.perm <- PERM
      }
    }
  }
}

# Save R object
save(hf.bray.plot,hf.bray.perm,hf.jac.plot,hf.jac.perm,hf.unif.plot,hf.unif.perm,hf.wuf.plot,hf.wuf.perm,lf.bray.plot,lf.bray.perm,lf.jac.plot,lf.jac.perm,lf.unif.plot,lf.unif.perm,lf.wuf.plot,lf.wuf.perm, file = paste0("plots/bdiv/Feces_d0/PCoA_plot_feces_day_0.Rdata"))

rm(pcoa.plot,PERM,centroid,centroid.1,centroid.2,bdisp,ANOVA,axis.tab,nmds.tab,nmds.used,phy.used)

# Load alpha-diversity for Day 0
load("scripts/adiv.Rdata")
load("plots/adiv/alpha_d0/adiv_Feces_d0.Rdata")

#Renaming bdiv HF plots
hf.bray.plot <- rm_legend(hf.bray.plot)
hf.jac.plot <- rm_legend(hf.jac.plot)
hf.unif.plot <- rm_legend(hf.unif.plot)
hf.wuf.plot <- rm_legend(hf.wuf.plot)

#Renaming bdiv HF plots
lf.bray.plot <- rm_legend(lf.bray.plot)
lf.jac.plot <- rm_legend(lf.jac.plot)
lf.unif.plot <- rm_legend(lf.unif.plot)
lf.wuf.plot <- rm_legend(lf.wuf.plot)

#remove leftovers
rm(bray.perm,bray.plot,jac.perm,jac.plot,unif.perm,unif.plot,wuf.perm,wuf.plot)

# Call plots
p.adiv <- ggarrange(p.obs,p.sha,p.fpd, 
                    nrow = 1, ncol = 3,
                    labels = c("A","",""),
                    common.legend = TRUE,
                    align = "hv",
                    legend = "top",
                    font.label = list(size = 24, face = "bold"))
p.adiv

p.bdiv <- ggarrange(hf.bray.plot, hf.jac.plot, hf.unif.plot, lf.bray.plot, lf.jac.plot, lf.unif.plot,
                   ncol = 3, nrow = 2,
                   align = "hv",
                   label.x = 0,
                   font.label = list(size = 24, face = "bold"),
                   labels = c("B","C","D","E","F","G"))
p.bdiv

p.all <- ggarrange(p.adiv, p.bdiv,
                   ncol = 1, nrow = 2,
                   heights = c(2,4))
p.all


p.all2 <- ggarrange(p.obs,p.sha,p.fpd,hf.bray.plot, hf.jac.plot, hf.unif.plot, lf.bray.plot, lf.jac.plot, lf.unif.plot,
                    ncol = 3, nrow = 3,
                    align = "hv",
                    common.legend = TRUE,
                    legend = "top",
                    font.label = list(size = 24, face = "bold"),
                    labels = c("A","","","B","","","C","",""),
                    heights = c(2,3,3))
p.all2

# Save combined graphics
ggsave(filename = "plots/figures/SUP_Figure A1 - diversity_feces_day_0.png", p.all2, device = "png", dpi = 300, height = 250, width = 300, units = "mm")
ggsave(filename = "plots/figures/SUP_Figure A1 - diversity_feces_day_0.pdf", p.all2, device = "pdf", dpi = 300, height = 250, width = 300, units = "mm")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)[ls(all.names = TRUE) != "params"])
invisible(gc())

```

## Figure A.2: ILEUM DIVERSITY

Figure A.2 (Appendix). Presents alpha- and beta-diversity for day 8 and 21 for Ileum samples.

```{r, eval=TRUE, echo=TRUE}
load("scripts/adiv.Rdata")

load("plots/adiv/alpha_d08/adiv_Ileum_d08.Rdata")
# renaming
p.eve.8 <- p.eve
p.obs.8 <- p.obs
p.sha.8 <- p.sha
p.fpd.8 <- p.fpd
load("plots/adiv/alpha_d21/adiv_Ileum_d21.Rdata")
p.eve.21 <- p.eve
p.obs.21 <- p.obs
p.sha.21 <- p.sha
p.fpd.21 <- p.fpd
rm(p.eve,p.sha,p.obs,p.fpd)

load("plots/bdiv/timeline/Ileum_HF/PCoA_plot_Ileum_HF.Rdata")
#Renaming bdiv HF plots
hf.bray.perm <- bray.perm
hf.bray.plot <- rm_legend(bray.plot)
hf.jac.perm <- jac.perm
hf.jac.plot <- rm_legend(jac.plot)
hf.unif.perm <- unif.perm
hf.unif.plot <- rm_legend(unif.plot)
hf.wuf.perm <- wuf.perm
hf.wuf.plot <- rm_legend(wuf.plot)

# Load bdiv LF plots
load("plots/bdiv/timeline/Ileum_LF/PCoA_plot_Ileum_LF.Rdata")

#Renaming bdiv HF plots
lf.bray.perm <- bray.perm
lf.bray.plot <- rm_legend(bray.plot)
lf.jac.perm <- jac.perm
lf.jac.plot <- rm_legend(jac.plot)
lf.unif.perm <- unif.perm
lf.unif.plot <- rm_legend(unif.plot)
lf.wuf.perm <- wuf.perm
lf.wuf.plot <- rm_legend(wuf.plot)

load("plots/bdiv/timeline/Ileum/PCoA_plot_Ileum.Rdata")

legend <- get_legend(bray.plot)

#remove leftovers
rm(bray.perm,bray.plot,jac.perm,jac.plot,unif.perm,unif.plot,wuf.perm,wuf.plot)

# Call plots
p.adiv <- ggarrange(p.obs.8,p.sha.8,p.fpd.8, p.obs.21,p.sha.21,p.fpd.21, 
                    nrow = 2, ncol = 3,
                    labels = c("A","","","B","",""),
                    common.legend = TRUE,
                    align = "hv",
                    legend = "top",
                    font.label = list(size = 24, face = "bold"))
p.adiv

p.bdiv <- ggarrange(hf.bray.plot, hf.jac.plot, lf.bray.plot, lf.jac.plot,
                   ncol = 2, nrow = 2,
                   align = "hv",
                   label.x = 0,
                   font.label = list(size = 24, face = "bold"),
                   labels = c("C","D","E","F"))
p.bdiv

p.all <- ggarrange(p.adiv, legend, p.bdiv,
                   ncol = 1, nrow = 3,
                   heights = c(3,0.5,4))
p.all

# Save combined graphics
ggsave(filename = "plots/figures/diversity_ileum_combined_bray_jac.png", p.all, device = "png", dpi = 300, height = 350, width = 350, units = "mm")
ggsave(filename = "plots/figures/diversity_ileum_combined_bray_jac.pdf", p.all, device = "pdf", dpi = 300, height = 350, width = 350, units = "mm")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)[ls(all.names = TRUE) != "params"])
invisible(gc())

```

## Figure A.3: CAECUM DIVERSITY

Figure A.3 (Appendix). Presents alpha- and beta-diversity for day 8 and 21 for Cecum samples.

```{r, eval=TRUE, echo=TRUE}
load("scripts/adiv.Rdata")

load("plots/adiv/alpha_d08/adiv_Cecum_d08.Rdata")
# renaming
p.eve.8 <- p.eve
p.obs.8 <- p.obs
p.sha.8 <- p.sha
p.fpd.8 <- p.fpd
load("plots/adiv/alpha_d21/adiv_Cecum_d21.Rdata")
p.eve.21 <- p.eve
p.obs.21 <- p.obs
p.sha.21 <- p.sha
p.fpd.21 <- p.fpd
rm(p.eve,p.sha,p.obs,p.fpd)

load("plots/bdiv/timeline/Cecum_HF/PCoA_plot_Cecum_HF.Rdata")
#Renaming bdiv HF plots
hf.bray.perm <- bray.perm
hf.bray.plot <- rm_legend(bray.plot)
hf.jac.perm <- jac.perm
hf.jac.plot <- rm_legend(jac.plot)
hf.unif.perm <- unif.perm
hf.unif.plot <- rm_legend(unif.plot)
hf.wuf.perm <- wuf.perm
hf.wuf.plot <- rm_legend(wuf.plot)

# Load bdiv LF plots
load("plots/bdiv/timeline/Cecum_LF/PCoA_plot_Cecum_LF.Rdata")

#Renaming bdiv HF plots
lf.bray.perm <- bray.perm
lf.bray.plot <- rm_legend(bray.plot)
lf.jac.perm <- jac.perm
lf.jac.plot <- rm_legend(jac.plot)
lf.unif.perm <- unif.perm
lf.unif.plot <- rm_legend(unif.plot)
lf.wuf.perm <- wuf.perm
lf.wuf.plot <- rm_legend(wuf.plot)

load("plots/bdiv/timeline/Cecum/PCoA_plot_Cecum.Rdata")

legend <- get_legend(bray.plot)

#remove leftovers
rm(bray.perm,bray.plot,jac.perm,jac.plot,unif.perm,unif.plot,wuf.perm,wuf.plot)

# Call plots
p.adiv <- ggarrange(p.obs.8,p.sha.8,p.fpd.8, p.obs.21,p.sha.21,p.fpd.21, 
                    nrow = 2, ncol = 3,
                    labels = c("A","","","B","",""),
                    common.legend = TRUE,
                    align = "hv",
                    legend = "top",
                    font.label = list(size = 24, face = "bold"))
p.adiv

p.bdiv <- ggarrange(hf.bray.plot, hf.jac.plot, lf.bray.plot, lf.jac.plot,
                   ncol = 2, nrow = 2,
                   align = "hv",
                   label.x = 0,
                   font.label = list(size = 24, face = "bold"),
                   labels = c("C","D","E","F"))
p.bdiv

p.all <- ggarrange(p.adiv, legend, p.bdiv,
                   ncol = 1, nrow = 3,
                   heights = c(3,0.5,4))
p.all

# Save combined graphics
ggsave(filename = "plots/figures/diversity_cecum_combined_bray_jac.png", p.all, device = "png", dpi = 300, height = 350, width = 350, units = "mm")
ggsave(filename = "plots/figures/diversity_cecum_combined_bray_jac.pdf", p.all, device = "pdf", dpi = 300, height = 350, width = 350, units = "mm")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)[ls(all.names = TRUE) != "params"])
invisible(gc())

```

## Figure A.4: Alternative representations
Figure A.4 (Appendix). Presents alpha- and beta-diversity for day 8 to 21 for Faeces samples.
```{r, eval=TRUE, echo=TRUE}
load("plots/bdiv/timeline/Feces_no020/PCoA_plot_Feces_no020.Rdata")

# Call plots
p1 <- ggarrange(bray.treat,jac.treat,bray.diet,jac.diet,
                ncol = 2, nrow = 2,
                align = "v",
                font.label = list(size = 24, face = "bold"),
                labels = c("A","B"),
                common.legend = TRUE)

p1

# Save combined graphics
ggsave(filename = "plots/figures/Figure_A4_Separation_bray_jac.png", p1, device = "png", dpi = 300, height = 180, width = 350, units = "mm")
ggsave(filename = "plots/figures/Figure_A4_Separation_bray_jac.pdf", p1, device = "pdf", dpi = 300, height = 180, width = 350, units = "mm")

# clear the environment and release memory
rm(list = ls(all.names = TRUE)[ls(all.names = TRUE) != "params"])
invisible(gc())

```


# SETTINGS {.tabset .tabset-fade .tabset-pills}

Overview of the parameters and packages that were used for this Rmarkdown.

## PARAMETERS

The following paramenters were set in for this analysis:

```{r parameters, eval=TRUE}
params <- readRDS( "R_objects/params_betadiv.RDS")


tmp <- unlist(params)
dat <- data.frame(Parameter = names(tmp), Value = unname(tmp))


kbl(dat, row.names = F) %>% kable_classic(lightable_options = "striped")

```

## SESSION INFO

The analysis was run in the following environment:

```{r packages, eval=TRUE}
sessionInfo()
```
